
# Goals

- Load and parse results of data from sequencing the amplicons from isolated genomes
    (/nas/FAC/FBM/DMF/pengel/spirit/D2c/aprasad/20211018_aprasad_ApisCrossSpeciesAnalysis/Analysis/20230531_PacBio_Sequencing/SequencingData-Plate2/00-raw_reads_renamed/ESL_strains)
    + Check if there are as many ASVs as expected in the isolate amplicons data
    + Any missing ASVs?
    + Many nonsensical ASVs?
- Load and parse results of long read sequenced full genomes
- Make a database to use to analyse experimental data
- Visualize the results of the priority effects pilot experiment (/nas/FAC/FBM/DMF/pengel/spirit/D2c/aprasad/20220921_aprasad_PriorityEffectsExperimentPilot/03_PilotExperiment/07_PacBIO_sequencing/01_ReadsRenamed)
- Visualize the results of the priority effects experiment

# Load packages and data

```{r}
.libPaths("/work/FAC/FBM/DMF/pengel/spirit/aprasad/miniconda3/envs/pacbio_ampli_r_env/lib/R/library")
source('03_Analysis/Utilities.R', chdir = TRUE)
# # save Rdata
# save.image('03_Analysis/Workspace.Rdata')
load('03_Analysis/Workspace.Rdata')

# note!!
# 1' and 1_rep both refer to the same
# delta* is referred to as m*, m for minus
```

# Analysis

## Compare Isolate genome sequences with amplicons from individual bacterial cultures

### Isolate whole genomes

Specifically the 16S rRNA sequences extracted from the isolate genomes of Lactos sequeced using PacBio and Bifidos by Nanopore by Malick. These sequences are found in `/nas/FAC/FBM/DMF/pengel/spirit/D2c/aprasad/20240399_aprasad_PriorityEffects/00_StrainSelection/Extract_compare_16S/barrnap/*` and handled by the python script that assigns taxonomy to the sequences. The script is, `00_StrainSelection/Extract_compare_16S/correct_counts_16S.py`

```{r}
system('mkdir -p 03_Analysis/IsolateGenomes/')
asv_copies_table <- read.csv("03_Analysis/IsolateGenomes/strain_ASV_copies_table.csv")
heatmap_matrix <- as.matrix(asv_copies_table)
df_info <- as.data.frame(species_name_spec_dict) %>%
            rownames_to_column("strain") %>%
            mutate(spec_name = species_name_spec_dict) %>%
            select(strain, spec_name)
heatmap_data <- read.csv("03_Analysis/IsolateGenomes/strain_ASV_copies_table.csv") %>%
                  select(!X) %>%
                  column_to_rownames("uid") %>%
                    as.matrix()
species_names <- heatmap_data %>% colnames() %>% as.data.frame() %>% rownames_to_column("strain") %>% left_join(df_info, by = c("." = "strain")) %>% pull(spec_name)
genus_names <- unlist(lapply(species_names, function(x) strsplit(x, " ")[[1]][[2]]))
col_split <- species_names %>% as.factor()
# highlight names of low coverage strains in red
genusColors <- c("Bombilactobacillus" = head(colorRampPalette(c(brewer.pal(11, "Spectral")[1], "#FFFFFF"))(10), -1)[1],
                 "Lactobacillus" = head(colorRampPalette(c(brewer.pal(11, "Spectral")[1], "#FFFFFF"))(10), -1)[4],
                 "Bifidobacterium" = brewer.pal(11, "Spectral")[3],
                 "Apibacter" = brewer.pal(11, "Spectral")[4],
                 "Bombella" = brewer.pal(11, "Spectral")[5],
                 "Commensalibacter" = brewer.pal(11, "Spectral")[6],
                 "Bartonella" = brewer.pal(11, "Spectral")[7],
                 "Frischella" = brewer.pal(11, "Spectral")[8],
                 "Apilactobacillus" = brewer.pal(11, "Spectral")[9],
                 "Snodgrassella" = brewer.pal(11, "Spectral")[10],
                 "Gilliamella" = brewer.pal(11, "Spectral")[11]
)
top_anno <- HeatmapAnnotation(Genus = as.character(genus_names),
                                col = list(Genus = genusColors)
                                )
heatmap_obj = Heatmap(heatmap_data,
                      name = "Present",
                      col = colorRamp2(c(0, 1, 2, 3, 4),
                                       c("#ffffff",
                                        "#4393c3",
                                        "#2166ac",
                                        "#fdb863",
                                        "#b2182b"
                                    )
                                ),
                      # heatmap_legend_param = list(at = c(0, 1, 2, 3, 4),
                      #                             legend_height = unit(5, "cm"),
                      #                             labels = c(0, 1, 2, 3, 4)),
                      column_split = col_split,
                      top_annotation = top_anno,
                      # left_annotation = left_anno,
                      cluster_rows = F,
                      cluster_columns = F,
                      column_title_rot = 40,
                      column_title_gp = gpar(fontsize = 12),
                      cluster_row_slices = T,
                      border = T,
                      show_row_names = T,
                      show_heatmap_legend = T)
pdf("/users/aprasad/nas_recherche/20240399_aprasad_PriorityEffects/03_Analysis/IsolateGenomes/strain_ASV_copies_heatmap.pdf", width = 10, height = 15)
draw(heatmap_obj, heatmap_legend_side = "right")
dev.off()
```

```{r}
df_anis <- read.csv("03_Analysis/IsolateGenomes/Compare_genomes/00_Genomes/ani_out.txt", sep = "\t", header = 0) %>%
            mutate(V1 = Vectorize(remove_extension)(V1, "_Assembly.fasta")) %>%
            mutate(V2 = Vectorize(remove_extension)(V2, "_Assembly.fasta")) %>%
            select(V1, V2, V3) %>%
            unique()
colnames(df_anis) <- c("g1", "g2", "ani")

# make a heatmap of ANIs!

df_anis <- df_anis %>%
            # filter(g1 %in% strains_uniq) %>%
            # filter(g2 %in% strains_uniq) %>%
            # mutate(g1 = factor(g1, levels = strains_uniq)) %>%
            # mutate(g2 = factor(g2, levels = strains_uniq)) %>%
            pivot_wider(names_from = g2, values_from = ani) %>%
            column_to_rownames("g1") %>%
            as.matrix()

# make rows and columns same order
df_anis <- df_anis[colnames(df_anis), colnames(df_anis)]

species_names <- df_anis %>% colnames() %>% as.data.frame() %>% rownames_to_column("strain") %>% 
                  left_join(as.data.frame(species_name_spec_dict) %>% rownames_to_column("strain"), by = c("." = "strain")) %>%
                  pull(species_name_spec_dict)

get_genus <- function(species_name) {
  if (species_name == "unknown") {
    return("unknown")
  } else {
    return(strsplit(species_name, " ")[[1]][[1]])
  }
}
                  
genus_names <- df_anis %>% rownames() %>% as.data.frame() %>% rownames_to_column("strain") %>% 
                  left_join(as.data.frame(species_name_spec_dict) %>% rownames_to_column("strain"), by = c("." = "strain")) %>%
                  mutate(genus = Vectorize(get_genus)(species_name_spec_dict)) %>%
                  pull(genus) %>% as.character()

col_split <- species_names %>% as.factor()

# anno color but also names
top_anno <- HeatmapAnnotation(Species = species_names,
                            col = list(Species = speciesColors_uniq)

)

# replace na with 0
df_anis[is.na(df_anis)] <- 70

col_fun <- colorRamp2(
  c(70, 85, 90, 95, 100),
  c("#ffffff", "#9ecae1", "#6baed6", "#1b7837", "#00441b")
)

heatmap_obj = Heatmap(df_anis,
                      name = "ANI",
                      col = col_fun,
                      bottom_annotation = top_anno,
                      # cluster_rows = F,
                      # cluster_columns = F,
                      column_title_rot = 40,
                      # column_split = col_split,
                      column_title_gp = gpar(fontsize = 12),
                      # cluster_row_slices = T,
                      border = F,
                      show_row_names = T,
                      show_heatmap_legend = T)
pdf("/users/aprasad/nas_recherche/20240399_aprasad_PriorityEffects/03_Analysis/IsolateGenomes/ani_heatmap.pdf", width = 10, height = 15)
draw(heatmap_obj, heatmap_legend_side = "right", annotation_legend_side = "bottom")
dev.off()


```

### Isolate amplicons

Strains of interest were cultured and subject to DNA extraction followed by amplicon sequencing as a measure to determine the ASVs present in the culture and a sanity check of the approach. This data was already parsed and run through dada2 in and the result available in `//nas/FAC/FBM/DMF/pengel/spirit/D2c/aprasad/20240399_aprasad_PriorityEffects/Archive/99_Analysis/CustomDada2DB_creation/RDS/dada2.rds`

```{r}
system('mkdir -p 03_Analysis/IsolateAmplicons/00_preprocessing')
dada2_isolate_amplicons <- readRDS("/nas/FAC/FBM/DMF/pengel/spirit/D2c/aprasad/20240399_aprasad_PriorityEffects/03_Analysis/IsolateAmplicons/00_preprocessing/RDS/dada2.rds")
seqtable_isolate_amplicons <- makeSequenceTable(dada2_isolate_amplicons); dim(seqtable_isolate_amplicons)
# only keep ESL strains from the experiment!
strain_to_keep <- c("ESL0825","ESL0820",
"ESL0822","ESL0170",
"ESL0824","ESL0198",
"ESL0827","ESL0199",
"ESL0200","ESL0819",
"ESL0197",
"ESL0294","ESL0394","ESL0295",
"ESL1028",
"ESL0353","ESL0263","ESL0185",
"ESL0183","ESL0262","ESL0835","ESL0354",
"ESL0393","ESL0259","ESL0260","ESL0184","ESL0350",
"ESL0186","ESL0261","ESL0351")
strains_to_keep <- sapply(strain_to_keep, function(x) paste0(x, "_filt_reads.fastq.gz"))
seqtable_isolate_amplicons <- seqtable_isolate_amplicons[rownames(seqtable_isolate_amplicons) %in% strains_to_keep, ]
seqtable_isolate_amplicons_nochim <- removeBimeraDenovo(seqtable_isolate_amplicons, method = "consensus", multithread = TRUE, verbose = TRUE); dim(seqtable_isolate_amplicons_nochim)
write.csv(seqtable_isolate_amplicons_nochim, "03_Analysis/IsolateAmplicons/seqtable_nochim.csv")
system("source ~/.bashrc && conda activate 20230313_scripts_env && python3 03_Analysis/infer_species_counts.py -s 03_Analysis/IsolateAmplicons/seqtable_nochim.csv -o 03_Analysis/IsolateAmplicons/species_table.csv")

# # write sequences
# asv_sequences <- colnames(seqtable_isolate_amplicons_nochim)
# sample_names <- unlist(lapply(rownames(seqtable_isolate_amplicons_nochim), function(x) remove_extension(x, "_filt_reads.fastq.gz")))
# sample_names <- unlist(lapply(sample_names, function(x) paste0("IsolateAmplicon_", x)))
# dna <- Biostrings::DNAStringSet(asv_sequences)
# names(dna) <- c(1:length(asv_sequences))
# # write
# Biostrings::writeXStringSet(dna, "03_Analysis/IsolateAmplicons/asv_sequences.fasta", format = "fasta", width = 10000)


# assign taxonomy
taxonomy_amps <- assignTaxonomy(seqtable_isolate_amplicons_nochim, silva_DB_assignTax, minBoot = 80, multithread=TRUE, verbose = TRUE)
# saveRDS(taxonomy_amps, "03_Analysis/IsolateAmplicons/RDS/taxonomy_before_species_assigned.RDS")
taxonomy_amps <- readRDS("03_Analysis/IsolateAmplicons/RDS/taxonomy_before_species_assigned.RDS")
taxonomy_amps_spec <- addSpecies(taxonomy_amps, silva_DB_assignSpec, allowMultiple = F, verbose = T)
# saveRDS(taxonomy_amps_spec, "03_Analysis/IsolateAmplicons/RDS/taxonomy_spec.RDS")
taxonomy_amps_spec <- readRDS("03_Analysis/IsolateAmplicons/RDS/taxonomy_spec.RDS")
taxonomy_amps_custom <- addSpecies(taxonomy_amps_spec, custom_taxonomy, allowMultiple = F, verbose = T)
saveRDS(taxonomy_amps_custom, "03_Analysis/IsolateAmplicons/RDS/taxonomy_custom.RDS")
write.csv(taxonomy_amps_custom, "03_Analysis/IsolateAmplicons/RDS/taxonomy_custom.csv")
# add counts to taxonomy table
taxonomy_amps_custom <- readRDS("03_Analysis/IsolateAmplicons/RDS/taxonomy_custom.RDS") %>%
                      as.data.frame(.name_repair = "minimal")
colnames(taxonomy_amps_custom) <- c("Kingdom","Phylum","Class","Order","Family","Genus","Species","Species_exact","Strain")
taxonomy_amps_custom_w_counts <- t(seqtable_isolate_amplicons_nochim) %>% as.data.frame %>% rownames_to_column("ASV") %>%
                                left_join(taxonomy_amps_custom %>% rownames_to_column("ASV"), by = c("ASV"))
write.csv(taxonomy_amps_custom_w_counts, "03_Analysis/IsolateAmplicons/RDS/taxonomy_custom_w_counts.csv")

```

### Compare amplicons with predicted ASVs

```{r}
df_info <- as.data.frame(species_name_spec_dict) %>%
            rownames_to_column("strain") %>%
            mutate(spec_name = species_name_spec_dict) %>%
              select(strain, spec_name)

df_strain_ampli <- read.csv("03_Analysis/IsolateAmplicons/species_table_all_asvs.csv", row.names = 1) %>%
                      select(-total) %>%
                      rownames_to_column("sample") %>%
                      mutate(sample = Vectorize(remove_extension)(sample, "_filt_reads.fastq.gz")) %>%
                      pivot_longer(!sample, names_to = "Strain", values_to = "Counts") %>%
                      mutate(Type = "Amplicons") %>%
                      filter(Counts > 0) %>%
                      left_join(df_info %>% select(strain, spec_name), by = c("sample" = "strain"))

df_predicted <- read.csv("03_Analysis/IsolateGenomes/strain_ASV_copies_table.csv") %>%
                  select(!X) %>%
                    pivot_longer(!uid, names_to = "Strain", values_to = "copies") %>%  
                    filter(copies > 0) %>%
                    mutate(sample = Strain) %>%
                    mutate(Strain = paste0(Strain, "_", uid)) %>%
                    filter(sample %in% unique(as.character(df_strain_ampli$sample))) %>%
                    select(!uid) %>%
                    mutate(copies = as.character(copies)) %>%
                    left_join(df_info %>% select(strain, spec_name), by = c("sample" = "strain"))

low_depth_samples <- read.csv("03_Analysis/IsolateAmplicons/species_table_all_asvs.csv", row.names = 1) %>%
                      select(-total) %>% rowSums() %>% data.frame() %>%
                      rownames_to_column("sample") %>%
                      mutate(sample = Vectorize(remove_extension)(sample, "_filt_reads.fastq.gz")) %>%
                      filter(. < 100) %>%
                      pull(sample) %>% as.character()


ggplot() +
        geom_tile(data = df_strain_ampli %>% #filter(!grepl("unk", Strain)) %>%
                          filter(!is.na(spec_name)),
                  aes(x = sample, y = Strain, fill = Counts)) +
        scale_fill_viridis_c(trans = "log10", option = "mako", direction = -1, limits = c(1, 58513)) +
        new_scale_fill() +
        geom_tile(data = df_predicted %>% mutate(Strain = paste0(Strain, "_p")) %>%
                          filter(!is.na(spec_name)),
                  aes(x = sample, y = Strain, fill = copies, alpha = copies), color = "black", linewidth = 1.5) +
        geom_tile(data = df_strain_ampli %>% filter(sample %in% low_depth_samples) %>% filter(!is.na(spec_name)),
                  aes(x = sample, y = Strain),
                  fill = "#999999",
                  color = "#b2182b", linewidth = 1.5) +
        scale_fill_manual(values = c("0" = "#ffffff", "1" = "#66a64e", "2" = "#2166ac", "3" = "#fdb863", "4" = "#b2182b"), drop = FALSE) +
        scale_alpha_manual(values = c("0" = 0, "1" = 1, "2" = 1, "3" = 1, "4" = 1)) +
        labs(x = "Strain",
          y = "Detected / Predicted ASV",
          fill = "Predicted ASV count",
          color = "Predicted ASV count"
          ) +
        # coord_equal() +
        facet_wrap(spec_name ~ ., scales = "free", ncol = 4) +
        make_theme(setFill = F, setCol = F, leg_pos = "bottom", modify_guide = F,
             x_angle = 45, x_hj = 1, x_vj = 1, guide_nrow = 2,
             y_hj = 1)
ggsave("03_Analysis/IsolateAmplicons/HeatmapsCompared/Strains_ASVs.pdf", width = 15, height = 10)

ggplot() +
        geom_tile(data = df_strain_ampli %>% filter(!grepl("unk", Strain)) %>%
                          filter(!is.na(spec_name)),
                  aes(x = sample, y = Strain, fill = Counts)) +
        scale_fill_viridis_c(trans = "log10", option = "mako", direction = -1, limits = c(1, 58513)) +
        new_scale_fill() +
        geom_tile(data = df_predicted %>% mutate(Strain = paste0(Strain, "_p")) %>%
                          filter(!is.na(spec_name)),
                  aes(x = sample, y = Strain, fill = copies, alpha = copies), color = "black", linewidth = 1.5) +
        geom_tile(data = df_strain_ampli %>% filter(sample %in% low_depth_samples) %>% filter(!is.na(spec_name)),
                  aes(x = sample, y = Strain),
                  fill = "#999999",
                  color = "#b2182b", linewidth = 1.5) +
        scale_fill_manual(values = c("0" = "#ffffff", "1" = "#66a64e", "2" = "#2166ac", "3" = "#fdb863", "4" = "#b2182b"), drop = FALSE) +
        scale_alpha_manual(values = c("0" = 0, "1" = 1, "2" = 1, "3" = 1, "4" = 1)) +
        labs(x = "Strain",
          y = "Measured / Predicted ASV count",
          fill = "Measured ASV count",
          color = "Predicted ASV count"
          ) +
        # coord_equal() +
        facet_wrap(spec_name ~ ., scales = "free", ncol = 4) +
        make_theme(setFill = F, setCol = F, leg_pos = "bottom", modify_guide = F,
             x_angle = 45, x_hj = 1, x_vj = 1, guide_nrow = 2,
             y_hj = 1)
ggsave("03_Analysis/IsolateAmplicons/HeatmapsCompared/Strains_ASVs_with_unk.pdf", width = 15, height = 10)

```


## Parse pilot experiment data

For the purpose of this analysis, I will start from the dada2 output generated at `/nas/FAC/FBM/DMF/pengel/spirit/D2c/aprasad/20220921_aprasad_PriorityEffectsExperimentPilot/03_PilotExperiment/07_PacBIO_sequencing/04_Results/RDS/dada2.rds` by `/nas/FAC/FBM/DMF/pengel/spirit/D2c/aprasad/20220921_aprasad_PriorityEffectsExperimentPilot/03_PilotExperiment/07_PacBIO_sequencing/Analysis_and_report.Rmd`

```{r}
dada2_pilot <- readRDS("/nas/FAC/FBM/DMF/pengel/spirit/D2c/aprasad/20220921_aprasad_PriorityEffectsExperimentPilot/03_PilotExperiment/07_PacBIO_sequencing/04_Results/RDS/dada2.rds")
seqtable_pilot <- makeSequenceTable(dada2_pilot); dim(seqtable_pilot)
seqtable_nochim_pilot <- removeBimeraDenovo(seqtable_pilot, method = "consensus", multithread = TRUE, verbose = TRUE); dim(seqtable_pilot)

# asv_sequences <- colnames(seqtable_nochim_pilot)
# sample_names <- unlist(lapply(rownames(seqtable_nochim_pilot), function(x) remove_extension(x, "_filt_reads.fastq.gz")))
# sample_names <- unlist(lapply(sample_names, function(x) paste0("PilotBees_", x)))
# dna <- Biostrings::DNAStringSet(asv_sequences)
# names(dna) <- c(1:length(asv_sequences))
# # write
# Biostrings::writeXStringSet(dna, "03_Analysis/PriorityEffectsPilotExperiment/asv_sequences.fasta", format = "fasta", width = 10000)

system('mkdir -p 03_Analysis/PriorityEffectsPilotExperiment')
write.csv(seqtable_nochim_pilot, "03_Analysis/PriorityEffectsPilotExperiment/seqtable_nochim.csv")
system("source ~/.bashrc && conda activate 20230313_scripts_env && python3 03_Analysis/infer_species_counts.py -s 03_Analysis/PriorityEffectsPilotExperiment/seqtable_nochim.csv -o 03_Analysis/PriorityEffectsPilotExperiment/species_table.csv")


```

### Combine species counts with qPCR data

```{r}
# LOD_plasmid_copies <- 1000
LOD_ct <- 26.53

# get_lod_by_type <- function(my_type="", my_treatment) {
#   if (grepl("bee", my_type) | grepl("alq", my_treatment)) {
#     LOD_copies <- UV_eff^(UV_int - LOD_ct)*elution_vol*dilution_factor_alq
#   } else {
#     LOD_copies <- UV_eff^(UV_int - LOD_ct)*elution_vol*dilution_factor
#   }
#   return(LOD_copies)
# }

source("../20220921_aprasad_PriorityEffectsExperimentPilot/03_PilotExperiment/05_qPCRs/qPCR_parse.R")
df_species_count_e0 <- read.csv("03_Analysis/PriorityEffectsPilotExperiment/species_table.csv", row.names = 1) %>%
                        rownames_to_column("ID") %>%
                        mutate(ID = Vectorize(remove_extension)(ID, "_filt_reads.fastq.gz")) %>%
                        pivot_longer(!ID, names_to = "Strain", values_to = "Counts") %>%
                        group_by(ID) %>%
                        mutate(total_counts = sum(Counts)) %>%
                        mutate(rel_counts = Counts/total_counts) %>%
                        ungroup() %>%
                        left_join(combined_df %>% select(!c(actin_copies, Type, Ct_Mean_host, Ct_SD_host, Mean_control, bac_copies_norm)) %>% unique, by = c("ID")) %>%
                        mutate(abs_counts_org = bac_copies * rel_counts) %>%
                        mutate(limit_passed = ifelse(Ct_Mean_bac > LOD_ct | is.na(Ct_Mean_bac), "No", "Yes")) %>%
                        mutate(limit_passed = as.factor(limit_passed)) %>%
                        mutate(present = ifelse(Counts > 0, 1, 0)) %>%
                        mutate(present = as.factor(present)) %>%
                        # calculate sample threshold and adjust zeros
                        group_by(ID) %>%
                        left_join(df_info %>% select(strain, spec_name), by = c("Strain" = "strain")) %>%
                        mutate(spec_name_uniq = paste0(spec_name, " (", Strain, ")")) %>%
                        mutate(Treatment_simp = Treatment) %>%
                          group_by(ID) %>%
                          mutate(total_reads = sum(Counts, na.rm = T)) %>%
                          # genome equivalents (geq) is the number of copies that corresponds to one read for that sample
                          # this is the ratio of the qpcr copy number and the total sequenced reads
                          mutate(sample_geq = bac_copies/total_reads) %>%
                          ungroup() %>%
                          # any strain having an abundance less than sample geq this is considered not detected
                          # so we can set the abundance of these strains to this value
                          # in plots, we can show the strains that are not detected as the threshold value
                          mutate(abs_counts = ifelse(is.na(abs_counts_org), 0, abs_counts_org)) %>%
                          mutate(geq_passed = ifelse(abs_counts >= sample_geq, 1, 0)) %>%
                          mutate(geq_passed = as.factor(geq_passed))
```

## Process data from the priority effects experiments 1 to 3

```{r}
system("mkdir -p 03_Analysis/PriorityEffectsExperiment")
#  sample_metadata.tsv Was transferred here from the experiment planning information
sample_metadata <- read.csv("03_Analysis/PriorityEffectsExperiment/sample_metadata.tsv", sep = "\t")
```

### Experiment1

#### Create required directories

```{r}
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RawReadsRenamed")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/PrimerRemoved")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/FilteredTrimmed")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/Figures")
```

#### Collect the raw reads

```{r}
sequencing_metadata_e1_1 <- read.csv("03_Analysis/PriorityEffectsExperiment/RawData/Plate1/AP_16S_Pl1_Circular_Consensus_Sequencing_Reads/m84177_241128_121402_s3.barcodes_summary.csv") %>%
                              filter(Sample.Name != "No Name")
sequencing_metadata_e1_2 <- read.csv("03_Analysis/PriorityEffectsExperiment/RawData/Plate2/AP_16S_Pl2_Circular_Consensus_Sequencing_Reads/m84177_241128_121402_s3.barcodes_summary.csv") %>%
                              filter(Sample.Name != "No Name")
read_counts_df_e1 <- rbind(sequencing_metadata_e1 %>% group_by(Sample.Name) %>% summarize(reads = HiFi.Reads), 
                           sequencing_metadata_e1_2 %>% group_by(Sample.Name) %>% summarize(reads = HiFi.Reads))
raw_reads_prefix_plate_1 <- "03_Analysis/PriorityEffectsExperiment/RawData/Plate1/AP_16S_Pl1_Circular_Consensus_Sequencing_Reads/fastx_files/m84177_241128_121402_s3.hifi_reads."
raw_read_files_1 <- as.character(lapply(sequencing_metadata_e1_1$Barcode, function(x) paste0(raw_reads_prefix_plate_1, x, ".hifi_reads.fastq.gz")))
raw_reads_prefix_plate_2 <- "03_Analysis/PriorityEffectsExperiment/RawData/Plate2/AP_16S_Pl2_Circular_Consensus_Sequencing_Reads/fastx_files/m84177_241128_121402_s3.hifi_reads."
raw_read_files_2 <- as.character(lapply(sequencing_metadata_e1_2$Barcode, function(x) paste0(raw_reads_prefix_plate_2, x, ".hifi_reads.fastq.gz")))

rename_barcode_to_sample <- function(x, file_path) {
  x <- gsub("m84177_241128_121402_s3.hifi_reads.", "", x)
  x <- gsub(".hifi_reads.fastq.gz", ".fastq.gz", x)
  if (grepl("Plate1", x)) {
    the_barecode <- strsplit(strsplit(x, "/")[[1]][7], ".fastq")[[1]][1]
    the_sample <- sequencing_metadata_e1_1 %>% filter(Barcode == the_barecode) %>% pull(Sample.Name)
    x <- gsub(the_barecode, the_sample, x)
  } 
  if (grepl("Plate2", x)) {
    the_barecode <- strsplit(strsplit(x, "/")[[1]][7], ".fastq")[[1]][1]
    the_sample <- sequencing_metadata_e1_2 %>% filter(Barcode == the_barecode) %>% pull(Sample.Name)
    x <- gsub(the_barecode, the_sample, x)
  }
  x <- gsub("03_Analysis/PriorityEffectsExperiment/RawData/Plate1/AP_16S_Pl1_Circular_Consensus_Sequencing_Reads/fastx_files", file_path, x)
  x <- gsub("03_Analysis/PriorityEffectsExperiment/RawData/Plate2/AP_16S_Pl2_Circular_Consensus_Sequencing_Reads/fastx_files", file_path, x)
  return(x)
}

raw_file_paths <- c(raw_read_files_1, raw_read_files_2)
raw_file_renamed_paths <- as.character(lapply(raw_file_paths, function(x) rename_barcode_to_sample(x, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RawReadsRenamed")))

for (i in 1:length(raw_file_paths)) {
  if (!file.exists(raw_file_renamed_paths[i])) {
    system(paste("cp", raw_file_paths[i], raw_file_renamed_paths[i]))
  } else {
    print(paste("File already exists:", raw_file_renamed_paths[i]))
  }
}
```

#### Remove primers and reorient reads

First, primers must be removed and files re-oriented. The sequences of the primers can be found https://www.pacb.com/wp-content/uploads/Procedure-checklist-Amplification-of-bacterial-full-length-16S-rRNA-gene-with-barcoded-primers.pdf.

In this step files are also renamed to use ID instead of barcode by way of creating the primer removal output names.

```{r}
primer_remove_in_files <- raw_file_renamed_paths
primer_remove_out_files <- as.character(lapply(primer_remove_in_files, function(x) gsub("/RawReadsRenamed", "/PrimerRemoved", x)))
f_primer <- "AGRGTTYGATYMTGGCTCAG"
r_primer <- "RGYTACCTTGTTACGACTT"
r_primer_comp <- rc(r_primer)
# primer_removal_summary <- removePrimers(primer_remove_in_files, primer_remove_out_files,
#               primer.fwd = f_primer,
#               primer.rev = r_primer_comp,
#               verbose = TRUE)
# saveRDS(primer_removal_summary, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/primer_removal_summary.rds")
primer_removal_summary <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/primer_removal_summary.rds")
```

#### Filter reads based on length

##### Read length distribution before

The adapters but not primers appear to have been removed upon CCS analysis and demultiplexing.
The step to get read lengths takes a long time so save the output and reload for next time.

```{r}
# read_lengths <- lapply(primer_remove_out_files, function(x) nchar(getSequences(x)))
# saveRDS(read_lengths, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/read_lengths.rds")
read_lengths <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/read_lengths.rds")
pdf("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/Figures/Read_length_histogram.pdf", width = 11, height = 8)
hist(do.call(c, read_lengths), 100)
dev.off()
hist(do.call(c, read_lengths), 100)
```

##### Filtering by length

We expect and amplicon size in the range of 1500. So make a cut-off of 1000 minimum and 1600 maximum.
This has already bee performed and will not be repeated when knitting.

```{r}
filt_read_files <- as.character(lapply(primer_remove_out_files, function(x) gsub("/PrimerRemoved", "/FilteredTrimmed", x)))
# trimming_summary <- filterAndTrim(primer_remove_out_files, filt_read_files, minLen=1000, maxLen=1600, rm.phix=FALSE, multithread=T)
# saveRDS(trimming_summary, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/trimming_summary.rds")
trimming_summary <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/trimming_summary.rds")
```

##### Read length distribution after

```{r}
read_lengths_filt <- lapply(filt_read_files, function(x) nchar(getSequences(x)))
saveRDS(read_lengths_filt, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/read_lengths_filt.rds")
read_lengths_filt <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/read_lengths_filt.rds")
pdf("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/Figures/Read_length_filtered_histogram.pdf", width = 11, height = 8)
hist(do.call(c, read_lengths_filt), 100)
dev.off()
hist(do.call(c, read_lengths_filt), 100)
```

#### Dada2 processing

```{r}
pdf("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/Figures/QualityProfilePlot1-5.pdf", width = 11, height = 8)
plotQualityProfile(filt_read_files[1:5])
dev.off()
plotQualityProfile(filt_read_files[1:5])
pdf("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/Figures/QualityProfilePlot20-25.pdf", width = 11, height = 8)
plotQualityProfile(filt_read_files[20:25])
dev.off()
plotQualityProfile(filt_read_files[20:25])

# derepd <- derepFastq(filt_read_files, verbose = T)
# errors <- learnErrors(derepd, errorEstimationFunction=PacBioErrfun, BAND_SIZE=32, multithread=TRUE)
# saveRDS(errors, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/errors.rds")
errors <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/errors.rds")
plotErrors(errors)
# dada2 <- dada(derepd, err=errors, BAND_SIZE=32, multithread=T)
# saveRDS(dada2, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/dada2.rds")
dada2 <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/dada2.rds")
seqtable <- makeSequenceTable(dada2); dim(seqtable)
seqtable_nochim <- removeBimeraDenovo(seqtable, method = "consensus", multithread = TRUE, verbose = TRUE); dim(seqtable)
sum(seqtable_nochim)/sum(seqtable)
hist(as.numeric(lapply(colnames(seqtable_nochim), function(x) nchar(x))))
write.csv(seqtable_nochim, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/seqtable_nochim.csv")
```

#### Summarise pre-processing

```{r}
df_summary <- primer_removal_summary %>% as.data.frame() %>%
                mutate(ID = Vectorize(remove_extension)(rownames(.), "_reads.fastq.gz")) %>%
                   pivot_longer(!ID, values_to = "reads", names_to = "type") %>%
                    mutate(type = ifelse(type == "reads.in", "0_Raw.reads", "1_Primer.removed")) %>%
                      left_join(sample_metadata, by = c("ID" = "ID"))
df_summary_trimming <- trimming_summary %>% as.data.frame %>% 
                                  mutate(ID = Vectorize(remove_extension)(rownames(.), "_primers_removed.fastq.gz")) %>%
                                  pivot_longer(!ID, values_to = "reads", names_to = "type") %>%
                                  mutate(type = ifelse(type == "reads.in", "Trimming.in", "2_Trimmed")) %>%
                                  left_join(sample_metadata, by = c("ID" = "ID")) %>%
                                  filter(type != "Trimming.in")
dada_denoise_summary <- data.frame("V1" = sapply(dada2, function(x) sum(getUniques(x))) %>% as.matrix) %>%
                          mutate("4_Denoised" = V1) %>% select(!V1) %>%
                          cbind("5_No.chimera" = rowSums(seqtable_nochim)) %>%
                          mutate(ID = Vectorize(remove_extension)(rownames(.), "_filt_reads.fastq.gz")) %>%
                            pivot_longer(!ID, values_to = "reads", names_to = "type")
df_summary <- bind_rows(rbind(df_summary, df_summary_trimming), dada_denoise_summary) %>%
                select(ID, type, reads) %>%
                mutate(ID = Vectorize(remove_extension)(ID, ".fastq.gz")) %>%
                left_join(sample_metadata %>% select(ID, Treatment)) %>%
                mutate(Treatment_type = ifelse(grepl("alq", Treatment), "aliquot", Treatment))
saveRDS(df_summary, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/df_summary.rds")
df_summary <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/df_summary.rds")

ggplot() +
  geom_bar(data = df_summary,
           aes(y = ID,
               x = reads,
               fill = type,
               group = type
           ), color = "black",
           stat = "identity", position = "dodge"
  ) +
  facet_wrap(~ Treatment, scale = "free") +
  scale_x_continuous(labels=unit_format(unit = "K", scale = 1e-3)) +
  labs(x = "Number of reads", y = "Sample", fill = "Preprocessing step    ") +
    make_theme(setFill = T, setCol = F, x_angle = 30)
ggsave("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/Figures/summary_of_preprocessing.pdf")
```

#### Species counts

```{r}
system("source ~/.bashrc && conda activate 20230313_scripts_env && python3 03_Analysis/infer_species_counts.py -s 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/seqtable_nochim.csv -o 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/species_table.csv")
```

### Experiment2

#### Create required directories

```{r}
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RawReadsRenamed")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/PrimerRemoved")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/FilteredTrimmed")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/Figures")
```

#### Collect the raw reads

```{r}
system("rsync -aviP ../../../../general_data/D2c/datasets/NGS_data/20250115-PacBioAmpliSeq-Plate3and4/analysis/Demultiplex_Barcodes/PB262_1_pool3_demux/user_biosamples.csv 03_Analysis/PriorityEffectsExperiment/RawData/Plate3/")
system("rsync -aviP ../../../../general_data/D2c/datasets/NGS_data/20250115-PacBioAmpliSeq-Plate3and4/analysis/Demultiplex_Barcodes/PB262_1_pool3_demux/fastx_files 03_Analysis/PriorityEffectsExperiment/RawData/Plate3/")
system("rsync -aviP ../../../../general_data/D2c/datasets/NGS_data/20250115-PacBioAmpliSeq-Plate3and4/analysis/Demultiplex_Barcodes/PB262_1_pool3_demux/*barcodes_summary.csv 03_Analysis/PriorityEffectsExperiment/RawData/Plate3/")

system("rsync -aviP ../../../../general_data/D2c/datasets/NGS_data/20250115-PacBioAmpliSeq-Plate3and4/analysis/Demultiplex_Barcodes/PB262_1_pool4_demux/user_biosamples.csv 03_Analysis/PriorityEffectsExperiment/RawData/Plate4/")
system("rsync -aviP ../../../../general_data/D2c/datasets/NGS_data/20250115-PacBioAmpliSeq-Plate3and4/analysis/Demultiplex_Barcodes/PB262_1_pool4_demux/fastx_files 03_Analysis/PriorityEffectsExperiment/RawData/Plate4/")
system("rsync -aviP ../../../../general_data/D2c/datasets/NGS_data/20250115-PacBioAmpliSeq-Plate3and4/analysis/Demultiplex_Barcodes/PB262_1_pool4_demux/*barcodes_summary.csv 03_Analysis/PriorityEffectsExperiment/RawData/Plate4/")

sequencing_metadata_e2_3 <- read.csv("03_Analysis/PriorityEffectsExperiment/RawData/Plate3/m84151_241230_115705_s1.barcodes_summary.csv") %>%
                              filter(Sample.Name != "No Name")
sequencing_metadata_e2_4 <- read.csv("03_Analysis/PriorityEffectsExperiment/RawData/Plate4/m84151_241230_115705_s1.barcodes_summary.csv") %>%
                              filter(Sample.Name != "No Name")
read_counts_df_e2 <- rbind(sequencing_metadata_e2_3 %>% group_by(Sample.Name) %>% summarize(reads = HiFi.Reads),
                            sequencing_metadata_e2_4 %>% group_by(Sample.Name) %>% summarize(reads = HiFi.Reads))
raw_reads_prefix_plate_3 <- "03_Analysis/PriorityEffectsExperiment/RawData/Plate3/fastx_files/m84151_241230_115705_s1.hifi_reads."
raw_read_files_3 <- as.character(lapply(sequencing_metadata_e2_3$Barcode, function(x) paste0(raw_reads_prefix_plate_3, x, ".hifi_reads.fastq.gz")))
raw_reads_prefix_plate_4 <- "03_Analysis/PriorityEffectsExperiment/RawData/Plate4/fastx_files/m84151_241230_115705_s1.hifi_reads."
raw_read_files_4 <- as.character(lapply(sequencing_metadata_e2_4$Barcode, function(x) paste0(raw_reads_prefix_plate_4, x, ".hifi_reads.fastq.gz")))

rename_barcode_to_sample_e2 <- function(x, file_path) {
  x <- gsub("m84151_241230_115705_s1.hifi_reads.", "", x)
  x <- gsub(".hifi_reads.fastq.gz", ".fastq.gz", x)
  if (grepl("Plate3", x)) {
    the_barecode <- strsplit(strsplit(x, "/")[[1]][6], ".fastq")[[1]][1]
    the_sample <- sequencing_metadata_e2_3 %>% filter(Barcode == the_barecode) %>% pull(Sample.Name)
    x <- gsub(the_barecode, the_sample, x)
  } 
  if (grepl("Plate4", x)) {
    the_barecode <- strsplit(strsplit(x, "/")[[1]][6], ".fastq")[[1]][1]
    the_sample <- sequencing_metadata_e2_4 %>% filter(Barcode == the_barecode) %>% pull(Sample.Name)
    x <- gsub(the_barecode, the_sample, x)
  }
  x <- gsub("03_Analysis/PriorityEffectsExperiment/RawData/Plate3/fastx_files", file_path, x)
  x <- gsub("03_Analysis/PriorityEffectsExperiment/RawData/Plate4/fastx_files", file_path, x)
  return(x)
}

raw_file_paths <- c(raw_read_files_3, raw_read_files_4)
raw_file_renamed_paths <- as.character(lapply(raw_file_paths, function(x) rename_barcode_to_sample_e2(x, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RawReadsRenamed")))

for (i in 1:length(raw_file_paths)) {
  if (!file.exists(raw_file_renamed_paths[i])) {
    system(paste("cp", raw_file_paths[i], raw_file_renamed_paths[i]))
  } else {
    print(paste("File already exists:", raw_file_renamed_paths[i]))
  }
}
```

#### Remove primers and reorient reads

For experiment 2, the primers are the same as in experiment 1. So the same primers will be used for primer removal.

```{r}
primer_remove_in_files <- raw_file_renamed_paths
primer_remove_out_files <- as.character(lapply(primer_remove_in_files, function(x) gsub("/RawReadsRenamed", "/PrimerRemoved", x)))
f_primer <- "AGRGTTYGATYMTGGCTCAG"
r_primer <- "RGYTACCTTGTTACGACTT"
r_primer_comp <- rc(r_primer)
# primer_removal_summary <- removePrimers(primer_remove_in_files, primer_remove_out_files,
#               primer.fwd = f_primer,
#               primer.rev = r_primer_comp,
#               verbose = TRUE)
# saveRDS(primer_removal_summary, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/primer_removal_summary.rds")
primer_removal_summary <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/primer_removal_summary.rds")
```

#### Filter reads based on length

#### Read length distribution before

```{r}
# read_lengths <- lapply(primer_remove_out_files, function(x) nchar(getSequences(x)))
# saveRDS(read_lengths, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/read_lengths.rds")
read_lengths <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/read_lengths.rds")
pdf("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/Figures/Read_length_histogram.pdf", width = 11, height = 8)
hist(do.call(c, read_lengths), 100)
dev.off()
hist(do.call(c, read_lengths), 100)
```

#### Filtering by length

```{r}
filt_read_files <- as.character(lapply(primer_remove_out_files, function(x) gsub("/PrimerRemoved", "/FilteredTrimmed", x)))
# trimming_summary <- filterAndTrim(primer_remove_out_files, filt_read_files, minLen=1000, maxLen=1600, rm.phix=FALSE, multithread=T)
# saveRDS(trimming_summary, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/trimming_summary.rds")
trimming_summary <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/trimming_summary.rds")
```

#### Read length distribution after

```{r}
read_lengths_filt <- lapply(filt_read_files, function(x) nchar(getSequences(x)))
saveRDS(read_lengths_filt, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/read_lengths_filt.rds")
read_lengths_filt <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/read_lengths_filt.rds")
pdf("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/Figures/Read_length_filtered_histogram.pdf", width = 11, height = 8)
hist(do.call(c, read_lengths_filt), 100)
dev.off()
hist(do.call(c, read_lengths_filt), 100)
```

#### Dada2 processing

```{r}
pdf("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/Figures/QualityProfilePlot1-5.pdf", width = 11, height = 8)
plotQualityProfile(filt_read_files[1:5])
dev.off()
plotQualityProfile(filt_read_files[1:5])
pdf("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/Figures/QualityProfilePlot20-25.pdf", width = 11, height = 8)
plotQualityProfile(filt_read_files[20:25])
dev.off()
plotQualityProfile(filt_read_files[20:25])

# derepd <- derepFastq(filt_read_files, verbose = T)
# errors <- learnErrors(derepd, errorEstimationFunction=PacBioErrfun, BAND_SIZE=32, multithread=TRUE)
# saveRDS(errors, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/errors.rds")
errors <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/errors.rds")
# plotErrors(errors)
# dada2 <- dada(derepd, err=errors, BAND_SIZE=32, multithread=T)
# saveRDS(dada2, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/dada2.rds")
dada2 <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/dada2.rds")
seqtable <- makeSequenceTable(dada2); dim(seqtable)
seqtable_nochim <- removeBimeraDenovo(seqtable, method = "consensus", multithread = TRUE, verbose = TRUE); dim(seqtable)
sum(seqtable_nochim)/sum(seqtable)
hist(as.numeric(lapply(colnames(seqtable_nochim), function(x) nchar(x))))
write.csv(seqtable_nochim, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/seqtable_nochim.csv")
```

#### Summarise pre-processing

```{r}
df_summary <- primer_removal_summary %>% as.data.frame() %>%
                mutate(ID = Vectorize(remove_extension)(rownames(.), "_reads.fastq.gz")) %>%
                   pivot_longer(!ID, values_to = "reads", names_to = "type") %>%
                    mutate(type = ifelse(type == "reads.in", "0_Raw.reads", "1_Primer.removed")) %>%
                      left_join(sample_metadata, by = c("ID" = "ID"))
df_summary_trimming <- trimming_summary %>% as.data.frame %>% 
                                  mutate(ID = Vectorize(remove_extension)(rownames(.), "_primers_removed.fastq.gz")) %>%
                                  pivot_longer(!ID, values_to = "reads", names_to = "type") %>%
                                  mutate(type = ifelse(type == "reads.in", "Trimming.in", "2_Trimmed")) %>%
                                  left_join(sample_metadata, by = c("ID" = "ID")) %>%
                                  filter(type != "Trimming.in")
dada_denoise_summary <- data.frame("V1" = sapply(dada2, function(x) sum(getUniques(x))) %>% as.matrix) %>%
                          mutate("4_Denoised" = V1) %>% select(!V1) %>%
                          cbind("5_No.chimera" = rowSums(seqtable_nochim)) %>%
                          mutate(ID = Vectorize(remove_extension)(rownames(.), "_filt_reads.fastq.gz")) %>%
                            pivot_longer(!ID, values_to = "reads", names_to = "type")
df_summary <- bind_rows(rbind(df_summary, df_summary_trimming), dada_denoise_summary) %>%
                select(ID, type, reads) %>%
                mutate(ID = Vectorize(remove_extension)(ID, ".fastq.gz")) %>%
                left_join(sample_metadata %>% select(ID, Treatment)) %>%
                mutate(Treatment_type = ifelse(grepl("alq", Treatment), "aliquot", Treatment))
saveRDS(df_summary, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/df_summary.rds")
df_summary <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/df_summary.rds")

ggplot() +
  geom_bar(data = df_summary,
           aes(y = ID,
               x = reads,
               fill = type,
               group = type
           ), color = "black",
           stat = "identity", position = "dodge"
  ) +
  facet_wrap(~ Treatment, scale = "free") +
  scale_x_continuous(labels=unit_format(unit = "K", scale = 1e-3)) +
  labs(x = "Number of reads", y = "Sample", fill = "Preprocessing step    ") +
    make_theme(setFill = T, setCol = F, x_angle = 30)
ggsave("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/Figures/summary_of_preprocessing.pdf")
```

#### Species counts

```{r}
system("source ~/.bashrc && conda activate 20230313_scripts_env && python3 03_Analysis/infer_species_counts.py -s 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/seqtable_nochim.csv -o 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/species_table.csv")
```

### Experiment3

#### Create required directories

```{r}
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RawReadsRenamed")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/PrimerRemoved")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/FilteredTrimmed")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS")
system("mkdir -p 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/Figures")
```

#### Collect the raw reads

```{r}
system("rsync -aviP ../../../../general_data/D2c/datasets/NGS_data/20250115-PacBioAmpliSeq-Plate5and6/analysis/Demultiplex_Barcodes/PB262_2_pool5_demux/user_biosamples.csv 03_Analysis/PriorityEffectsExperiment/RawData/Plate5/")
system("rsync -aviP ../../../../general_data/D2c/datasets/NGS_data/20250115-PacBioAmpliSeq-Plate5and6/analysis/Demultiplex_Barcodes/PB262_2_pool5_demux/fastx_files 03_Analysis/PriorityEffectsExperiment/RawData/Plate5/")
system("rsync -aviP ../../../../general_data/D2c/datasets/NGS_data/20250115-PacBioAmpliSeq-Plate5and6/analysis/Demultiplex_Barcodes/PB262_2_pool5_demux/*barcodes_summary.csv 03_Analysis/PriorityEffectsExperiment/RawData/Plate5/")

system("rsync -aviP ../../../../general_data/D2c/datasets/NGS_data/20250115-PacBioAmpliSeq-Plate5and6/analysis/Demultiplex_Barcodes/PB262_2_pool6_demux/user_biosamples.csv 03_Analysis/PriorityEffectsExperiment/RawData/Plate6/")
system("rsync -aviP ../../../../general_data/D2c/datasets/NGS_data/20250115-PacBioAmpliSeq-Plate5and6/analysis/Demultiplex_Barcodes/PB262_2_pool6_demux/fastx_files 03_Analysis/PriorityEffectsExperiment/RawData/Plate6/")
system("rsync -aviP ../../../../general_data/D2c/datasets/NGS_data/20250115-PacBioAmpliSeq-Plate5and6/analysis/Demultiplex_Barcodes/PB262_2_pool6_demux/*barcodes_summary.csv 03_Analysis/PriorityEffectsExperiment/RawData/Plate6/")

sequencing_metadata_e3_5 <- read.csv("03_Analysis/PriorityEffectsExperiment/RawData/Plate5/m84151_241230_135808_s2.barcodes_summary.csv") %>%
                              filter(Sample.Name != "No Name")
sequencing_metadata_e3_6 <- read.csv("03_Analysis/PriorityEffectsExperiment/RawData/Plate6/m84151_241230_135808_s2.barcodes_summary.csv") %>%
                              filter(Sample.Name != "No Name")
read_counts_df_e3 <- rbind(sequencing_metadata_e3_5 %>% group_by(Sample.Name) %>% summarize(reads = HiFi.Reads),
                            sequencing_metadata_e3_6 %>% group_by(Sample.Name) %>% summarize(reads = HiFi.Reads))
raw_reads_prefix_plate_5 <- "03_Analysis/PriorityEffectsExperiment/RawData/Plate5/fastx_files/m84151_241230_135808_s2.hifi_reads."
raw_read_files_5 <- as.character(lapply(sequencing_metadata_e3_5$Barcode, function(x) paste0(raw_reads_prefix_plate_5, x, ".hifi_reads.fastq.gz")))
raw_reads_prefix_plate_6 <- "03_Analysis/PriorityEffectsExperiment/RawData/Plate6/fastx_files/m84151_241230_135808_s2.hifi_reads."
raw_read_files_6 <- as.character(lapply(sequencing_metadata_e3_6$Barcode, function(x) paste0(raw_reads_prefix_plate_6, x, ".hifi_reads.fastq.gz")))

rename_barcode_to_sample_e3 <- function(x, file_path) {
  x <- gsub("m84151_241230_135808_s2.hifi_reads.", "", x)
  x <- gsub(".hifi_reads.fastq.gz", ".fastq.gz", x)
  if (grepl("Plate5", x)) {
    the_barecode <- strsplit(strsplit(x, "/")[[1]][6], ".fastq")[[1]][1]
    the_sample <- sequencing_metadata_e3_5 %>% filter(Barcode == the_barecode) %>% pull(Sample.Name)
    x <- gsub(the_barecode, the_sample, x)
  } 
  if (grepl("Plate6", x)) {
    the_barecode <- strsplit(strsplit(x, "/")[[1]][6], ".fastq")[[1]][1]
    the_sample <- sequencing_metadata_e3_6 %>% filter(Barcode == the_barecode) %>% pull(Sample.Name)
    x <- gsub(the_barecode, the_sample, x)
  }
  x <- gsub("03_Analysis/PriorityEffectsExperiment/RawData/Plate5/fastx_files", file_path, x)
  x <- gsub("03_Analysis/PriorityEffectsExperiment/RawData/Plate6/fastx_files", file_path, x)
  return(x)
}

raw_file_paths <- c(raw_read_files_5, raw_read_files_6)
raw_file_renamed_paths <- as.character(lapply(raw_file_paths, function(x) rename_barcode_to_sample_e3(x, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RawReadsRenamed")))

for (i in 1:length(raw_file_paths)) {
  if (!file.exists(raw_file_renamed_paths[i])) {
    system(paste("cp", raw_file_paths[i], raw_file_renamed_paths[i]))
  } else {
    print(paste("File already exists:", raw_file_renamed_paths[i]))
  }
}
```

#### Remove primers and reorient reads

For experiment 3, the primers are the same as in experiment 1. So the same primers will be used for primer removal.

```{r}
primer_remove_in_files <- raw_file_renamed_paths
primer_remove_out_files <- as.character(lapply(primer_remove_in_files, function(x) gsub("/RawReadsRenamed", "/PrimerRemoved", x)))
f_primer <- "AGRGTTYGATYMTGGCTCAG"
r_primer <- "RGYTACCTTGTTACGACTT"
r_primer_comp <- rc(r_primer)
# primer_removal_summary <- removePrimers(primer_remove_in_files, primer_remove_out_files,
#               primer.fwd = f_primer,
#               primer.rev = r_primer_comp,
#               verbose = TRUE)
# saveRDS(primer_removal_summary, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/primer_removal_summary.rds")
primer_removal_summary <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/primer_removal_summary.rds")
```

#### Filter reads based on length

#### Read length distribution before

```{r}
# read_lengths <- lapply(primer_remove_out_files, function(x) nchar(getSequences(x)))
# saveRDS(read_lengths, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/read_lengths.rds")
read_lengths <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/read_lengths.rds")
pdf("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/Figures/Read_length_histogram.pdf", width = 11, height = 8)
hist(do.call(c, read_lengths), 100)
dev.off()
hist(do.call(c, read_lengths), 100)
```

##### Filtering by length

```{r}
filt_read_files <- as.character(lapply(primer_remove_out_files, function(x) gsub("/PrimerRemoved", "/FilteredTrimmed", x)))
# trimming_summary <- filterAndTrim(primer_remove_out_files, filt_read_files, minLen=1000, maxLen=1600, rm.phix=FALSE, multithread=T)
# saveRDS(trimming_summary, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/trimming_summary.rds")
trimming_summary <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/trimming_summary.rds")
```

#### Read length distribution after

```{r}
read_lengths_filt <- lapply(filt_read_files, function(x) nchar(getSequences(x)))
saveRDS(read_lengths_filt, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/read_lengths_filt.rds")
read_lengths_filt <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/read_lengths_filt.rds")
pdf("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/Figures/Read_length_filtered_histogram.pdf", width = 11, height = 8)
hist(do.call(c, read_lengths_filt), 100)
dev.off()
hist(do.call(c, read_lengths_filt), 100)
```

#### Dada2 processing

```{r}
pdf("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/Figures/QualityProfilePlot1-5.pdf", width = 11, height = 8)
plotQualityProfile(filt_read_files[1:5])
dev.off()
plotQualityProfile(filt_read_files[1:5])
pdf("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/Figures/QualityProfilePlot20-25.pdf", width = 11, height = 8)
plotQualityProfile(filt_read_files[20:25])
dev.off()
plotQualityProfile(filt_read_files[20:25])

# derepd <- derepFastq(filt_read_files, verbose = T)
# errors <- learnErrors(derepd, errorEstimationFunction=PacBioErrfun, BAND_SIZE=32, multithread=TRUE)
# saveRDS(errors, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/errors.rds")
errors <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/errors.rds")
plotErrors(errors)
# dada2 <- dada(derepd, err=errors, BAND_SIZE=32, multithread=T)
# saveRDS(dada2, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/dada2.rds")
dada2 <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/dada2.rds")
seqtable <- makeSequenceTable(dada2); dim(seqtable)
seqtable_nochim <- removeBimeraDenovo(seqtable, method = "consensus", multithread = TRUE, verbose = TRUE); dim(seqtable)
sum(seqtable_nochim)/sum(seqtable)
hist(as.numeric(lapply(colnames(seqtable_nochim), function(x) nchar(x))))
write.csv(seqtable_nochim, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/seqtable_nochim.csv")
```

#### Summarise pre-processing

```{r}
df_summary <- primer_removal_summary %>% as.data.frame() %>%
                mutate(ID = Vectorize(remove_extension)(rownames(.), "_reads.fastq.gz")) %>%
                   pivot_longer(!ID, values_to = "reads", names_to = "type") %>%
                    mutate(type = ifelse(type == "reads.in", "0_Raw.reads", "1_Primer.removed")) %>%
                      left_join(sample_metadata, by = c("ID" = "ID"))
df_summary_trimming <- trimming_summary %>% as.data.frame %>% 
                                  mutate(ID = Vectorize(remove_extension)(rownames(.), "_primers_removed.fastq.gz")) %>%
                                  pivot_longer(!ID, values_to = "reads", names_to = "type") %>%
                                  mutate(type = ifelse(type == "reads.in", "Trimming.in", "2_Trimmed")) %>%
                                  left_join(sample_metadata, by = c("ID" = "ID")) %>%
                                  filter(type != "Trimming.in")
dada_denoise_summary <- data.frame("V1" = sapply(dada2, function(x) sum(getUniques(x))) %>% as.matrix) %>%
                          mutate("4_Denoised" = V1) %>% select(!V1) %>%
                          cbind("5_No.chimera" = rowSums(seqtable_nochim)) %>%
                          mutate(ID = Vectorize(remove_extension)(rownames(.), "_filt_reads.fastq.gz")) %>%
                            pivot_longer(!ID, values_to = "reads", names_to = "type")
df_summary <- bind_rows(rbind(df_summary, df_summary_trimming), dada_denoise_summary) %>%
                select(ID, type, reads) %>%
                mutate(ID = Vectorize(remove_extension)(ID, ".fastq.gz")) %>%
                left_join(sample_metadata %>% select(ID, Treatment)) %>%
                mutate(Treatment_type = ifelse(grepl("alq", Treatment), "aliquot", Treatment))
saveRDS(df_summary, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/df_summary.rds")
df_summary <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/df_summary.rds")

ggplot() +
  geom_bar(data = df_summary,
           aes(y = ID,
               x = reads,
               fill = type,
               group = type
           ), color = "black",
           stat = "identity", position = "dodge"
  ) +
  facet_wrap(~ Treatment, scale = "free") +
  scale_x_continuous(labels=unit_format(unit = "K", scale = 1e-3)) +
  labs(x = "Number of reads", y = "Sample", fill = "Preprocessing step    ") +
    make_theme(setFill = T, setCol = F, x_angle = 30)
ggsave("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/Figures/summary_of_preprocessing.pdf")
```

#### Species counts

```{r}
system("source ~/.bashrc && conda activate 20230313_scripts_env && python3 03_Analysis/infer_species_counts.py -s 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/seqtable_nochim.csv -o 03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/species_table.csv")
```

### Get qPCR copy numbers

```{r}
source("03_Analysis/qpcr_data_parse.R") # issue with "source"ing but open the file and run, gives no errors
df_qpcr_counts <- read.csv("03_Analysis/PriorityEffectsExperiment/01_qpcrdata/qPCR_counts.csv")
```

### Combine species counts and qPCR counts

```{r}
df_species_count_e1 <- read.csv("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/species_table.csv", row.names = 1) %>%
                        rownames_to_column("ID") %>%
                        mutate(ID = Vectorize(remove_extension)(ID, ".fastq.gz")) %>%
                        pivot_longer(!ID, names_to = "Strain", values_to = "Counts") %>%
                        group_by(ID) %>%
                        mutate(total_counts = sum(Counts)) %>%
                        mutate(rel_counts = Counts/total_counts) %>%
                        ungroup() %>%
                        left_join(df_qpcr_counts %>% select(ID, bac_copies, bac_copies_raw, Ct_Mean_bac, Ct_SD_bac), by = c("ID")) %>%
                        mutate(abs_counts_org = bac_copies * rel_counts) %>%
                        mutate(limit_passed = ifelse(Ct_Mean_bac > LOD_ct | is.na(Ct_Mean_bac), "No", "Yes")) %>%
                        mutate(limit_passed = as.factor(limit_passed)) %>%
                        left_join(sample_metadata, by = c("ID" = "ID")) %>%
                        mutate(present = ifelse(Counts > 0, 1, 0)) %>%
                        mutate(present = as.factor(present)) %>%
                        left_join(df_info %>% select(strain, spec_name), by = c("Strain" = "strain")) %>%
                        mutate(spec_name_uniq = paste0(spec_name, " (", Strain, ")"))
df_species_count_e2 <- read.csv("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/species_table.csv", row.names = 1) %>%
                        rownames_to_column("ID") %>%
                        mutate(ID = Vectorize(remove_extension)(ID, ".fastq.gz")) %>%
                        pivot_longer(!ID, names_to = "Strain", values_to = "Counts") %>%
                        group_by(ID) %>%
                        mutate(total_counts = sum(Counts)) %>%
                        mutate(rel_counts = Counts/total_counts) %>%
                        ungroup() %>%
                        left_join(df_qpcr_counts %>% select(ID, bac_copies, bac_copies_raw, Ct_Mean_bac, Ct_SD_bac), by = c("ID")) %>%
                        mutate(abs_counts_org = bac_copies * rel_counts) %>%
                        mutate(limit_passed = ifelse(Ct_Mean_bac > LOD_ct | is.na(Ct_Mean_bac), "No", "Yes")) %>%
                        mutate(limit_passed = as.factor(limit_passed)) %>%
                        left_join(sample_metadata, by = c("ID" = "ID")) %>%
                        mutate(present = ifelse(Counts > 0, 1, 0)) %>%
                        mutate(present = as.factor(present)) %>%
                        left_join(df_info %>% select(strain, spec_name), by = c("Strain" = "strain")) %>%
                        mutate(spec_name_uniq = paste0(spec_name, " (", Strain, ")"))
df_species_count_e3 <- read.csv("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/species_table.csv", row.names = 1) %>%
                        rownames_to_column("ID") %>%
                        mutate(ID = Vectorize(remove_extension)(ID, ".fastq.gz")) %>%
                        pivot_longer(!ID, names_to = "Strain", values_to = "Counts") %>%
                        group_by(ID) %>%
                        mutate(total_counts = sum(Counts)) %>%
                        mutate(rel_counts = Counts/total_counts) %>%
                        ungroup() %>%
                        left_join(df_qpcr_counts %>% select(ID, bac_copies, bac_copies_raw, Ct_Mean_bac, Ct_SD_bac), by = c("ID")) %>%
                        mutate(abs_counts_org = bac_copies * rel_counts) %>%
                        mutate(limit_passed = ifelse(Ct_Mean_bac > LOD_ct | is.na(Ct_Mean_bac), "No", "Yes")) %>%
                        mutate(limit_passed = as.factor(limit_passed)) %>%
                        left_join(sample_metadata, by = c("ID" = "ID")) %>%
                        mutate(present = ifelse(Counts > 0, 1, 0)) %>%
                        mutate(present = as.factor(present)) %>%
                        left_join(df_info %>% select(strain, spec_name), by = c("Strain" = "strain")) %>%
                        mutate(spec_name_uniq = paste0(spec_name, " (", Strain, ")"))
df_species_count_all <- bind_rows(df_species_count_e1, df_species_count_e2, df_species_count_e3)

# report on strains that were in the database but never detected! These are removed from further analysis
df_species_count_all %>%
  group_by(Strain) %>%
  summarize(sum(Counts)) %>% print(n = 50)

# for the aliqupts, the actual copies fed to bees is 5/165 * qPCR_counts = 0.0303 * qPCR_counts
df_species_count_all <- df_species_count_all %>%
                          mutate(abs_counts_org = ifelse(Type == "aliquot", 0.0303 * abs_counts_org, abs_counts_org)) %>%
                          mutate(abs_counts_org = ifelse(Type == "bacteria", 0.0303 * abs_counts_org, abs_counts_org)) %>%
                          # Type is bee for honeybee samples not additional fraction to calculate - extraction was done same way for
                          # aliquots as for guts except instead of gut homogenate the source material was 165 uL of the aliquot fed
                          mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment)) %>%
                          group_by(ID) %>%
                          mutate(total_reads = sum(Counts, na.rm = T)) %>%
                          # genome equivalents (geq) is the number of copies that corresponds to one read for that sample
                          # this is the ratio of the qpcr copy number and the total sequenced reads
                          mutate(sample_geq = bac_copies/total_reads) %>%
                          ungroup() %>%
                          # any strain having an abundance less than sample geq this is considered not detected
                          # so we can set the abundance of these strains to this value
                          # in plots, we can show the strains that are not detected as the threshold value
                          mutate(abs_counts = ifelse(is.na(abs_counts_org), 0, abs_counts_org)) %>%
                          mutate(geq_passed = ifelse(abs_counts >= sample_geq, 1, 0)) %>%
                          mutate(geq_passed = as.factor(geq_passed)) %>%
                          group_by(Strain) %>%
                          mutate(strain_sum = sum(Counts)) %>%
                          filter(strain_sum > 0) %>%
                          select(-strain_sum) %>%
                          ungroup()

# get samples that were too low to detect anything in lower than LOD - water gets signal higher - filter
samples_low_detection <- df_species_count_all %>%
                          group_by(ID) %>%
                          mutate(known_reads = sum(Counts[!Strain %in% c("unknown")])) %>%
                          ungroup() %>%
                          filter(limit_passed == "No" | is.na(limit_passed) | known_reads < 100) %>%
                          filter(Type == "bee") %>%
                          filter(!Treatment %in% c("MD", "extraction_blank")) %>%
                          pull(ID) %>% unique

write.csv(samples_low_detection, "03_Analysis/PriorityEffectsExperiment/samples_low_detection.csv", row.names = F)

#

# write the data out
# 1-23 and 1-12 are incorrect in metadata column treatment!
df_species_count_e0_mod <- df_species_count_e0 %>% 
                            mutate(Type = ifelse(Sample_type == "bee_gut", "bee", Sample_type)) %>%
                            mutate(Type = ifelse(Sample_type == "community_alq", "aliquot", Type)) %>%
                            mutate(Experiment = 'pilot') %>%
                            mutate(CommunityCombo = 'pilot') %>%
                            mutate(Notes = 'pilot') %>%
                            mutate(Treatment = ifelse(ID == "1-24", "A-3_14", Treatment)) %>%
                            mutate(Treatment = ifelse(ID == "1-23", "B-3_14", Treatment)) %>%
                            select(all_of(df_species_count_all %>% names()))

low_depth_samples_pilot <- df_species_count_e0_mod %>%
                          filter(limit_passed == "No" | is.na(limit_passed)) %>%
                          filter(Type == "bee") %>%
                          filter(!Treatment %in% c("MD", "extraction_blank")) %>%
                          pull(ID) %>% unique

df_species_count_all_combined <- rbind(df_species_count_e0_mod %>% filter(!ID %in% low_depth_samples_pilot), df_species_count_all) %>%
                        mutate(arrival = Vectorize(get_first_second_only_arriver)(Treatment, Strain)) %>%
                        mutate(arrival = ifelse(Treatment == "0-3", "MD", arrival)) %>%
                        mutate(CommunityCombo = (ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)))


df_species_count_all_combined %>%
  filter(Type == "bee") %>%
  group_by(ID) %>%
  mutate(known_reads = sum(Counts[!Strain %in% c("unknown")])) %>%
  mutate(unknown_reads = sum(Counts[Strain %in% c("unknown")])) %>%
  summarize(Treatment, Type, total_reads, known_reads, unknown_reads) %>%
  unique() %>%
  filter(!ID %in% samples_low_detection) %>%
  filter(Treatment != "MD") %>%
  mutate(unknown_perc = unknown_reads/total_reads*100) %>% pull(unknown_perc) %>%
  quantile(., 0.9)



# make a column called arrival mod for the sake of melliventris strains alone
# same as arrival column except that for the 394 anf the other melliventris strain that it comes with, append a modifier "2"
# do this "by hand"

A_first_treatments <- c("A1-B1", "A2-B2", "A3-B3", "A4-B4", "A5-B5", "A6-B6")
A_only_treatments <- c("A1-0", "A2-0", "A3-0", "A4-0", "A5-0", "A6-0")
B_first_treatments <- c("B1-A1", "B2-A2", "B3-A3", "B4-A4", "B5-A5", "B6-A6")
B_only_treatments <- c("B1-0", "B2-0", "B3-0", "B4-0", "B5-0", "B6-0")

df_species_count_all_combined_mod <- df_species_count_all_combined %>%
                                      mutate(arrival_mod = arrival) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0394" & Treatment %in% A_first_treatments & CommunityCombo %in% c("1", "2", "4", "5", "6"), "second_2", arrival_mod)) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0394" & Treatment %in% A_first_treatments & CommunityCombo %in% c("3"), "first_2", arrival_mod)) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0350" & Treatment %in% A_first_treatments & CommunityCombo %in% c("1", "2", "5", "6"), "second_2", arrival_mod)) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0350" & Treatment %in% A_first_treatments & CommunityCombo %in% c("3"), "first_2", arrival_mod)) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0184" & Treatment %in% A_first_treatments & CommunityCombo %in% c("4"), "second_2", arrival_mod)) %>%
                                      # then
                                      mutate(arrival_mod = ifelse(Strain == "ESL0394" & Treatment %in% A_only_treatments & CommunityCombo %in% c("1", "2", "3", "4", "5", "6"), "only_2", arrival_mod)) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0350" & Treatment %in% A_only_treatments & CommunityCombo %in% c("1", "2", "3", "5", "6"), "only_2", arrival_mod)) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0184" & Treatment %in% A_only_treatments & CommunityCombo %in% c("4"), "only_2", arrival_mod)) %>%
                                      # then
                                      mutate(arrival_mod = ifelse(Strain == "ESL0394" & Treatment %in% B_first_treatments & CommunityCombo %in% c("1", "2", "4", "5", "6"), "first_2", arrival_mod)) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0394" & Treatment %in% B_first_treatments & CommunityCombo %in% c("3"), "second_2", arrival_mod)) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0350" & Treatment %in% B_first_treatments & CommunityCombo %in% c("1", "2", "5", "6"), "first_2", arrival_mod)) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0350" & Treatment %in% B_first_treatments & CommunityCombo %in% c("3"), "second_2", arrival_mod)) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0184" & Treatment %in% B_first_treatments & CommunityCombo %in% c("4"), "first_2", arrival_mod)) %>%
                                      # then
                                      mutate(arrival_mod = ifelse(Strain == "ESL0394" & Treatment %in% B_only_treatments & CommunityCombo %in% c("1", "2", "3", "4", "5", "6"), "only_2", arrival_mod)) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0350" & Treatment %in% B_only_treatments & CommunityCombo %in% c("1", "2", "3", "5", "6"), "only_2", arrival_mod)) %>%
                                      mutate(arrival_mod = ifelse(Strain == "ESL0184" & Treatment %in% B_only_treatments & CommunityCombo %in% c("4"), "only_2", arrival_mod))

# system("mkdir -p 03_Analysis/Results/tables")
write.csv(rbind(df_species_count_e0_mod, df_species_count_all) %>%
                        mutate(is_low = ifelse(ID %in% samples_low_detection, "Yes", "No")) %>%
                        mutate(arrival = Vectorize(get_first_second_only_arriver)(Treatment, Strain)) %>%
                        mutate(arrival = ifelse(Treatment == "0-3", "MD", arrival)) %>%
                        mutate(CommunityCombo = (ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo))), "03_Analysis/Results/tables/df_species_count_all_combined.csv", row.names = F, quote = F)
write.csv(df_species_count_all_combined_mod, "03_Analysis/Results/tables/df_species_count_all_combined_mod.csv", row.names = F, quote = F)
```


```{r}
df_total_reads_compare <- df_species_count_all %>%
                          group_by(ID) %>%
                          filter(!CommunityCombo %in% c("m2", "m3", "m5", "m8", "m9", "m10")) %>%
                          mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment)) %>%
                          summarize(sample_geq, bac_copies, Type, CommunityCombo,Treatment_simp, total_reads = sum(Counts), total_abs = sum(abs_counts)) %>%
                          unique()


ggplot() +
  geom_bar(data = df_total_reads_compare,
        aes(x = ID, y = total_abs, fill = Treatment_simp),
        stat = "identity", position = "stack") +
  geom_bar(data = df_total_reads_compare,
        aes(x = ID, y = sample_geq, fill = Treatment_simp), color = "#000000", alpha = 0.5, stat = "identity") +
  geom_bar(data = df_total_reads_compare %>% filter(!ID %in% samples_low_detection),
        aes(x = ID, y = total_reads), fill = "#ffffff", alpha = 0.5, stat = "identity") +
  geom_bar(data = df_total_reads_compare %>% filter(ID %in% samples_low_detection),
        aes(x = ID, y = total_reads), fill = "#000000", stat = "identity") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  geom_hline(yintercept = median(df_total_reads_compare$sample_geq, na.rm = T), linetype = "solid") +
  geom_hline(yintercept = 1e+03, linetype = "dashed") +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  facet_wrap(~CommunityCombo, scale = "free_x", ncol = 4) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 7,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/total_reads_geq_and_abundance.pdf", width = 11, height = 8)


ggplot() +
  # geom_bar(data = df_species_count_all %>% filter(!ID %in% samples_low_detection) %>%
  #                   filter(!Strain %in% c("unknown")) %>%
  #                   mutate(facet_category = Type) %>%
  #                   mutate(facet_category = ifelse(facet_category == "bee", paste0("0. ", facet_category, "(", Treatment_simp, ")"), facet_category)) %>%
  #                   mutate(facet_category = ifelse(facet_category == "bacteria", "1. bacteria", facet_category)) %>%
  #                   mutate(facet_category = ifelse(facet_category == "aliquot", "1. bacteria", facet_category)) %>%
  #                   mutate(facet_category = ifelse(facet_category == "blank", "2. blank", facet_category)) %>%
  #                   mutate(facet_category = ifelse(Treatment == "extraction_blank", "2. blank", facet_category)) %>%
  #                   mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)),
  #       aes(x = factor(Strain, strains_uniq), y = sample_geq, group = ID),
  #       position = "identity", linewidth = 0,
  #       fill = "#b9b9b9", stat = "identity", alpha = 0.05
  # ) +
  geom_boxplot(data = df_species_count_all %>% filter(!ID %in% samples_low_detection) %>%
                    filter(!Strain %in% c("unknown")) %>%
                    mutate(facet_category = Type) %>%
                    mutate(facet_category = ifelse(facet_category == "bee", paste0("0. ", facet_category, "(", Treatment_simp, ")"), facet_category)) %>%
                    mutate(facet_category = ifelse(facet_category == "bacteria", "1. bacteria", facet_category)) %>%
                    mutate(facet_category = ifelse(facet_category == "aliquot", "1. bacteria", facet_category)) %>%
                    mutate(facet_category = ifelse(facet_category == "blank", "2. blank", facet_category)) %>%
                    mutate(facet_category = ifelse(Treatment == "extraction_blank", "2. blank", facet_category)) %>%
                    filter(geq_passed == "1") %>%
                    mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)),
        aes(x = factor(Strain, strains_uniq), y = abs_counts, fill = Treatment_simp),
        outlier.shape = NA,
        position = "identity", linewidth = 0.5) +
  geom_point(data = df_species_count_all %>% filter(!ID %in% samples_low_detection) %>%
                    filter(!Strain %in% c("unknown")) %>%
                    mutate(facet_category = Type) %>%
                    mutate(facet_category = ifelse(facet_category == "bee", paste0("0. ", facet_category, "(", Treatment_simp, ")"), facet_category)) %>%
                    mutate(facet_category = ifelse(facet_category == "bacteria", "1. bacteria", facet_category)) %>%
                    mutate(facet_category = ifelse(facet_category == "aliquot", "1. bacteria", facet_category)) %>%
                    mutate(facet_category = ifelse(facet_category == "blank", "2. blank", facet_category)) %>%
                    mutate(facet_category = ifelse(Treatment == "extraction_blank", "2. blank", facet_category)) %>%
                    mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)),
        aes(x = factor(Strain, strains_uniq), y = abs_counts, color = geq_passed, alpha = geq_passed),
        size = 0.5) +
  facet_grid(~facet_category, space = "free") +
  geom_hline(yintercept = median(df_total_reads_compare$sample_geq, na.rm = T), linetype = "solid") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = TreatmentColors) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Strain", y = "Absolute Abundance", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 4,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/Strain_abundace_per_sample.pdf", width = 20, height = 15)
```

## Check community bacterial community samples

### Check just the aliquots used in the experiments

```{r}
system("mkdir -p 03_Analysis/Results/figures/aliquots")

df_comm_heatmap <- df_species_count_all %>%
                    filter(Type == "bacteria") %>%
                    filter(Experiment != "none")
mat_comm_heatmap <- df_comm_heatmap %>%
                    select(ID, Strain, abs_counts) %>%
                    pivot_wider(names_from = Strain, values_from = abs_counts) %>%
                    column_to_rownames("ID") %>%
                    as.matrix()
mat_comm_heatmap <- mat_comm_heatmap[, c(strains, "unknown")]
species_names <- data.frame(strain = colnames(mat_comm_heatmap)) %>% left_join(df_info, by = c("strain" = "strain")) %>% pull(spec_name)
genus_names <- unlist(lapply(species_names, function(x) strsplit(x, " ")[[1]][[1]]))
treatment <- data.frame(ID = rownames(mat_comm_heatmap)) %>% left_join(sample_metadata %>% select(ID, Treatment), by = c("ID")) %>% pull(Treatment)
col_split <- genus_names %>% as.factor() %>% as.character()
row_split <- c("1","1","2","2","3","3","1","1","2","2",
    "3","3","4","4","5","5","6","6","4","4",
    "5","5","6","6", "alq", "alq", "alq", "alq", "m1","m1","m4","m4","m6","m6",
    "m7","m7")
top_anno <- HeatmapAnnotation(Species = as.character(species_names),
                                col = list(Species = speciesColors_uniq)
                                )
colnames(mat_comm_heatmap) <- data.frame(strain = colnames(mat_comm_heatmap)) %>%
                                left_join(df_info %>% select(spec_name, strain), by = c("strain")) %>%
                                mutate(spec_name = paste0(spec_name, " (", strain, ")")) %>%
                                pull(spec_name)
heatmap_obj = Heatmap(mat_comm_heatmap,
                      name = "Absolute count",
                      col = colorRamp2(c(0, 10, 100, 1000, 10000, 1000000, 10000000), 
                                       c(brewer.pal(9, "PuBu")[1], brewer.pal(9, "PuBu")[2], brewer.pal(9, "PuBu")[3], brewer.pal(9, "PuBu")[4], brewer.pal(9, "PuBu")[6], brewer.pal(9, "PuBu")[8], brewer.pal(9, "PuBu")[9])),
                      heatmap_legend_param = list(at = c(0, 10, 100, 1000, 10000, 1000000, 10000000),
                                                  legend_height = unit(5, "cm"),
                                                  labels = c(0, 10, 100, 1000, 10000, 1000000, 10000000)),
                      row_split = row_split,
                      column_split = col_split,
                      top_annotation = top_anno,
                      cluster_rows = F,
                      cluster_columns = F,
                      column_title_rot = 40,
                      cluster_row_slices = T,
                      border = T,
                      column_names_side = "top",
                      column_names_rot = 45,
                      column_title_side = "bottom",
                      row_title_rot = 0,
                      show_row_names = T,
                      show_row_dend = F,
                      show_column_dend = F,
                      show_heatmap_legend = T)
draw(heatmap_obj)
pdf("03_Analysis/Results/figures/aliquots/strain_abundance_heatmap.pdf", width = 10, height = 15)
draw(heatmap_obj)
dev.off()


mat_comm_heatmap <- df_comm_heatmap %>%
                    select(ID, Strain, abs_counts) %>%
                    mutate(present = ifelse(abs_counts > 0, 1, 0)) %>%
                    select(-abs_counts) %>%
                    pivot_wider(names_from = Strain, values_from = present) %>%
                    column_to_rownames("ID") %>%
                    as.matrix()
mat_comm_heatmap <- mat_comm_heatmap[, c(strains, "unknown")]
species_names <- data.frame(strain = colnames(mat_comm_heatmap)) %>% left_join(df_info, by = c("strain" = "strain")) %>% pull(spec_name)
genus_names <- unlist(lapply(species_names, function(x) strsplit(x, " ")[[1]][[1]]))
treatment <- data.frame(ID = rownames(mat_comm_heatmap)) %>% left_join(sample_metadata %>% select(ID, Treatment), by = c("ID")) %>% pull(Treatment)
col_split <- genus_names %>% as.factor() %>% as.character()
row_split <- c("1","1","2","2","3","3","1","1","2","2",
    "3","3","4","4","5","5","6","6","4","4",
    "5","5","6","6", "alq", "alq", "alq", "alq", "m1","m1","m4","m4","m6","m6",
    "m7","m7")
top_anno <- HeatmapAnnotation(Species = as.character(species_names),
                                col = list(Species = speciesColors_uniq)
                                )
colnames(mat_comm_heatmap) <- data.frame(strain = colnames(mat_comm_heatmap)) %>%
                                left_join(df_info %>% select(spec_name, strain), by = c("strain")) %>%
                                mutate(spec_name = paste0(spec_name, " (", strain, ")")) %>%
                                pull(spec_name)
heatmap_obj = Heatmap(mat_comm_heatmap,
                      name = "Present",
                      col = c("#ffffff", "#1f78b4"),
                      row_split = row_split,
                      column_split = col_split,
                      top_annotation = top_anno,
                      cluster_rows = F,
                      cluster_columns = F,
                      column_title_rot = 40,
                      cluster_row_slices = T,
                      border = T,
                      column_names_side = "top",
                      column_names_rot = 45,
                      column_title_side = "bottom",
                      row_title_rot = 0,
                      show_row_names = T,
                      show_row_dend = F,
                      show_column_dend = F,
                      show_heatmap_legend = T)
draw(heatmap_obj)
pdf("03_Analysis/Results/figures/aliquots/strain_presence_heatmap.pdf", width = 10, height = 15)
draw(heatmap_obj)
dev.off()
```

Plot community composition as PCoA for each set of treatments

```{r}
system("mkdir -p 03_Analysis/Results/figures/priority_effects_experiment/community_composition")
system("mkdir -p 03_Analysis/Results/tables")
system("mkdir -p 03_Analysis/Results/tables/community_composition")
system("mkdir -p 03_Analysis/Results/figures/priority_effects_experiment/community_composition/heatmaps")

combos_for_plot <- c("1", 
                     "4",
                     "1\'", 
                     "m6",
                     
                     "2",
                     "5",
                     "m1",
                     "m7",
                     
                     "3",
                     "6",
                     "m4")
# # combos_for_plot <- c("1", "2", "3", "4", "5", "6", "1\'", "m1", "m4", "m6", "m7")

p_list <- list()
for (my_comunity_combo in combos_for_plot) {
  if (grepl("m", my_comunity_combo)) {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        filter(Type == "bee") %>%
                        filter(Experiment == "3") %>%
                        filter(!ID %in% samples_low_detection) %>%
                        filter(CommunityCombo %in% c("1\'", my_comunity_combo)) %>%
                        select(ID, Strain, abs_counts) %>%
                        pivot_wider(names_from = Strain, values_from = abs_counts) %>%
                        column_to_rownames("ID") %>% as.matrix()
  } else {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                          filter(Type == "bee") %>%
                          mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                          filter(!ID %in% samples_low_detection) %>%
                          filter(CommunityCombo == my_comunity_combo) %>%
                          select(ID, Strain, abs_counts) %>%
                          pivot_wider(names_from = Strain, values_from = abs_counts) %>%
                          column_to_rownames("ID") %>% as.matrix()
  }
 

  df_abundance_mat_org <- df_abundance_mat
  
  df_abundance_mat <- decostand(df_abundance_mat[, -1], method = "log")

  # df_abundance_mat[is.na(df_abundance_mat)] <- 0

  df_abundance_info <- df_species_count_all %>%
                        select(ID, Type, Experiment, CommunityCombo, Treatment) %>%
                        filter(Type == "bee") %>%
                        unique() %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment))

  dist_mat_bray <- vegdist(df_abundance_mat, method = "bray")

  matrix <- as.matrix(dist_mat_bray)
  dist <- as.dist(matrix)
  res_pcoa <- pcoa(dist)
  ev1 <- res_pcoa$vectors[,1]
  ev2 <- res_pcoa$vectors[,2]
  df_pcoa_new <- data.frame(cbind(ev1,ev2))
  df_pcoa_new$Sample <- rownames(df_pcoa_new)
  rownames(df_pcoa_new) <- NULL
  df_pcoa_new <- left_join(df_pcoa_new, df_abundance_info, by = c("Sample" = "ID"))
  perc_axis <- round(((res_pcoa$values$Relative_eig[c(1,2)])*100), digits=1)
  axis_x_title <- paste0("PCo1 (",perc_axis[1],"%)")
  axis_y_title <- paste0("PCo2 (",perc_axis[2],"%)")

  ado_res = adonis2(dist_mat_bray ~ Treatment_simp, 
          rownames(dist_mat_bray %>% as.matrix) %>% as.data.frame() %>%
          left_join(df_abundance_info, by = c(. = "ID"))
  )
  adonis_OmegaSq(ado_res)

  p <- ggplot(df_pcoa_new, aes(x = ev1,
                          y = ev2,
                          colour = Treatment_simp)) +
                  geom_point(stat="identity", size=4) +
                    labs(x=axis_x_title, y = axis_y_title, color = "Treatment", title = paste0("Community combination: ", my_comunity_combo)) +
                      make_theme(setFill = F, setCol = F, guide_nrow = 1, leg_size = 10 ) +
                        scale_color_manual(values=TreatmentColors) +
                        annotate("text", x = 0,
                                 y = 0,
                                 size = 6,
                                 label = paste0("omega2 = ", round(adonis_OmegaSq(ado_res)$parOmegaSq[[1]], 3),
                                               ", \np = ", adonis_OmegaSq(ado_res)$`Pr(>F`[[1]], "  (N=", 1+ado_res$Df[[3]], ")")
                                 )
  p_list[[my_comunity_combo]] <- p
  
  if (grepl("m", my_comunity_combo)) {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        filter(Type == "bee") %>%
                        filter(Experiment == "3") %>%
                        filter(CommunityCombo %in% c("1\'", my_comunity_combo)) %>%
                        select(ID, spec_name_uniq, abs_counts) %>%
                        pivot_wider(names_from = spec_name_uniq, values_from = abs_counts) %>%
                        column_to_rownames("ID") %>% as.matrix()
  } else {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                          mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                          filter(Type == "bee") %>%
                          filter(CommunityCombo == my_comunity_combo) %>%
                          select(ID, spec_name_uniq, abs_counts) %>%
                          pivot_wider(names_from = spec_name_uniq, values_from = abs_counts) %>%
                          column_to_rownames("ID") %>% as.matrix()
  }
  df_abundance_mat %>% as.data.frame() %>% rownames_to_column("ID") %>% left_join(df_abundance_info %>% select(ID, Treatment), by = c("ID" = "ID")) %>%
  write.csv(paste0("03_Analysis/Results/tables/community_composition/", my_comunity_combo,"df_abundance_mat.csv"), row.names = F)

  treatments_vec <- rownames(df_abundance_mat_org) %>% as.data.frame() %>%
              left_join(df_abundance_info, by = c("." = "ID")) %>%
              mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment)) %>%
              select(., Treatment_simp) %>% pull(Treatment_simp) %>% as.character
  right_anno <- rowAnnotation(Treatment = treatments_vec,
                              col = list(Treatment = TreatmentColors)
                          )
  row_sep <- treatments_vec
  heatmap_obj <- Heatmap(df_abundance_mat_org,
                        name = "Abundance",
                col = colorRamp2(c(0, 1e+2, 1e+4, 1e+6, 1e+8, 1e+10), c(brewer.pal(9, "PuBu")[1], brewer.pal(9, "PuBu")[2], brewer.pal(9, "PuBu")[3], brewer.pal(9, "PuBu")[6], brewer.pal(9, "PuBu")[8], brewer.pal(9, "PuBu")[9])),
                cluster_rows = TRUE,
                cluster_columns = TRUE,
                column_title_rot = 0,
                column_title_gp = gpar(fontsize = 12),
                cluster_row_slices = T,
                border = T,
                right_annotation = right_anno,
                row_title_gp = gpar(fontsize = 12),
                row_title_rot = 0,
                show_row_names = T,
                show_heatmap_legend = T,
                row_title = "ID",
                column_title = "Species",
                # row_split = row_sep,
                #  column_split = col_sep,
                # heatmap_legend_param = list(title = "Presence", at = c(0, 1), labels = c("0", "1")))
                heatmap_legend_param = list(title = "Counts", at = c(0, 1e+2, 1e+4, 1e+6, 1e+8, 1e+10), labels = c("0", "1e+2", "1e+4", "1e+6", "1e+8", "1e+10")))
  pdf(paste0("03_Analysis/Results/figures/priority_effects_experiment/community_composition/heatmaps/", my_comunity_combo, "_jaccard.pdf"), width = 20, height = 20)
  draw(heatmap_obj, heatmap_legend_side = "right", annotation_legend_side = "right")
  dev.off()
}

pdf("03_Analysis/Results/figures/priority_effects_experiment/community_composition/pcoa_bray.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 4, nrow = 4)
dev.off()


p_list <- list()
for (my_comunity_combo in combos_for_plot) {
  if (grepl("m", my_comunity_combo)) {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        filter(Type == "bee") %>%
                        filter(Experiment == "3") %>%
                        filter(!ID %in% samples_low_detection) %>%
                        filter(CommunityCombo %in% c("1\'", my_comunity_combo)) %>%
                        select(ID, Strain, rel_counts) %>%
                        group_by(ID) %>%
                        mutate(sum = sum(rel_counts)) %>%
                        filter(sum > 0) %>%
                        ungroup() %>%
                        select(-sum) %>%
                        pivot_wider(names_from = Strain, values_from = rel_counts) %>%
                        column_to_rownames("ID") %>% as.matrix()
  } else {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                          mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                          filter(Type == "bee") %>%
                          filter(!ID %in% samples_low_detection) %>%
                          filter(CommunityCombo == my_comunity_combo) %>%
                          select(ID, Strain, rel_counts) %>%
                          group_by(ID) %>%
                          mutate(sum = sum(rel_counts)) %>%
                          filter(sum > 0) %>%
                          ungroup() %>%
                          select(-sum) %>%
                          pivot_wider(names_from = Strain, values_from = rel_counts) %>%
                          column_to_rownames("ID") %>% as.matrix()
  }

  # df_abundance_mat[is.na(df_abundance_mat)] <- 0

  df_abundance_info <- df_species_count_all %>%
                        select(ID, Type, Experiment, CommunityCombo, Treatment) %>%
                        filter(Type == "bee") %>%
                        unique() %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment))

  dist_mat_rel <- vegdist(df_abundance_mat, method = "robust.aitchison")

  matrix <- as.matrix(dist_mat_rel)
  dist <- as.dist(matrix)
  res_pcoa <- pcoa(dist)
  ev1 <- res_pcoa$vectors[,1]
  ev2 <- res_pcoa$vectors[,2]
  df_pcoa_new <- data.frame(cbind(ev1,ev2))
  df_pcoa_new$Sample <- rownames(df_pcoa_new)
  rownames(df_pcoa_new) <- NULL
  df_pcoa_new <- left_join(df_pcoa_new, df_abundance_info, by = c("Sample" = "ID"))
  perc_axis <- round(((res_pcoa$values$Relative_eig[c(1,2)])*100), digits=1)
  axis_x_title <- paste0("PCo1 (",perc_axis[1],"%)")
  axis_y_title <- paste0("PCo2 (",perc_axis[2],"%)")

  ado_res = adonis2(dist_mat_rel ~ Treatment_simp, 
          rownames(dist_mat_rel %>% as.matrix) %>% as.data.frame() %>%
          left_join(df_abundance_info, by = c(. = "ID"))
  )
  adonis_OmegaSq(ado_res)

  p <- ggplot(df_pcoa_new, aes(x = ev1,
                          y = ev2,
                          colour = Treatment_simp)) +
                  geom_point(stat="identity", size=4) +
                    labs(x=axis_x_title, y = axis_y_title, color = "Treatment", title = paste0("Community combination: ", my_comunity_combo)) +
                      make_theme(setFill = F, setCol = F, guide_nrow = 1, leg_size = 10 ) +
                        scale_color_manual(values=TreatmentColors) +
                        annotate("text", x = 0,
                                 y = 0,
                                 size = 6,
                                 label = paste0("omega2 = ", round(adonis_OmegaSq(ado_res)$parOmegaSq[[1]], 3),
                                               ", \np = ", adonis_OmegaSq(ado_res)$`Pr(>F`[[1]], "  (N=", 1+ado_res$Df[[3]], ")")
                                 )
  p_list[[my_comunity_combo]] <- p
  
  if (grepl("m", my_comunity_combo)) {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        filter(Type == "bee") %>%
                        filter(Experiment == "3") %>%
                        filter(CommunityCombo %in% c("1\'", my_comunity_combo)) %>%
                        select(ID, spec_name_uniq, abs_counts) %>%
                        pivot_wider(names_from = spec_name_uniq, values_from = abs_counts) %>%
                        column_to_rownames("ID") %>% as.matrix()
  } else {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                          mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                          filter(Type == "bee") %>%
                          filter(CommunityCombo == my_comunity_combo) %>%
                          select(ID, spec_name_uniq, abs_counts) %>%
                          pivot_wider(names_from = spec_name_uniq, values_from = abs_counts) %>%
                          column_to_rownames("ID") %>% as.matrix()
  }
  df_abundance_mat %>% as.data.frame() %>% rownames_to_column("ID") %>% left_join(df_abundance_info %>% select(ID, Treatment), by = c("ID" = "ID")) %>%
  write.csv(paste0("03_Analysis/Results/tables/community_composition/", my_comunity_combo,"df_abundance_rel_mat.csv"), row.names = F)
}

pdf("03_Analysis/Results/figures/priority_effects_experiment/community_composition/pcoa_aitchison.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 4, nrow = 4)
dev.off()

p_list <- list()
for (my_comunity_combo in combos_for_plot) {
  
  if (grepl("m", my_comunity_combo)) {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        filter(Type == "bee") %>%
                        filter(Experiment == "3") %>%
                        filter(CommunityCombo %in% c("1\'", my_comunity_combo)) %>%
                        filter(!ID %in% samples_low_detection) %>%
                        mutate(present = ifelse(abs_counts > 0, 1, 0)) %>%
                        mutate(present = as.numeric(present)) %>%
                        select(ID, Strain, present) %>%
                        pivot_wider(names_from = Strain, values_from = present) %>%
                        column_to_rownames("ID") %>% as.matrix()
  } else {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                          mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                          filter(Type == "bee") %>%
                          filter(CommunityCombo == my_comunity_combo) %>%
                          filter(!ID %in% samples_low_detection) %>%
                          mutate(present = ifelse(abs_counts > 0, 1, 0)) %>%
                          mutate(present = as.numeric(present)) %>%
                          select(ID, Strain, present) %>%
                          pivot_wider(names_from = Strain, values_from = present) %>%
                          column_to_rownames("ID") %>% as.matrix()
  }
 
  # drop empty rows
  df_abundance_mat <- df_abundance_mat[rowSums(df_abundance_mat) > 0, ]

  df_abundance_info <- df_species_count_all %>%
                        select(ID, Type, Experiment, CommunityCombo, Treatment) %>%
                        filter(Type == "bee") %>%
                        unique() %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment))

  dist_mat_jacc <- vegdist(df_abundance_mat, method = "jaccard")

  matrix <- as.matrix(dist_mat_jacc)
  dist <- as.dist(matrix)
  res_pcoa <- pcoa(dist)
  ev1 <- res_pcoa$vectors[,1]
  ev2 <- res_pcoa$vectors[,2]
  df_pcoa_new <- data.frame(cbind(ev1,ev2))
  df_pcoa_new$Sample <- rownames(df_pcoa_new)
  rownames(df_pcoa_new) <- NULL
  df_pcoa_new <- left_join(df_pcoa_new, df_abundance_info, by = c("Sample" = "ID"))
  perc_axis <- round(((res_pcoa$values$Relative_eig[c(1,2)])*100), digits=1)
  axis_x_title <- paste0("PCo1 (",perc_axis[1],"%)")
  axis_y_title <- paste0("PCo2 (",perc_axis[2],"%)")

  ado_res = adonis2(dist_mat_jacc ~ Treatment_simp, 
          rownames(dist_mat_jacc %>% as.matrix) %>% as.data.frame() %>%
          left_join(df_abundance_info, by = c(. = "ID"))
  )
  adonis_OmegaSq(ado_res)

  p <- ggplot(df_pcoa_new, aes(x = ev1,
                          y = ev2,
                          colour = Treatment_simp)) +
                  geom_point(stat="identity", size=4) +
                    labs(x=axis_x_title, y = axis_y_title, color = "Treatment", title = paste0("Community combination: ", my_comunity_combo)) +
                      make_theme(setFill = F, setCol = F, guide_nrow = 1, leg_size = 10 ) +
                        scale_color_manual(values=TreatmentColors) +
                        annotate("text", x = 0,
                                 y = 0,
                                 size = 6,
                                 label = paste0("omega2 = ", round(adonis_OmegaSq(ado_res)$parOmegaSq[[1]], 3),
                                               ", \np = ", adonis_OmegaSq(ado_res)$`Pr(>F`[[1]], "  (N=", 1+ado_res$Df[[3]], ")")
                                 )
  p_list[[my_comunity_combo]] <- p

  treatments_vec <- rownames(df_abundance_mat) %>% as.data.frame() %>%
              left_join(df_abundance_info, by = c("." = "ID")) %>%
              mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment)) %>%
              select(., Treatment_simp) %>% pull(Treatment_simp) %>% as.character
  right_anno <- rowAnnotation(Treatment = treatments_vec,
                              col = list(Treatment = TreatmentColors)
                          )
  row_sep <- treatments_vec
  heatmap_obj <- Heatmap(df_abundance_mat, 
                        name = "Presence",
                # col = colorRamp2(c(0, 1000, 100000, 1000000, 100000000), c(brewer.pal(9, "PuBu")[1], brewer.pal(9, "PuBu")[2], brewer.pal(9, "PuBu")[6], brewer.pal(9, "PuBu")[7], brewer.pal(9, "PuBu")[8])),
                 col = colorRamp2(c(0, 1), c("#ffffff", "#2166ac")),
                cluster_rows = TRUE,
                cluster_columns = TRUE,
                column_title_rot = 0,
                column_title_gp = gpar(fontsize = 12),
                cluster_row_slices = T,
                border = T,
                right_annotation = right_anno,
                row_title_gp = gpar(fontsize = 12),
                row_title_rot = 0,
                show_row_names = T,
                show_heatmap_legend = T,
                row_title = "ID",
                column_title = "Species",
                # row_split = row_sep,
                #  column_split = col_sep,
                 heatmap_legend_param = list(title = "Presence", at = c(0, 1), labels = c("0", "1")))
                # heatmap_legend_param = list(title = "Counts", at = c(0, 1000, 100000, 1000000, 100000000), labels = c("0", "10", "100", "1000", "10000")))
  pdf(paste0("03_Analysis/Results/figures/priority_effects_experiment/community_composition/heatmaps/", my_comunity_combo, "_jaccard.pdf"), width = 20, height = 20)
  draw(heatmap_obj, heatmap_legend_side = "right", annotation_legend_side = "right")
  dev.off()
}

pdf("03_Analysis/Results/figures/priority_effects_experiment/community_composition/pcoa_jaccard.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 4, nrow = 4)
dev.off()
```

At the species level..

```{r}
system("mkdir -p 03_Analysis/Results/figures/priority_effects_experiment/community_composition_species_level")
system("mkdir -p 03_Analysis/Results/tables")
system("mkdir -p 03_Analysis/Results/tables/community_composition_species_level")
system("mkdir -p 03_Analysis/Results/figures/priority_effects_experiment/community_composition_species_level/heatmaps/")

combos_for_plot <- c("1",
                     "4",
                     "1\'", 
                     "m6",
                     
                     "2",
                     "5",
                     "m1",
                     "m7",
                     
                     "3",
                     "6",
                     "m4")
# # combos_for_plot <- c("1", "2", "3", "4", "5", "6", "1\'", "m1", "m4", "m6", "m7")

p_list <- list()
for (my_comunity_combo in combos_for_plot) {
  if (grepl("m", my_comunity_combo)) {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        filter(Type == "bee") %>%
                        filter(Experiment == "3") %>%
                        filter(!ID %in% samples_low_detection) %>%
                        filter(CommunityCombo %in% c("1\'", my_comunity_combo)) %>%
                        select(ID, spec_name, abs_counts) %>%
                        group_by(ID, spec_name) %>%
                        reframe(ID, spec_name, abs_counts = sum(abs_counts)) %>%
                        unique() %>%
                        pivot_wider(names_from = spec_name, values_from = abs_counts) %>%
                        column_to_rownames("ID") %>% as.matrix()
  } else {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                          filter(Type == "bee") %>%
                          mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                          filter(!ID %in% samples_low_detection) %>%
                          filter(CommunityCombo == my_comunity_combo) %>%
                          select(ID, spec_name, abs_counts) %>%
                          group_by(ID, spec_name) %>%
                          reframe(ID, spec_name, abs_counts = sum(abs_counts)) %>%
                          unique() %>%
                          pivot_wider(names_from = spec_name, values_from = abs_counts) %>%
                          column_to_rownames("ID") %>% as.matrix()
  }
 

  df_abundance_mat_org <- df_abundance_mat
  
  df_abundance_mat <- decostand(df_abundance_mat[, -1], method = "log")

  # df_abundance_mat[is.na(df_abundance_mat)] <- 0

  df_abundance_info <- df_species_count_all %>%
                        select(ID, Type, Experiment, CommunityCombo, Treatment) %>%
                        filter(Type == "bee") %>%
                        unique() %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment))

  dist_mat_bray <- vegdist(df_abundance_mat, method = "bray")

  matrix <- as.matrix(dist_mat_bray)
  dist <- as.dist(matrix)
  res_pcoa <- pcoa(dist)
  ev1 <- res_pcoa$vectors[,1]
  ev2 <- res_pcoa$vectors[,2]
  df_pcoa_new <- data.frame(cbind(ev1,ev2))
  df_pcoa_new$Sample <- rownames(df_pcoa_new)
  rownames(df_pcoa_new) <- NULL
  df_pcoa_new <- left_join(df_pcoa_new, df_abundance_info, by = c("Sample" = "ID"))
  perc_axis <- round(((res_pcoa$values$Relative_eig[c(1,2)])*100), digits=1)
  axis_x_title <- paste0("PCo1 (",perc_axis[1],"%)")
  axis_y_title <- paste0("PCo2 (",perc_axis[2],"%)")

  ado_res = adonis2(dist_mat_bray ~ Treatment_simp, 
          rownames(dist_mat_bray %>% as.matrix) %>% as.data.frame() %>%
          left_join(df_abundance_info, by = c(. = "ID"))
  )
  adonis_OmegaSq(ado_res)

  p <- ggplot(df_pcoa_new, aes(x = ev1,
                          y = ev2,
                          colour = Treatment_simp)) +
                  geom_point(stat="identity", size=4) +
                    labs(x=axis_x_title, y = axis_y_title, color = "Treatment", title = paste0("Community combination: ", my_comunity_combo)) +
                      make_theme(setFill = F, setCol = F, guide_nrow = 1, leg_size = 10 ) +
                        scale_color_manual(values=TreatmentColors) +
                        annotate("text", x = 0,
                                 y = 0,
                                 size = 6,
                                 label = paste0("omega2 = ", round(adonis_OmegaSq(ado_res)$parOmegaSq[[1]], 3),
                                               ", \np = ", adonis_OmegaSq(ado_res)$`Pr(>F`[[1]], "  (N=", 1+ado_res$Df[[3]], ")")
                                 )
  p_list[[my_comunity_combo]] <- p

  treatments_vec <- rownames(df_abundance_mat_org) %>% as.data.frame() %>%
              left_join(df_abundance_info, by = c("." = "ID")) %>%
              mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment)) %>%
              select(., Treatment_simp) %>% pull(Treatment_simp) %>% as.character
  right_anno <- rowAnnotation(Treatment = treatments_vec,
                              col = list(Treatment = TreatmentColors)
                          )
  row_sep <- treatments_vec
  heatmap_obj <- Heatmap(df_abundance_mat_org,
                        name = "Abundance",
                col = colorRamp2(c(0, 1e+2, 1e+4, 1e+6, 1e+8, 1e+10), c(brewer.pal(9, "PuBu")[1], brewer.pal(9, "PuBu")[2], brewer.pal(9, "PuBu")[3], brewer.pal(9, "PuBu")[6], brewer.pal(9, "PuBu")[8], brewer.pal(9, "PuBu")[9])),
                cluster_rows = TRUE,
                cluster_columns = TRUE,
                column_title_rot = 0,
                column_title_gp = gpar(fontsize = 12),
                cluster_row_slices = T,
                border = T,
                right_annotation = right_anno,
                row_title_gp = gpar(fontsize = 12),
                row_title_rot = 0,
                show_row_names = T,
                show_heatmap_legend = T,
                row_title = "ID",
                column_title = "Species",
                # row_split = row_sep,
                #  column_split = col_sep,
                # heatmap_legend_param = list(title = "Presence", at = c(0, 1), labels = c("0", "1")))
                heatmap_legend_param = list(title = "Counts", at = c(0, 1e+2, 1e+4, 1e+6, 1e+8, 1e+10), labels = c("0", "1e+2", "1e+4", "1e+6", "1e+8", "1e+10")))
  pdf(paste0("03_Analysis/Results/figures/priority_effects_experiment/community_composition_species_level/heatmaps/", my_comunity_combo, "_jaccard.pdf"), width = 20, height = 20)
  draw(heatmap_obj, heatmap_legend_side = "right", annotation_legend_side = "right")
  dev.off()
  
  if (grepl("m", my_comunity_combo)) {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        filter(Type == "bee") %>%
                        filter(Experiment == "3") %>%
                        filter(CommunityCombo %in% c("1\'", my_comunity_combo)) %>%
                        select(ID, spec_name_uniq, abs_counts) %>%
                        pivot_wider(names_from = spec_name_uniq, values_from = abs_counts) %>%
                        column_to_rownames("ID") %>% as.matrix()
  } else {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                          mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                          filter(Type == "bee") %>%
                          filter(CommunityCombo == my_comunity_combo) %>%
                          select(ID, spec_name_uniq, abs_counts) %>%
                          pivot_wider(names_from = spec_name_uniq, values_from = abs_counts) %>%
                          column_to_rownames("ID") %>% as.matrix()
  }
  df_abundance_mat %>% as.data.frame() %>% rownames_to_column("ID") %>% left_join(df_abundance_info %>% select(ID, Treatment), by = c("ID" = "ID")) %>%
  write.csv(paste0("03_Analysis/Results/tables/community_composition_species_level/", my_comunity_combo,"df_abundance_mat.csv"), row.names = F)
}

pdf("03_Analysis/Results/figures/priority_effects_experiment/community_composition_species_level/pcoa_bray.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 4, nrow = 4)
dev.off()

p_list <- list()
for (my_comunity_combo in combos_for_plot) {
  
  if (grepl("m", my_comunity_combo)) {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        filter(Type == "bee") %>%
                        filter(Experiment == "3") %>%
                        filter(CommunityCombo %in% c("1\'", my_comunity_combo)) %>%
                        filter(!ID %in% samples_low_detection) %>%
                        select(ID, spec_name, abs_counts) %>%
                        group_by(ID, spec_name) %>%
                        reframe(ID, spec_name, abs_counts = sum(abs_counts)) %>%
                        unique() %>%
                        mutate(present = ifelse(abs_counts > 0, 1, 0)) %>%
                        mutate(present = as.numeric(present)) %>%
                        select(ID, spec_name, present) %>%
                        pivot_wider(names_from = spec_name, values_from = present) %>%
                        column_to_rownames("ID") %>% as.matrix()
  } else {
    df_abundance_mat <- df_species_count_all %>% filter(Strain != "unknown") %>%
                          mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                          filter(Type == "bee") %>%
                          filter(CommunityCombo == my_comunity_combo) %>%
                          filter(!ID %in% samples_low_detection) %>%
                          select(ID, spec_name, abs_counts) %>%
                          group_by(ID, spec_name) %>%
                          reframe(ID, spec_name, abs_counts = sum(abs_counts)) %>%
                          unique() %>%
                          mutate(present = ifelse(abs_counts > 0, 1, 0)) %>%
                          mutate(present = as.numeric(present)) %>%
                          select(ID, spec_name, present) %>%
                          pivot_wider(names_from = spec_name, values_from = present) %>%
                          column_to_rownames("ID") %>% as.matrix()
  }
 
  # drop empty rows
  df_abundance_mat <- df_abundance_mat[rowSums(df_abundance_mat) > 0, ]

  df_abundance_info <- df_species_count_all %>%
                        select(ID, Type, Experiment, CommunityCombo, Treatment) %>%
                        filter(Type == "bee") %>%
                        unique() %>%
                        mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                        mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment))

  dist_mat_jacc <- vegdist(df_abundance_mat, method = "jaccard")

  matrix <- as.matrix(dist_mat_jacc)
  dist <- as.dist(matrix)
  res_pcoa <- pcoa(dist)
  ev1 <- res_pcoa$vectors[,1]
  ev2 <- res_pcoa$vectors[,2]
  df_pcoa_new <- data.frame(cbind(ev1,ev2))
  df_pcoa_new$Sample <- rownames(df_pcoa_new)
  rownames(df_pcoa_new) <- NULL
  df_pcoa_new <- left_join(df_pcoa_new, df_abundance_info, by = c("Sample" = "ID"))
  perc_axis <- round(((res_pcoa$values$Relative_eig[c(1,2)])*100), digits=1)
  axis_x_title <- paste0("PCo1 (",perc_axis[1],"%)")
  axis_y_title <- paste0("PCo2 (",perc_axis[2],"%)")

  ado_res = adonis2(dist_mat_jacc ~ Treatment_simp, 
          rownames(dist_mat_jacc %>% as.matrix) %>% as.data.frame() %>%
          left_join(df_abundance_info, by = c(. = "ID"))
  )
  adonis_OmegaSq(ado_res)

  p <- ggplot(df_pcoa_new, aes(x = ev1,
                          y = ev2,
                          colour = Treatment_simp)) +
                  geom_point(stat="identity", size=4) +
                    labs(x=axis_x_title, y = axis_y_title, color = "Treatment", title = paste0("Community combination: ", my_comunity_combo)) +
                      make_theme(setFill = F, setCol = F, guide_nrow = 1, leg_size = 10 ) +
                        scale_color_manual(values=TreatmentColors) +
                        annotate("text", x = 0,
                                 y = 0,
                                 size = 6,
                                 label = paste0("omega2 = ", round(adonis_OmegaSq(ado_res)$parOmegaSq[[1]], 3),
                                               ", \np = ", adonis_OmegaSq(ado_res)$`Pr(>F`[[1]], "  (N=", 1+ado_res$Df[[3]], ")")
                                 )
  p_list[[my_comunity_combo]] <- p

  treatments_vec <- rownames(df_abundance_mat) %>% as.data.frame() %>%
              left_join(df_abundance_info, by = c("." = "ID")) %>%
              mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment)) %>%
              select(., Treatment_simp) %>% pull(Treatment_simp) %>% as.character
  right_anno <- rowAnnotation(Treatment = treatments_vec,
                              col = list(Treatment = TreatmentColors)
                          )
  row_sep <- treatments_vec
  heatmap_obj <- Heatmap(df_abundance_mat, 
                        name = "Presence",
                # col = colorRamp2(c(0, 1000, 100000, 1000000, 100000000), c(brewer.pal(9, "PuBu")[1], brewer.pal(9, "PuBu")[2], brewer.pal(9, "PuBu")[6], brewer.pal(9, "PuBu")[7], brewer.pal(9, "PuBu")[8])),
                 col = colorRamp2(c(0, 1), c("#ffffff", "#2166ac")),
                cluster_rows = TRUE,
                cluster_columns = TRUE,
                column_title_rot = 0,
                column_title_gp = gpar(fontsize = 12),
                cluster_row_slices = T,
                border = T,
                right_annotation = right_anno,
                row_title_gp = gpar(fontsize = 12),
                row_title_rot = 0,
                show_row_names = T,
                show_heatmap_legend = T,
                row_title = "ID",
                column_title = "Species",
                # row_split = row_sep,
                #  column_split = col_sep,
                 heatmap_legend_param = list(title = "Presence", at = c(0, 1), labels = c("0", "1")))
                # heatmap_legend_param = list(title = "Counts", at = c(0, 1000, 100000, 1000000, 100000000), labels = c("0", "10", "100", "1000", "10000")))
  pdf(paste0("03_Analysis/Results/figures/priority_effects_experiment/community_composition_species_level/heatmaps/", my_comunity_combo, "_jaccard.pdf"), width = 20, height = 20)
  draw(heatmap_obj, heatmap_legend_side = "right", annotation_legend_side = "right")
  dev.off()
}

pdf("03_Analysis/Results/figures/priority_effects_experiment/community_composition_species_level/pcoa_jaccard.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 4, nrow = 4)
dev.off()
```

## Quantify strength of priority effects

Ensure strains below LOD are considered absent. Using two approaches. One only using AB and BA treatments and another also including A0 and B0. First we start with the simpler approach.

# Priority effect strength calculated using the python script, `priority_effect_strength.py`.

```{r}
# C is absolute counts - filtered or adjusted as needed
# p_ij = ln( C_ji / C_ij )
# There are 8 community-combinaitons in total where 0 is the pilot
# 1 - 6 are in the actual experiment and 7 is the repeat of 1 in experiment 3
# then there are 4 drop-out experiments only combinations 1 was repeated in experiment 3


# p_ij = ln( C_ji / C_0i ) - ln( C_ij / C_i0 )
# consider C_i0 and C_0i as the same perhaps..

# run 03_Analysis/priority_effect_strength.py

system("mkdir -p 03_Analysis/Results/tables/priority_effect_strength")
system("mkdir -p 03_Analysis/Results/tables/priority_effect_strength/temp")

df_species_count_all_combined_mod %>%
                  filter(Type == "bee") %>%
                  filter(!ID %in% samples_low_detection) %>%
                  filter(CommunityCombo %in% c("1", "2", "3", "4", "5", "6", "1\'")) %>%
                  # filter(geq_passed == "1") %>%
                  write.csv("03_Analysis/Results/tables/priority_effect_strength/temp/counts_all_full_combos.csv", row.names = F)

system("source ~/.bashrc && conda activate 20230313_scripts_env && python3 03_Analysis/priority_effect_strength.py --counts_table_file 03_Analysis/Results/tables/priority_effect_strength/temp/counts_all_full_combos.csv --low_detection_samples_file 03_Analysis/PriorityEffectsExperiment/samples_low_detection.csv --output_file 03_Analysis/Results/tables/priority_effect_strength/full_combos_all.csv")

df_species_count_all_combined %>%
                  filter(Type == "bee") %>%
                  filter(!ID %in% samples_low_detection) %>%
                  filter(CommunityCombo %in% c("1", "2", "3", "4", "5", "6", "1\'")) %>%
                  filter(!Strain %in% c("ESL0394")) %>%
                  # filter(geq_passed == "1") %>%
                  write.csv("03_Analysis/Results/tables/priority_effect_strength/temp/counts_full_combos.csv", row.names = F)

system("source ~/.bashrc && conda activate 20230313_scripts_env && python3 03_Analysis/priority_effect_strength.py --counts_table_file 03_Analysis/Results/tables/priority_effect_strength/temp/counts_full_combos.csv --low_detection_samples_file 03_Analysis/PriorityEffectsExperiment/samples_low_detection.csv --output_file 03_Analysis/Results/tables/priority_effect_strength/full_combos_clean.csv")

```

```{r}
df_species_count_plot_diff <- df_species_count_all_combined %>%
                                filter(Experiment %in% c("1", "2")) %>%
                                filter(Type == "bee") %>%
                                filter(arrival %in% c("first", "second", "only")) %>%
                                mutate(abs_counts = ifelse(abs_counts < 1, 1, abs_counts))

pvals <- df_species_count_plot_diff %>%
  filter(!arrival %in% c("dropout")) %>%
  group_by(Strain) %>%
  wilcox_test(abs_counts ~ arrival, data = .) %>%
  # correct p-values for multiple tes
  mutate(y.position = case_when(
            group1 == "first" & group2 == "second" ~ 11,
            group1 == "first" & group2 == "only" ~ 10.5,
            group1 == "only" & group2 == "second" ~ 10,
            TRUE ~ 10.6
  )) %>% # Prevent overlap
  mutate(p = p.adjust(p, method = "BH")) %>%
  mutate(pval = ifelse(p < 0.05, "*", "ns.")) %>%
  mutate(pval = ifelse(p < 0.01, "**", pval)) %>%
  mutate(pval = ifelse(p < 0.001, "***", pval)) %>%
  left_join(df_species_count_all_combined %>%
              select(Strain, spec_name_uniq) %>%
                unique(), by = c("Strain" = "Strain"))

write.csv(pvals, "03_Analysis/Results/tables/priority_effect_strength/fig3_beeswarm_pvals.csv", row.names = F)

# pvals_all <- df_species_count_plot_diff %>%
#   filter(!arrival %in% c("only", "dropout")) %>%
#   group_by(Strain) %>%
#   summarize(pval = wilcox.test(abs_counts ~ arrival, data = ., p.adjust.method = "BH")$p.value) %>%
#   mutate(pval = ifelse(pval < 0.05, "*", "")) %>%
#   mutate(pval = ifelse(pval < 0.01, "**", pval)) %>%
#   mutate(pval = ifelse(pval < 0.001, "***", pval))

ggplot() +
  # geom_bar(data = df_species_count_plot_diff,
  #           aes(x = arrival, y = sample_geq, group = Strain),
  #           position = "identity",
  #           fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_species_count_plot_diff %>% filter(geq_passed == "1"),
                aes(x = arrival, y = abs_counts, fill = Strain),
                linewidth = 0.5,
                outlier.shape = NA
                ) +
  geom_point(data = df_species_count_plot_diff %>% filter(geq_passed != "1"),
                aes(x = arrival, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5, shape = 21, position = position_jitterdodge(jitter.width = 0.25, dodge.width = 0.25)
                ) +
  geom_point(data = df_species_count_plot_diff %>% filter(geq_passed == "1"),
                aes(x = arrival, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5, shape = 21, position = position_jitterdodge(jitter.width = 0.25, dodge.width = 0.25)
                ) +
  facet_wrap(~spec_name_uniq, ncol = 5) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.8, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  geom_signif(data = pvals, manual = TRUE,
              aes(xmin = group1, xmax = group2, annotations = pval, y_position = y.position),
              y_position = 1e+06, tip_length = 0.01
  ) +
  make_theme(setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "none",
             x_hj = 1, x_vj = 1, y_hj = 1) +
  #  add horizontal lines theme.
  theme(
    panel.grid.major.y = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.y = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )
ggsave("03_Analysis/Results/figures/colonization_success/counts_by_arrival_per_strain.pdf", width = 10, height = 15)

ggplot() +
  # geom_bar(data = df_species_count_plot_diff,
  #           aes(x = arrival, y = sample_geq, group = Strain),
  #           position = "identity",
  #           fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_species_count_plot_diff %>% filter(geq_passed != "1"),
                aes(x = arrival, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  geom_beeswarm(data = df_species_count_plot_diff %>% filter(geq_passed == "1"),
                aes(x = arrival, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  facet_wrap(~spec_name_uniq, ncol = 5) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.8, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  geom_signif(data = pvals, manual = TRUE,
              aes(xmin = group1, xmax = group2, annotations = pval, y_position = y.position),
              y_position = 1e+06, tip_length = 0.01
  ) +
  make_theme(setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "none",
             x_hj = 1, x_vj = 1, y_hj = 1) +
  #  add horizontal lines theme.
  theme(
    panel.grid.major.y = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.y = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )
ggsave("03_Analysis/Results/figures/colonization_success/counts_by_arrival_per_strain_beeswarm.pdf", width = 10, height = 15)


df_species_count_plot_diff <- df_species_count_all_combined_mod %>%
                                filter(Experiment %in% c("1", "2")) %>%
                                filter(Type == "bee") %>%
                                filter(arrival %in% c("first", "second", "only")) %>%
                                mutate(abs_counts = ifelse(abs_counts < 1, 1, abs_counts))

pvals <- df_species_count_plot_diff %>%
  filter(!arrival %in% c("dropout")) %>%
  group_by(Strain) %>%
  wilcox_test(abs_counts ~ arrival_mod, data = .) %>%
  # correct p-values for multiple tes
  mutate(y.position = case_when(
            group1 == "first" & group2 == "second" ~ 11,
            group1 == "first" & group2 == "only" ~ 10.5,
            group1 == "only" & group2 == "second" ~ 10,
            TRUE ~ 10.6
  )) %>% # Prevent overlap
  filter(!(grepl("2", group1) | grepl("2", group2)) & !Strain %in% c("ESL0394", "ESL0350")) %>%
  mutate(p = p.adjust(p, method = "BH")) %>%
  mutate(pval = ifelse(p < 0.05, "*", "ns.")) %>%
  mutate(pval = ifelse(p < 0.01, "**", pval)) %>%
  mutate(pval = ifelse(p < 0.001, "***", pval))

ggplot() +
  # geom_bar(data = df_species_count_plot_diff,
  #           aes(x = arrival_mod, y = sample_geq, group = Strain),
  #           position = "identity",
  #           fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_species_count_plot_diff %>% filter(geq_passed != "1"),
                aes(x = arrival_mod, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  geom_beeswarm(data = df_species_count_plot_diff %>% filter(geq_passed == "1"),
                aes(x = arrival_mod, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  facet_wrap(~spec_name_uniq, ncol = 5, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.8, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  geom_signif(data = pvals, manual = TRUE,
              aes(xmin = group1, xmax = group2, annotations = pval, y_position = y.position),
              y_position = 1e+06, tip_length = 0.01
  ) +
  make_theme(setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "none",
             x_hj = 1, x_vj = 1, y_hj = 1) +
  #  add horizontal lines theme.
  theme(
    panel.grid.major.y = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.y = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )
ggsave("03_Analysis/Results/figures/colonization_success/counts_by_arrival_mod_per_strain_beeswarm_ext.pdf", width = 10, height = 15)


df_species_count_plot_diff <- df_species_count_all_combined %>%
                                filter(Experiment %in% c("1", "2")) %>%
                                filter(Type == "bee") %>%
                                filter(arrival %in% c("first", "second", "only")) %>%
                                mutate(abs_counts = ifelse(abs_counts < 1, 1, abs_counts)) %>%
                                mutate(arrival_org = arrival) %>%
                                mutate(arrival_comm = paste0(arrival, "-", CommunityCombo))

ggplot() +
  # geom_bar(data = df_species_count_plot_diff,
  #           aes(x = arrival_comm, y = sample_geq, group = Strain),
  #           position = "identity",
  #           fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_species_count_plot_diff %>% filter(geq_passed != "1"),
                aes(x = arrival_comm, y = abs_counts, fill = arrival, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  geom_beeswarm(data = df_species_count_plot_diff %>% filter(geq_passed == "1"),
                aes(x = arrival_comm, y = abs_counts, fill = arrival, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  facet_wrap(~spec_name_uniq, ncol = 5, scale = "free_x") +
  # scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.8, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(setFill = T, palettefill = "Set1", setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1) +
  #  add horizontal lines theme.
  theme(
    panel.grid.major.y = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.y = element_blank(),
    panel.grid.major.x = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.x = element_blank()
  )
ggsave("03_Analysis/Results/figures/colonization_success/counts_by_arrival_per_comm_per_strain_beeswarm.pdf", width = 15, height = 20)


df_species_count_plot_diff <- df_species_count_all_combined_mod %>%
                                filter(Experiment %in% c("1", "2")) %>%
                                filter(Type == "bee") %>%
                                filter(arrival %in% c("first", "second", "only")) %>%
                                mutate(abs_counts = ifelse(abs_counts < 1, 1, abs_counts)) %>%
                                mutate(arrival_comm = paste0(arrival_mod, CommunityCombo))

ggplot() +
  # geom_bar(data = df_species_count_plot_diff,
  #           aes(x = arrival_comm, y = sample_geq, group = Strain),
  #           position = "identity",
  #           fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_species_count_plot_diff %>% filter(geq_passed != "1"),
                aes(x = arrival_comm, y = abs_counts, fill = arrival, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  geom_beeswarm(data = df_species_count_plot_diff %>% filter(geq_passed == "1"),
                aes(x = arrival_comm, y = abs_counts, fill = arrival, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  facet_wrap(~spec_name_uniq, ncol = 5, scale = "free_x") +
  # scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.8, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(setFill = T, palettefill = "Set1", setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1) +
  #  add horizontal lines theme.
  theme(
    panel.grid.major.y = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.y = element_blank(),
    # panel.grid.major.x = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.x = element_blank()
  )
ggsave("03_Analysis/Results/figures/colonization_success/counts_by_arrival_mod_per_comm_per_strain_beeswarm.pdf", width = 15, height = 20)


# dropout treatments

df_species_count_plot_diff <- df_species_count_all_combined %>%
                                filter(Experiment %in% c("3")) %>%
                                filter(Type == "bee") %>%
                                filter(arrival %in% c("first", "second")) %>%
                                mutate(abs_counts = ifelse(abs_counts < 1, 1, abs_counts))

pvals <- df_species_count_plot_diff %>%
  filter(!arrival %in% c("dropout")) %>%
  group_by(Strain) %>%
  wilcox_test(abs_counts ~ arrival, data = .) %>%
  # correct p-values for multiple tes
  mutate(y.position = case_when(
            group1 == "first" & group2 == "second" ~ 11,
            group1 == "first" & group2 == "only" ~ 10.5,
            group1 == "only" & group2 == "second" ~ 10,
            TRUE ~ 10.6
  )) %>% # Prevent overlap
  filter(!(grepl("2", group1) | grepl("2", group2)) & !Strain %in% c("ESL0394", "ESL0350")) %>%
  mutate(p = p.adjust(p, method = "BH")) %>%
  mutate(pval = ifelse(p < 0.05, "*", "ns.")) %>%
  mutate(pval = ifelse(p < 0.01, "**", pval)) %>%
  mutate(pval = ifelse(p < 0.001, "***", pval))

ggplot() +
  # geom_bar(data = df_species_count_plot_diff,
  #           aes(x = arrival, y = sample_geq, group = Strain),
  #           position = "identity",
  #           fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_species_count_plot_diff %>% filter(geq_passed != "1"),
                aes(x = arrival, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  geom_beeswarm(data = df_species_count_plot_diff %>% filter(geq_passed == "1"),
                aes(x = arrival, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  facet_wrap(~spec_name_uniq, ncol = 5, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.8, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  geom_signif(data = pvals, manual = TRUE,
              aes(xmin = group1, xmax = group2, annotations = pval, y_position = y.position),
              y_position = 1e+06, tip_length = 0.01
  ) +
  make_theme(setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "none",
             x_hj = 1, x_vj = 1, y_hj = 1) +
  #  add horizontal lines theme.
  theme(
    panel.grid.major.y = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.y = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )
ggsave("03_Analysis/Results/figures/colonization_success/dropouts_counts_by_arrival_per_strain_beeswarm.pdf", width = 10, height = 15)


df_species_count_plot_diff <- df_species_count_all_combined %>%
                                filter(Experiment %in% c("3")) %>%
                                filter(Type == "bee") %>%
                                filter(arrival %in% c("first", "second")) %>%
                                mutate(abs_counts = ifelse(abs_counts < 1, 1, abs_counts)) %>%
                                mutate(arrival_comm = paste0(arrival, CommunityCombo))

pvals_all <- data.frame()
for (strain_iter in unique(df_species_count_plot_diff$Strain)) {
  print(strain_iter)
  df_species_count_plot_diff_strain <- df_species_count_plot_diff %>%
                                        filter(Strain == strain_iter)
  pvals_all_strain <- df_species_count_plot_diff_strain %>%
                  select(Strain, arrival_comm, abs_counts) %>%
                  group_by(arrival_comm) %>%
                  filter(!is.na(abs_counts), sum(abs_counts>1) > 1) %>%
                  ungroup() %>%
                  wilcox_test(abs_counts ~ arrival_comm, data = ., p.adjust.method = "BH") %>%
                  mutate(y.position = case_when(
                            grepl("first", group1) & grepl("second", group2) ~ 11,
                            grepl("first", group1) & grepl("first", group2) ~ 10.5,
                            grepl("second", group1) & grepl("second", group2) ~ 10,
                            TRUE ~ 10.6
                  )) %>% # Prevent overlap
                  filter(!(grepl("first", group1) | grepl("first", group2))) %>%
                  mutate(Strain = strain_iter)
  pvals_all <- rbind(pvals_all, pvals_all_strain)
}

pvals <- data.frame()
for (strain_iter in unique(df_species_count_plot_diff$Strain)) {
  df_species_count_plot_diff_strain <- df_species_count_plot_diff %>%
                                        filter(Strain == strain_iter)
  pvals_strain <- df_species_count_plot_diff_strain %>%
                  filter(sample_geq != "0") %>%
                  select(Strain, arrival_comm, abs_counts) %>%
                  group_by(arrival_comm) %>%
                  filter(!is.na(abs_counts), sum(abs_counts>1) > 1) %>%
                  ungroup() %>%
                  wilcox_test(abs_counts ~ arrival_comm, data = ., p.adjust.method = "BH") %>%
                  mutate(y.position = case_when(
                            grepl("first", group1) & grepl("second", group2) ~ 11,
                            grepl("first", group1) & grepl("first", group2) ~ 10.5,
                            grepl("second", group1) & grepl("second", group2) ~ 10,
                            TRUE ~ 10.6
                  )) %>% # Prevent overlap
                  filter(!(grepl("first", group1) | grepl("first", group2))) %>%
                  mutate(Strain = strain_iter)
  pvals <- rbind(pvals, pvals_strain)
}

write.csv(pvals_all, "03_Analysis/Results/tables/priority_effect_strength/pvals_all.csv", row.names = F)


ggplot() +
  # geom_bar(data = df_species_count_plot_diff,
  #           aes(x = arrival_comm, y = sample_geq, group = Strain),
  #           position = "identity",
  #           fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_species_count_plot_diff %>% filter(geq_passed != "1"),
                aes(x = arrival_comm, y = abs_counts, fill = CommunityCombo, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  geom_beeswarm(data = df_species_count_plot_diff %>% filter(geq_passed == "1"),
                aes(x = arrival_comm, y = abs_counts, fill = CommunityCombo, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  facet_wrap(~factor(Strain, strains_uniq), ncol = 5) +
  geom_signif(data = pvals %>% filter(p.adj.signif != "ns"), manual = TRUE,
              aes(xmin = group1, xmax = group2, annotations = p.adj.signif, y_position = y.position),
              tip_length = 0.01
  ) +
  scale_fill_manual(values = c("1\'" = "#000000",
                                "m1" = "#a6cee3",
                                "m4" = "#1f78b4",
                                "m6" = "#fb9a99",
                                "m7" = "#a50f15"
  )) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.8, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(setFill = F, palettefill = "Set1", setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "none",
             x_hj = 1, x_vj = 1, y_hj = 1) +
  #  add horizontal lines theme.
  theme(
    panel.grid.major.y = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.y = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )
ggsave("03_Analysis/Results/figures/colonization_success/dropouts_counts_by_arrival_comm_per_strain_beeswarm.pdf", width = 10, height = 15)

```

Quantify the difference between abundance in full and dropout treatments for 1,4 and 7

```{r}

identify_dropout_type <- function(Strain, CommunityCombo, Treatment) {
  if (CommunityCombo %in% c("1", "1\'", "2", "3", "4", "5", "6")) {
    return("Full")
  } else {
    if (CommunityCombo == "m1") {
      if (Strain %in% c("ESL0825", "ESL0820")) {
        return("Conspecific")
      } else if (Strain %in% c("ESL0170", "ESL0824", "ESL0198", "ESL0827", "ESL0199", "ESL0200", "ESL0819", "ESL0197")) {
        return("Same genus")
      } else {
        return("Different genus")
      }
    }
    if (CommunityCombo == "m4") {
      if (Strain %in% c("ESL0827", "ESL0199")) {
        return("Conspecific")
      } else if (Strain %in% c("ESL0170", "ESL0824", "ESL0198", "ESL0825", "ESL0820", "ESL0200", "ESL0819", "ESL0197")) {
        return("Same genus")
      } else {
        return("Different genus")
      }
    }
    # if (CommunityCombo == "m6") {
    #   # conspecifics ESL0825, ESL0820
    #   # same genus ESL0822
    #   if (Strain %in% c("ESL0825", "ESL0820")) {
    #     return("Conspecific")
    #   } else if (Strain %in% ) {
    #     return("Same genus")
    #   } else {
    #     return("Different genus")
    #   }
    # }
    if (CommunityCombo == "m7") {
      if (Strain %in% c("ESL0263", "ESL0185")) {
        return("Conspecific")
      } else if (Strain %in% c("ESL0183", "ESL0183_ESL0262", "ESL0835", "ESL0184", "ESL0350", "ESL0394", "ESL0186", "ESL0261")) {
        return("Same genus")
      } else {
        return("Different genus")
      }
    }
    return("Unknown")
  }
}

df_comparison_full_drop <- df_species_count_all_combined %>%
                            filter(!ID %in% samples_low_detection) %>%
                              filter(Type == "bee") %>%
                              filter(Experiment %in% c("3")) %>%
                              filter(!grepl("m6", CommunityCombo)) %>%
                              mutate(abs_counts = ifelse(abs_counts < 1, 1, abs_counts)) %>%
                              mutate(arrival_comm = paste0(arrival, CommunityCombo)) %>%
                              filter((arrival %in% c("second") & grepl("m", Treatment)) | arrival %in% c("second", "first") & !grepl("m", Treatment)) %>%
                              filter(!Strain %in% c(strain_without_species_pair)) %>%
                              mutate(dropout_type = Vectorize(identify_dropout_type)(Strain, CommunityCombo, Treatment)) %>%
                              mutate(dropout_type = ifelse(arrival == "first" & dropout_type == "Full", "Full (first)", dropout_type)) %>%
                              mutate(dropout_type = ifelse(arrival == "second" & dropout_type == "Full", "Full (second)", dropout_type))

drop_order <- c("Full (second)", "Conspecific", "Same genus", "Different genus", "Full (first)")
comm_order <- c("1\' (second)", "m1", "m4", "m7", "1' (first)")

strain_order <- c(
  # m1
  "ESL0825",
  "ESL0820",
  # m4
  "ESL0827",
  "ESL0199",
  # m7
  "ESL0185",
  "ESL0263",
  
  "ESL0822",
  "ESL0170",
  "ESL0824",
  "ESL0198",
  "ESL0200",
  "ESL0819",
  # "ESL0197",
  # "ESL0295",
  # "ESL1028",
  "ESL0183",
  "ESL0835",
  # "ESL0394",
  # "ESL0184",
  # "ESL0350",
  "ESL0186",
  "ESL0261"
)

p_list <- list()
for (my_strain in strain_order) {
  df_comparison_full_drop_iter <- df_comparison_full_drop %>%
                                    filter(Strain == my_strain) %>%
                                    mutate(CommunityCombo = ifelse(arrival == "first" & CommunityCombo == "1\'", "1\' (first)", CommunityCombo)) %>%
                                    mutate(CommunityCombo = ifelse(arrival == "second" & CommunityCombo == "1\'", "1\' (second)", CommunityCombo)) %>%
                                    mutate(dropout_type = factor(dropout_type, levels = drop_order))
              
  
  if (my_strain %in% c("ESL0825", "ESL0820")) {
      comparisons_list <- list(c("m1", "1\' (second)"),
                              c("1\' (first)", "1\' (second)"))
      a <- df_comparison_full_drop_iter %>% filter(CommunityCombo == "1' (first)") %>% pull(abs_counts)
      b <- df_comparison_full_drop_iter %>% filter(CommunityCombo == "m1") %>% pull(abs_counts)
    } else {
      if (my_strain %in% c("ESL0827", "ESL0199")) {
        comparisons_list <- list(c("m4", "1\' (second)"),
                              c("1\' (first)", "1\' (second)"))
        a <- df_comparison_full_drop_iter %>% filter(CommunityCombo == "1' (first)") %>% pull(abs_counts)
        b <- df_comparison_full_drop_iter %>% filter(CommunityCombo == "m4") %>% pull(abs_counts)
      } else {
        if (my_strain %in% c("ESL0263", "ESL0185")) {
          comparisons_list <- list(c("m7", "1\' (second)"),
                              c("1\' (first)", "1\' (second)"))
          a <- df_comparison_full_drop_iter %>% filter(CommunityCombo == "1' (first)") %>% pull(abs_counts)
          b <- df_comparison_full_drop_iter %>% filter(CommunityCombo == "m7") %>% pull(abs_counts)
        }
        else {
          comparisons_list <- list(c("1\' (first)", "1\' (second)"))
          a <- NA
          b <- NA
        }
      }
    }


  spec_name_iter <- df_comparison_full_drop_iter$spec_name[1]
  spec_name_iter <- strsplit(spec_name_iter, " ")[[1]][2]

  obs_diff <- median(a, na.rm = TRUE) - median(b, na.rm = TRUE)

  # Combine and resample
  combined <- c(a, b)
  n_a <- length(a)

  n_iter <- 1000
  perm_diffs <- replicate(n_iter, {
    perm <- sample(combined)
    median(perm[1:n_a]) - median(perm[(n_a + 1):length(combined)])
  })

  p_val <- mean(abs(perm_diffs) >= abs(obs_diff))

  # calculates the permutation p-value as the proportion of permutations where the test statistic is as extreme or more extreme than observed

  p <- ggplot() +
    geom_beeswarm(data = df_comparison_full_drop_iter %>% filter(geq_passed != "1"),
                  aes(x = factor(CommunityCombo, comm_order), y = abs_counts, fill = dropout_type, color = geq_passed, alpha = geq_passed),
                  # aes(x = factor(dropout_type, drop_order), y = abs_counts, fill = dropout_type, color = geq_passed, alpha = geq_passed),
                  size = 2.5,
                  cex = 4,
                  corral = "wrap",
                  corral.width = 0.75,
                  shape = 21,
                  stroke = 1
                  ) +
    geom_beeswarm(data = df_comparison_full_drop_iter %>% filter(geq_passed == "1"),
                  aes(x = factor(CommunityCombo, comm_order), y = abs_counts, fill = dropout_type, color = geq_passed, alpha = geq_passed),
                  # aes(x = factor(dropout_type, drop_order), y = abs_counts, fill = dropout_type, color = geq_passed, alpha = geq_passed),
                  size = 2.5,
                  cex = 4,
                  corral = "wrap",
                  corral.width = 0.75,
                  shape = 21,
                  stroke = 1
                  ) +
    # geom_boxplot(data = df_comparison_full_drop_iter %>% filter(geq_passed == "1"),
    #           aes(x = factor(CommunityCombo, comm_order), y = abs_counts, fill = dropout_type, color = geq_passed, alpha = geq_passed),
    #           # aes(x = factor(dropout_type, drop_order), y = abs_counts, fill = dropout_type),
    #           color = "#000000",
    #           width = 0.5
    #           ) +
    # facet_wrap(~spec_name_uniq, ncol = 5) +
    ggtitle(paste0("Strain: ", my_strain, "\n ", spec_name_iter)) +
    labs(x = "Dropout type", y = "Absolute Abundance") +
    scale_fill_manual(values = c("Full (first)" = "#000000",
                                  "Full (second)" = "#999999",
                                  "Full" = "#000000",
                                  "Different genus" = "#a6cee3",
                                  "Same genus" = "#1f78b4",
                                  "Conspecific" = "#a50f15"
    )) +
    # scale_fill_manual(values = c("1\'" = "#000000",
    #                               "m1" = "#a6cee3",
    #                               "m4" = "#1f78b4",
    #                               "m6" = "#fb9a99",
    #                               "m7" = "#a50f15"
    # )) +
    scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
    scale_alpha_manual(values = c("0" = 0.8, "1" = 0.9)) +
    geom_hline(yintercept = median(df_comparison_full_drop_iter %>%
                                    filter(dropout_type == "Full (first)") %>%
                                    pull(abs_counts)
                              ), linetype = "dashed", color = "#000000") +
    geom_hline(yintercept = median(df_comparison_full_drop_iter %>%
                                    filter(dropout_type == "Full (second)") %>%
                                    pull(abs_counts)
                              ), linetype = "dashed", color = "#999999") +
    scale_y_log10(
      breaks = scales::trans_breaks("log10", function(x) 10^x),
      labels = scales::trans_format("log10", scales::math_format(10^.x))
    ) +
    scale_x_discrete(drop = FALSE) +
    # geom_signif(data = df_comparison_full_drop_iter %>% mutate(abs_counts = ifelse(abs_counts < 1, 1, abs_counts))
    #                         , #%>% filter(geq_passed == "1"),
    #             aes(x = factor(CommunityCombo, comm_order), y = abs_counts),
    #             comparisons = comparisons_list,
    #             map_signif_level = TRUE,
    #             tip_length = 0, step_increase = 0.05, textsize = 2
    #     ) +
    annotate("text", x = 1, y = 1e+06, label = paste0("p = ", round(p_val, 3)), size = 4, color = "black") +
    make_theme(setFill = F, palettefill = "Set1", setCol = F, x_angle = 45,
              guide_nrow = 23, leg_pos = "none", leg_size = 8,
              x_hj = 1, x_vj = 1, y_hj = 1) +
    #  add horizontal lines theme.
    theme(
      # panel.grid.major.y = element_line(color = "#999999", size = 0.5),
      panel.grid.major.y = element_blank(),
      panel.grid.minor.y = element_blank(),
      panel.grid.major.x = element_blank(),
      panel.grid.minor.x = element_blank()
    )
  if (df_comparison_full_drop_iter %>% filter(dropout_type == "Conspecific") %>% nrow() > 0) {
    p <- p +
      geom_hline(yintercept = median(df_comparison_full_drop_iter %>%
                                    filter(dropout_type == "Conspecific") %>%
                                    pull(abs_counts)
                              ), linetype = "dashed", color = "#a50f15")
  }
  
  p_list[[my_strain]] <- p
}


pdf("03_Analysis/Results/figures/colonization_success/dropout_comparison_per_strain.pdf", width = 15, height = 20)
marrangeGrob(p_list, ncol = 4, nrow = 5)
dev.off()


```


```{r}
# cleaner plot and quantification of dropout effects

df_comparison_full_drop
write.csv(df_comparison_full_drop, "03_Analysis/Results/tables/dropout_comparison_full_drop.csv", row.names = F)
View(df_comparison_full_drop)

drop_order <- c("Full (second)", "Different genus", "Same genus", "Conspecific", "Full (first)")

df_comparison_full_drop_mod <- df_comparison_full_drop %>%
                                  # filter(!dropout_type %in% c("Full (first)")) %>%
                                  mutate(dropout_type_spl = ifelse(grepl("Conspecific", dropout_type), "Conspecific", "Other")) %>%
                                  filter(!spec_name_uniq %in% c("Lactobacillus melliventris (ESL0394)",
                                                                "Lactobacillus melliventris (ESL0350)",
                                                                "Lactobacillus melliventris (ESL0184)")) %>%
                                  # drop samples which were earlier from Shaded panels indicate missing data due to insufficient sequencing depth
                                  filter(Treatment != "Bm4-A1")

strain_order_mod <- c("Bifidobacterium apousia (ESL0825)",
                      "Bifidobacterium apousia (ESL0820)",
                      "Bifidobacterium sp1. (ESL0827)",
                      "Bifidobacterium sp1. (ESL0199)",
                      "Lactobacillus apis (ESL0185)",
                      "Lactobacillus apis (ESL0263)",
                      "Bifidobacterium asteroides (ESL0822)",
                      "Bifidobacterium asteroides (ESL0170)" ,
                      "Bifidobacterium polysaccharolyticum (ESL0824)",
                      "Bifidobacterium polysaccharolyticum (ESL0198)",
                      "Bifidobacterium sp2. (ESL0200)",
                      "Bifidobacterium sp2. (ESL0819)",
                      "Lactobacillus helsingborgensis (ESL0183)",
                      "Lactobacillus helsingborgensis (ESL0835)",
                      "Lactobacillus kullabergensis (ESL0261)",
                      "Lactobacillus kullabergensis (ESL0186)"
                      # "Lactobacillus melliventris (ESL0394)",
                      # "Lactobacillus melliventris (ESL0350)",
                      # "Lactobacillus melliventris (ESL0184)"
                      )

# --- SETTINGS ---
label_mult <- 1.12  # how far above median to place labels (multiplicative; works on log10 scale)

# --- MEDIANS & % vs Full (first) ---
meds <- df_comparison_full_drop_mod %>%
  # choose whether to restrict medians to passed-only:
  # filter(geq_passed == "1") %>%
  group_by(spec_name_uniq, dropout_type) %>%
  summarise(med = median(abs_counts), .groups = "drop") %>%
  group_by(spec_name_uniq) %>%
  mutate(
    med_full_first = med[dropout_type == "Full (first)"][1],
    log10_med_full = log10(med_full_first),
    log10_med = log10(med),
    pct_vs_full_first = log10_med / log10_med_full * 100,
    fold_change    = med / med_full_first,
    log2_fc        = log2(fold_change)
  ) %>%
  ungroup() %>%
  # keep only rows where baseline exists
  filter(!is.na(med_full_first)) %>%
  mutate(
    dropout_type = factor(dropout_type, levels = drop_order),
    y_lab = 10e+9,
    # y_lab = med * label_mult,
    pct_lab = ifelse(dropout_type == "Full (first)", "100%", sprintf("%.1f%%", pct_vs_full_first)),
    lab_log2fc = ifelse(dropout_type == "Full (first)", "1x", sprintf("%.3fx", log2_fc)),
    lab_fc = ifelse(dropout_type == "Full (first)", "1x", sprintf("%.2fx", fold_change))
  )


ggplot() +
  geom_beeswarm(data = df_comparison_full_drop_mod %>% filter(geq_passed != "1"),
                # aes(x = factor(dropout_type, drop_order), y = abs_counts, fill = dropout_type, color = geq_passed, alpha = geq_passed),
                aes(x = factor(dropout_type, drop_order), y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 0.5
                ) +
  geom_beeswarm(data = df_comparison_full_drop_mod %>% filter(geq_passed == "1"),
                # aes(x = factor(dropout_type, drop_order), y = abs_counts, fill = dropout_type, color = geq_passed, alpha = geq_passed),
                aes(x = factor(dropout_type, drop_order), y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 0.5
                ) +
  # geom_boxplot(data = df_comparison_full_drop_mod %>% filter(geq_passed == "1"),
  #           # aes(x = factor(dropout_type, drop_order), y = abs_counts, fill = Strain),
  #           aes(x = factor(dropout_type, drop_order), y = abs_counts, fill = dropout_type),
  #           alpha = 0.25,
  #           # aes(x = factor(dropout_type, drop_order), y = abs_counts, fill = dropout_type),
  #           color = "#000000",
  #           width = 0.5
  #           ) +
  # stat_summary(data = df_comparison_full_drop_mod,
  #               fun = median, geom = "crossbar", width = 0.5, color = "#000000",
  #               aes(x = factor(dropout_type, drop_order), y = abs_counts, fill = dropout_type),
  #               # aes(x = factor(dropout_type, drop_order), y = abs_counts, fill = Strain),
  # ) +
  geom_point(data = meds,
             aes(x = dropout_type, y = med),
             shape = 45, size = 15, color = "black", stroke = 1.5, inherit.aes = FALSE
  ) +
  geom_text(
    data = meds,
    aes(x = dropout_type, y = y_lab, label = pct_lab),
    size = 3, color = "black", inherit.aes = FALSE
  ) +
  facet_wrap(~factor(spec_name_uniq, strain_order_mod), ncol = 3) +
  # ggtitle(paste0("Strain: ", my_strain, "\
  labs(x = "Dropout type", y = "Absolute Abundance") +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  # scale_fill_manual(values = c("Full (second)" = "#b9b9b9",
  #                              "Full (first)" = "#000000",
  #                              "Different genus" = "#a6cee3",
  #                              "Same genus" = "#1f78b4",
  #                              "Conspecific" = "#a50f15"
  #   )) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.8, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(setFill = F, palettefill = "Set1", setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", leg_size = 8,
             x_hj = 1, x_vj = 1, y_hj = 1) +
  # add verticle grid lines
  theme(
    panel.grid.major.x = element_line(color = "#999999", size = 0.5, linetype = "dashed"),
    panel.grid.minor.x = element_blank(),
    panel.grid.major.y = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.y = element_blank()
  )
  # ggsave("03_Analysis/Results/figures/colonization_success/dropout_comparison_per_strain_summary.pdf", width = 10, height = 15)
  ggsave("03_Analysis/Results/figures/colonization_success/dropout_comparison_per_strain_summary.pdf", width = 10, height = 15)
```


```{r}
# run 03_Analysis/priority_effect_strength.py uncommenting simple and advanced

df_priority_effect_strength_simple <- read.csv("03_Analysis/Results/tables/priority_effect_strength/full_combos_all.csv") %>%
                                        mutate(CommunityCombo = as.factor(CommunityCombo)) %>%
                                          filter(!Strain %in% c("ESL0295"))
# df_priority_effect_strength_simple_sim <- read.csv("03_Analysis/Results/tables/priority_effect_strength/full_combos_all_by_sim.csv") %>%
#                                         mutate(CommunityCombo = as.factor(CommunityCombo)) %>%
#                                           filter(!Strain %in% c("ESL0295"))


ggplot() +
  geom_boxplot(data = df_priority_effect_strength_simple %>% filter(!grepl("m", CommunityCombo)),
            aes(x = factor(Strain, strains_uniq), y = priority_effect_strength_median, fill = factor(Strain, strains_uniq)),
            color = "#000000",
            width = 0.5
            ) +
  geom_point(data = df_priority_effect_strength_simple %>% filter(!grepl("m", CommunityCombo)),
            aes(x = factor(Strain, strains_uniq), y = priority_effect_strength_median, shape = CommunityCombo),
            # shape = 2,
            color = "#000000",
            fill = "#000000",
            size = 2
            ) +
  # geom_path(data = df_priority_effect_strength_simple %>% filter(!grepl("m", CommunityCombo)),
  #           aes(x = factor(Strain, strains_uniq), y = priority_effect_strength_median, group = CommunityCombo),
  #           color = "#000000",
  #           size = 0.25
  #           ) +
  facet_grid(~spec_name, scale = "free_x") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "black") +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_shape_manual(values = c(
    "1" = 22,
    "2" = 21,
    "3" = 20,
    "4" = 23,
    "5" = 24,
    "6" = 25,
    "1\'" = 0
  )) +
  make_theme(theme = theme_bw(), setFill = F, setCol = T, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/priority_effect_strength_compared/priority_effect_strength_simple.pdf", width = 20, height = 15)


ggplot() +
  geom_boxplot(data = df_priority_effect_strength_simple %>% filter(!grepl("m", CommunityCombo)),
            aes(x = factor(Strain, strains_uniq), y = priority_effect_strength_median, fill = factor(Strain, strains_uniq)),
            color = "#000000",
            width = 0.5
            ) +
  geom_point(data = df_priority_effect_strength_simple %>% filter(!grepl("m", CommunityCombo)),
            aes(x = factor(Strain, strains_uniq), y = priority_effect_strength_median),
            # shape = 2,
            color = "#000000",
            fill = "#000000",
            size = 2
            ) +
  # geom_path(data = df_priority_effect_strength_simple %>% filter(!grepl("m", CommunityCombo)),
  #           aes(x = factor(Strain, strains_uniq), y = priority_effect_strength_median, group = CommunityCombo),
  #           color = "#000000",
  #           size = 0.25
  #           ) +
  facet_grid(~spec_name, scale = "free_x") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "black") +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  make_theme(theme = theme_bw(), setFill = F, setCol = T, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/priority_effect_strength_compared/priority_effect_strength_simple_wo_shape.pdf", width = 20, height = 15)
```



```{r}
# check stats using a linear mixed effects model
combos_full <- c("1", "2", "3", "4", "5", "6")

df_lme_data <- df_species_count_all_combined %>%
                  filter(CommunityCombo %in% combos_full) %>%
                  filter(Type == "bee") %>%
                  filter(arrival %in% c("first", "second", "only")) %>%
                  mutate(arrival = recode(arrival,
                                          "second" = "2_second",
                                          "first" = "1_first",
                                          "only" = "0_only")) %>%
                  mutate(abs_counts = ifelse(abs_counts < 1, 1, abs_counts)) %>%
                  # log transform the abs_counts
                  mutate(abs_counts_trans = log10(abs_counts)) %>%
                  mutate(abs_counts_trans = ifelse(abs_counts_trans == 0, 0.1, abs_counts_trans))
                  

model_results_trans <- list()
model_summaries_trans <- list()

for (strain in strains_uniq) {
  # Subset data for the current strain
  dat <- df_lme_data %>%
    filter(Strain == strain)  # Filter for the specific strain

  # Check if there's enough data for modeling
  if (nrow(dat) > 5 && length(unique(dat$arrival)) > 1) {  # Ensure sufficient data & variability

  # Fit the linear mixed-effects model
  mod <- glmer(abs_counts_trans ~ arrival + (1 | CommunityCombo), 
                 family = Gamma(link = "log"),
                 data = dat)
  
  model_results_trans[[as.character(strain)]] <- mod
  # Store the model summary
  model_summaries_trans[[as.character(strain)]] <- summary(mod)

      # Print results
    cat("\n\n----------\nStrain:", strain, "\n")
    print(summary(mod))
  } else {
    cat("\n\n----------\nStrain:", strain, " - Not enough data for modeling\n")
  }
}
# Initialize a data frame to hold the results
summary_results <- data.frame(
  Strain = character(0),
  Arrival_First_Estimate = numeric(0),
  Arrival_Second_Estimate = numeric(0),
  Arrival_First_pValue = numeric(0),
  Arrival_Second_pValue = numeric(0),
  Random_Effect_Variance = numeric(0)
)

# Loop through each strain and extract the relevant information
for (strain in strains_uniq) {
  # Subset the data for the current strain
  dat <- df_lme_data %>%
    filter(Strain == strain)
  
  if (nrow(dat) > 5 && length(unique(dat$arrival)) > 1) {
    # Fit the model for transformed abundance data
    mod <- model_results_trans[[as.character(strain)]]
    
    # Extract fixed effects
    fixed_effects <- fixef(mod)
    
    # Extract p-values (if available)
    p_values <- summary(mod)$coefficients[, 4]
    
    # Extract the random effect variance for CommunityCombo
    random_effects_var <- as.numeric(VarCorr(mod)$CommunityCombo[1])  # Extract variance for random effect
    
    # Store the results in the summary data frame
    summary_results <- rbind(summary_results, data.frame(
      Strain = strain,
      Arrival_First_Estimate = fixed_effects["arrival1_first"],
      Arrival_Second_Estimate = fixed_effects["arrival2_second"],
      Arrival_First_pValue = p_values["arrival1_first"],
      Arrival_Second_pValue = p_values["arrival2_second"],
      Random_Effect_Variance = random_effects_var,
      row.names = strain
    ))
  } else {
    cat("\n\n----------\nStrain:", strain, " - Not enough data for modeling\n")
  }
}

# View the summarized results across all strains
print(summary_results)


# Initialize a data frame to hold the results
summary_results <- data.frame(
  Strain = character(0),
  Arrival_First_Estimate = numeric(0),
  Arrival_Second_Estimate = numeric(0),
  Arrival_First_pValue = numeric(0),
  Arrival_Second_pValue = numeric(0),
  Arrival_First_Adj_pValue = numeric(0),
  Arrival_Second_Adj_pValue = numeric(0),
  Arrival_First_Adj_bh_pValue = numeric(0),
  Arrival_Second_Adj_bh_pValue = numeric(0),
  Random_Effect_Variance = numeric(0),
  Model_AIC = numeric(0),
  Sample_Size = numeric(0)
)

# Initialize lists to store the p-values for BH adjustment
all_pvalues_first <- c()
all_pvalues_second <- c()
strains <- c()

# Loop through each strain and extract the relevant information
for (strain in strains_uniq) {
  # Subset the data for the current strain
  dat <- df_lme_data %>%
    filter(Strain == strain)
  
  # Check if there's enough data for modeling
  if (nrow(dat) > 5 && length(unique(dat$arrival)) > 1) {
    # Fit the model for transformed abundance data
    mod <- model_results_trans[[as.character(strain)]]
    
    # Extract fixed effects
    fixed_effects <- fixef(mod)
    
    # Extract p-values (if available)
    p_values <- summary(mod)$coefficients[, 4]
    
    # Extract the random effect variance for CommunityCombo
    random_effects_var <- as.numeric(VarCorr(mod)$CommunityCombo[1])  # Extract variance for random effect
    
    # Get the AIC for the model
    model_aic <- AIC(mod)
    
    # Get the sample size (number of rows used in the model)
    sample_size <- nrow(dat)
    
    # Store the p-values for the BH adjustment later
    all_pvalues_first <- c(all_pvalues_first, p_values["arrival1_first"])
    all_pvalues_second <- c(all_pvalues_second, p_values["arrival2_second"])
    strains <- c(strains, strain)
    
    # Store the results in the summary data frame
    summary_results <- rbind(summary_results, data.frame(
      Strain = strain,
      Arrival_First_Estimate = fixed_effects["arrival1_first"],
      Arrival_Second_Estimate = fixed_effects["arrival2_second"],
      Arrival_First_pValue = p_values["arrival1_first"],
      Arrival_Second_pValue = p_values["arrival2_second"],
      Random_Effect_Variance = random_effects_var,
      Model_AIC = model_aic,
      Sample_Size = sample_size,
      row.names = strain  # Assign the strain name as the row name
    ))
  } else {
    cat("\n\n----------\nStrain:", strain, " - Not enough data for modeling\n")
  }
}

# Adjust p-values for multiple testing using Benjamini-Hochberg (FDR) method
adjusted_pvalues_first_bon <- p.adjust(all_pvalues_first, method = "bonferroni")
adjusted_pvalues_second_bon <- p.adjust(all_pvalues_second, method = "bonferroni")
adjusted_pvalues_first_bh <- p.adjust(all_pvalues_first, method = "BH")
adjusted_pvalues_second_bh <- p.adjust(all_pvalues_second, method = "BH")

# Add adjusted p-values to the summary
summary_results$Arrival_First_Adj_pValue <- adjusted_pvalues_first_bon
summary_results$Arrival_Second_Adj_pValue <- adjusted_pvalues_second_bon
summary_results$Arrival_First_Adj_bh_pValue <- adjusted_pvalues_first_bh
summary_results$Arrival_Second_Adj_bh_pValue <- adjusted_pvalues_second_bh

# visualize with *** in a column

summary_results$Arrival_First_Signif <- ifelse(summary_results$Arrival_First_Adj_pValue < 0.05, "*", "ns")
summary_results$Arrival_Second_Signif <- ifelse(summary_results$Arrival_Second_Adj_pValue < 0.05, "*", "ns")
summary_results$Arrival_First_Signif <- ifelse(summary_results$Arrival_First_Adj_pValue < 0.01, "***", summary_results$Arrival_First_Signif)
summary_results$Arrival_Second_Signif <- ifelse(summary_results$Arrival_Second_Adj_pValue < 0.01, "***", summary_results$Arrival_Second_Signif)

summary_results$Arrival_First_bh_Signif <- ifelse(summary_results$Arrival_First_Adj_bh_pValue < 0.05, "*", "ns")
summary_results$Arrival_Second_bh_Signif <- ifelse(summary_results$Arrival_Second_Adj_bh_pValue < 0.05, "*", "ns")
summary_results$Arrival_First_bh_Signif <- ifelse(summary_results$Arrival_First_Adj_bh_pValue < 0.01, "***", summary_results$Arrival_First_bh_Signif)
summary_results$Arrival_Second_bh_Signif <- ifelse(summary_results$Arrival_Second_Adj_bh_pValue < 0.01, "***", summary_results$Arrival_Second_bh_Signif)

# View the final summary results
summary_results

write.csv(summary_results, "03_Analysis/Results/tables/df_priority_effect_strength_model_results.csv")

# summary(model_results_trans[[as.character("ESL0263")]])
```


At the species level show it is not different!

```{r}
df_counts_heatmap <- df_species_count_all_combined %>%
              # mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
              # filter(CommunityCombo %in% c(my_comunity_combo, "1\'")) %>%
              filter(Type == "bee") %>% filter(Experiment != "pilot") %>%
              filter(!Treatment %in% c("extraction_blank")) %>%
              filter(!ID %in% samples_low_detection) %>%
              select(spec_name, ID, Strain, geq_passed, CommunityCombo, Treatment_simp) %>%
              mutate(geq_passed = as.numeric(as.character(geq_passed))) %>%
              unique() %>%
              group_by(ID, spec_name) %>%
              reframe(ID, CommunityCombo, Treatment_simp, geq_passed = sum(geq_passed)) %>%
              ungroup() %>%
              mutate(geq_passed = as.factor(geq_passed))

p_list <- list()
for (comm in unique(df_counts_heatmap$CommunityCombo)) {
  if (grepl("m", comm)) {
    a_ratio <- 1
  } else {
    a_ratio <- 1.5
  }
  p <- ggplot() +
        geom_tile(data = df_counts_heatmap %>% filter(CommunityCombo == comm),
                    aes(y = spec_name, x = ID, fill = geq_passed),
                    color = NA) +
        facet_grid(~factor(Treatment_simp, levels = c("A", "B", "AB", "BA", "AmB", "BmA", "MD")), scale = "free") +
        labs(x = "Sample ID", y = "Species", fill = "Number of strains above detection limit") +
        scale_fill_manual(values = c("0" = "#d9d9d9", "1" = "#377eb8", "2" = "#4daf4a", "3" = "#e30f10")) +
        ggtitle(paste0("Community ", comm)) +
        make_theme(setFill = F, palettefill = "Set1", y_size = 15,
                  setCol = F, x_angle = 45, modify_guide = F,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1) +
        theme(aspect.ratio = a_ratio)
  p_list[[comm]] <- p
}

pdf("03_Analysis/Results/figures/priority_effects_experiment/spec_abundance_heatmap.pdf", width = 25, height = 20)
marrangeGrob(p_list, nrow = 3, ncol = 2)
dev.off()

df_counts_heatmap <- df_species_count_all_combined %>%
              # mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
              # filter(CommunityCombo %in% c(my_comunity_combo, "1\'")) %>%
              filter(Type == "bee") %>% filter(Experiment != "pilot") %>%
              filter(!Treatment %in% c("extraction_blank")) %>%
              filter(!ID %in% samples_low_detection) %>%
              select(spec_name, ID, Strain, abs_counts, CommunityCombo, Treatment_simp) %>%
              unique() %>%
              group_by(ID, spec_name) %>%
              reframe(ID, CommunityCombo, Treatment_simp, abs_counts_spec = sum(abs_counts)) %>%
              ungroup()

p_list <- list()
for (comm in unique(df_counts_heatmap$CommunityCombo)) {
  if (grepl("m", comm)) {
    a_ratio <- 1
  } else {
    a_ratio <- 1.5
  }
  p <- ggplot() +
        geom_tile(data = df_counts_heatmap %>% filter(CommunityCombo == comm),
                    aes(y = spec_name, x = ID, fill = abs_counts_spec),
                    color = NA) +
        facet_grid(~factor(Treatment_simp, levels = c("A", "B", "AB", "BA", "AmB", "BmA", "MD")), scale = "free") +
        labs(x = "Sample ID", y = "Species", fill = "Number of strains above detection limit") +
        ggtitle(paste0("Community ", comm)) +
        scale_fill_gradient2(
            low = "#ffffff",
            mid = brewer.pal(9, "PuBu")[4],
            high = brewer.pal(9, "BuGn")[7],
            na.value = "#ffffff",
            midpoint = 7,
            trans = "log10"
          ) +
        make_theme(setFill = F, palettefill = "Set1", y_size = 15,
                  setCol = F, x_angle = 45, modify_guide = F,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1) +
        theme(aspect.ratio = a_ratio)
  p_list[[comm]] <- p
}

pdf("03_Analysis/Results/figures/priority_effects_experiment/spec_abundance_counts_heatmap.pdf", width = 25, height = 20)
marrangeGrob(p_list, nrow = 3, ncol = 2)
dev.off()

ggplot() +
  geom_tile(data = df_species_count_all %>%
                    filter(Experiment != "none") %>%
                    # filter(Treatment != "dropout_community") %>%
                    mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                    mutate(treattype = ifelse(Treatment == "dropout_community", "dropout_community", "full")) %>%
                    unique() %>%
                    filter(Type != "bee"),
  aes(y = ID, x = Strain, fill = abs_counts_org, color = geq_passed)) +
  facet_grid(treattype+factor(CommunityCombo)~spec_name, scale = "free", space = "free") +
  scale_fill_viridis_c(trans = "log10", option = "mako", direction = -1, na.value = NA) +
  # scale_fill_manual(values = c("0" = "#ffffff", "1" = "#000000"), na.value = NA) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000"), na.value = NA) +
  make_theme(setCol = F, setFill = F, leg_pos = "right", modify_guide = F,
             x_angle = 45, x_hj = 1, x_vj = 1,
             y_hj = 1) +
  theme(strip.text = element_text(size = 5, angle = 55))
ggsave("03_Analysis/Results/figures/priority_effects_experiment/community_alq_heatmap_all_used.pdf")


df_count_vs_load <- df_species_count_all %>%
                      filter(Type == "bee") %>%
                      filter(geq_passed == "1") %>%
                      filter(!ID %in% samples_low_detection) %>%
                      group_by(ID) %>%
                      summarise(Treatment_simp, CommunityCombo, total_load = bac_copies, n_strains = n_distinct(Strain))

ggplot() +
  geom_point(data = df_count_vs_load,
                # aes(x = n_strains, y = total_load),
                aes(x = n_strains, y = total_load, color = Treatment_simp),
    size = 2
  ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  # facet_wrap(~Treatment_simp) +
  stat_cor(data = df_count_vs_load,
                aes(x = n_strains, y = total_load),
            p.accuracy = 0.001, r.accuracy = 0.01, method = "spearman") +
  geom_smooth(data = df_count_vs_load,
                aes(x = n_strains, y = total_load),
                method = "lm", se = T) +
  stat_cor(data = df_count_vs_load %>% filter(n_strains >= 10),
                aes(x = n_strains, y = total_load),
            label.x.npc = "right", label.y.npc = "top",
            label.x = 15,
            p.accuracy = 0.001, r.accuracy = 0.01, method = "spearman") +
  geom_smooth(data = df_count_vs_load %>% filter(n_strains >= 10),
                aes(x = n_strains, y = total_load),
                method = "lm", se = T) +
  labs(y = "Total bacterial load", x = "Number of strains detected", color = "Treatment") +
  geom_vline(xintercept = 10, linetype = "dashed") +
  geom_vline(xintercept = 22, linetype = "solid") +
  scale_color_manual(values = TreatmentColors) +
  make_theme(setFill = F, setCol = F, x_angle = 45,
             leg_pos = "right", guide_nrow = 6,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/total_load_vs_n_strains.pdf")

df_load_per_species <- df_species_count_all %>%
                      filter(Experiment != "3") %>%
                      mutate(Treatment_simp = ifelse(Treatment_simp == "extraction_blank", "blank", Treatment_simp)) %>%
                      filter(spec_name != "unknown") %>%
                      filter(!is.na(spec_name)) %>%
                      filter(Type == "bee") %>%
                      # filter(geq_passed == "1") %>%
                      filter(!ID %in% samples_low_detection) %>%
                      group_by(ID, spec_name) %>%
                      summarise(Treatment_simp, CommunityCombo, species_load = sum(abs_counts)) %>%
                      mutate(species_load = ifelse(species_load < 1, 1, species_load))

pvals <- df_load_per_species %>%
            group_by(spec_name) %>%
            wilcox_test(species_load ~ Treatment_simp, p.adjust.method = "BH",
                        comparisons = list(c("A", "AB"),
                                           c("B", "BA")
                                      )
            ) %>%
            mutate(y.position = case_when(
                      group1 == "A" & group2 == "AB" ~ 10.5,
                      group1 == "B" & group2 == "BA" ~ 10.5,
                      group1 == "A" & group2 == "BA" ~ 10.5,
                      group1 == "B" & group2 == "AB" ~ 10.5,
                      TRUE ~ 10.6
            )) # Prevent overlap

            

ggplot() +
  # geom_boxplot(data = df_load_per_species,
  #               aes(x = Treatment_simp, y = species_load, fill = Treatment_simp),
  #               # aes(x = spec_name, y = species_load, color = Treatment_simp),
  #   outlier.shape = NA
  # ) +
  # geom_point(data = df_load_per_species,
  #               aes(x = Treatment_simp, y = species_load),
  #   size = 1,
  #   color = "#000000"
  # ) +
  geom_beeswarm(data = df_load_per_species,
                aes(x = Treatment_simp, y = species_load, fill = Treatment_simp),
                size = 2.5,
                cex = 4,
                shape = 21,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.9,
                stroke = 1
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  # # add wilcox on top of the boxplots and add BH correction of p values
  # geom_signif(data = df_load_per_species,
  #               aes(x = Treatment_simp, y = species_load),
  #               test = "wilcox.test", test.args = list(exact = F, p.adjust.method = "BH"),
  #               comparisons = list(c("A", "AB"),
  #                                  c("B", "BA")
  #                             ),
  #               map_signif_level = T,
  #               facet_variable = "spec_name") +
  geom_signif(data = pvals, aes(xmin = group1, xmax = group2, annotations = p.adj.signif, y_position = y.position),
              manual = TRUE,
              inherit.aes = FALSE, tip_length = 0.01) +
  facet_wrap(~spec_name, scale = "free_x", ncol = 3) +
  labs(y = "Total bacterial load (sum of strain of species)",
       x = "Treatment", fill = "Treatment") +
  scale_color_manual(values = TreatmentColors) +
  scale_fill_manual(values = TreatmentColors) +
  make_theme(setFill = F, setCol = F, x_angle = 45,
             leg_pos = "right", guide_nrow = 6,
             x_hj = 1, x_vj = 1, y_hj = 1) +
  #  add horizontal lines theme.
  theme(
    panel.grid.major.y = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.y = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/species_load_vs_treatment.pdf", width = 15, height = 15)

df_species_count_all_adj <- df_species_count_all_combined %>%
                              filter(!ID %in% samples_low_detection) %>%
                              filter(Type == "bee") %>%
                              filter(Experiment %in% c("1", "2", "3")) %>%
                              mutate(abs_counts = ifelse(abs_counts < 1, 1, abs_counts)) %>%
                              mutate(arrival_comm = paste0(arrival, CommunityCombo)) %>%
                              filter(arrival %in% c("first", "second"))

ggplot() +
  # geom_bar(data = df_species_count_all_adj,
  #           aes(x = arrival_comm, y = sample_geq, group = Strain),
  #           position = "identity",
  #           fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_species_count_all_adj %>% filter(geq_passed != "1"),
                aes(x = arrival_comm, y = abs_counts, fill = arrival, color = geq_passed, alpha = geq_passed),
                size = 2,
                cex = 3,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  geom_beeswarm(data = df_species_count_all_adj %>% filter(geq_passed == "1"),
                aes(x = arrival_comm, y = abs_counts, fill = arrival, color = geq_passed, alpha = geq_passed),
                size = 2,
                cex = 3,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1
                ) +
  facet_wrap(~spec_name_uniq, ncol = 5, scale = "free_x") +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_fill_manual(values = c("first" = "#a6cee3",
                               "second" = "#ff7f00", "MD" = "#999999")) +
  scale_alpha_manual(values = c("0" = 0.8, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(setFill = F, palettefill = "Set1", setCol = F, x_angle = 45,
             x_size = 10,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1) +
  #  add horizontal lines theme.
  theme(
    panel.grid.major.y = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.y = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )
ggsave("03_Analysis/Results/figures/colonization_success/full_counts_by_arrival_comm_per_strain_beeswarm.pdf", width = 20, height = 20)

```

```{r}

desired_levels <- c("only", "first", "both", "second", "MD")

df_species_count_all_adj <- df_species_count_all_combined %>%
                              filter(!ID %in% samples_low_detection) %>%
                              filter(Type == "bee") %>%
                              filter(Experiment %in% c("pilot")) %>%
                              mutate(abs_counts = ifelse(abs_counts < 1, 1, abs_counts)) %>%
                              filter(arrival %in% c("first", "second", "only", "both", "MD")) %>%
                              group_by(Strain) %>%
                              mutate(sum = sum(abs_counts > 1)) %>%
                              filter(sum > 2) %>%
                              ungroup() %>%
                              select(-sum) %>%
                              mutate(Treatment_mod = case_when(
                                Treatment == "0-3" ~ "MD",
                                Treatment == "A-0" ~ "Day0",
                                Treatment == "B-0" ~ "Day0",
                                Treatment == "A-3" ~ "Day3",
                                Treatment == "B-3" ~ "Day3",
                                Treatment == "AB-0" ~ "Day0",
                                Treatment == "AB-3" ~ "Day3",
                                Treatment == "BA-3" ~ "Day3",
                                Treatment == "AB-1" ~ "Day1",
                                Treatment == "BA-1" ~ "Day1",
                                Treatment == "A-3_14" ~ "Day3",
                                Treatment == "B-3_14" ~ "Day3",
                                # Treatment == "A-3_14" ~ "Day3_extra",
                                # Treatment == "B-3_14" ~ "Day3_extra",
                                TRUE ~ Treatment
                              )) %>%
                              filter(Strain != "unknown") %>%
                              filter(Strain != "ESL0197")
                              # filter(Strain != "ESL0197") %>%
                              # mutate(arrival = paste0(arrival, Treatment))

p_list <- list()
for (a_strain in unique(df_species_count_all_adj$Strain)) {
  df_species_count_all_adj_strain <- df_species_count_all_adj %>%
                                  filter(Strain == a_strain)
  p <- ggplot() +
  # geom_boxplot(data = df_species_count_all_adj_strain %>% filter(geq_passed == "1"),
  #               aes(x = arrival, y = abs_counts, fill = Treatment_mod),
  #               # aes(x = Treatment, y = abs_counts, fill = arrival, color = geq_passed, alpha = geq_passed),
  #               ) +
  geom_beeswarm(data = df_species_count_all_adj_strain %>% filter(geq_passed != "1"),
                # aes(x = arrival, y = abs_counts, fill = Treatment_mod, color = geq_passed, alpha = geq_passed),
                aes(x = Treatment, y = abs_counts, fill = arrival, color = geq_passed, alpha = geq_passed),
                size = 3,
                cex = 4,
                corral = "wrap",
                corral.width = 0.25,
                shape = 21,
                stroke = 1
                ) +
  geom_beeswarm(data = df_species_count_all_adj_strain %>% filter(geq_passed == "1"),
                # aes(x = arrival, y = abs_counts, fill = Treatment_mod, color = geq_passed, alpha = geq_passed),
                aes(x = Treatment, y = abs_counts, fill = arrival, color = geq_passed, alpha = geq_passed),
                size = 3,
                cex = 4,
                corral = "wrap",
                corral.width = 0.25,
                shape = 21,
                stroke = 1
                ) +
  # facet_grid(~spec_name_uniq+factor(arrival, levels = desired_levels), scale = "free_x") +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.8, "1" = 0.9)) +
  ggtitle(paste0("Strain: ", a_strain)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  # scale_x_discrete(limits = desired_levels) +
  # scale_fill_manual(values = c("Day0" = "#377eb8",
  #                              "Day1" = "#4daf4a",
  #                              "Day3" = "#e30f10",
  #                              "Day3_extra" = "#ff7f00", "MD" = "#999999")) +
  scale_fill_manual(values = c("first" = "#377eb8",
                               "second" = "#4daf4a",
                               "only" = "#e30f10",
                               "both" = "#ff7f00", "MD" = "#999999")) +
  make_theme(setFill = F, palettefill = "Set1", setCol = F, x_angle = 45,
             x_size = 10,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1) +
  #  add horizontal lines theme.
  theme(
    panel.grid.major.y = element_line(color = "#999999", size = 0.5),
    panel.grid.minor.y = element_blank(),
    panel.grid.major.x = element_line(color = "#b9b9b9", size = 0.5),
    panel.grid.minor.x = element_blank()
  )
  p_list[[a_strain]] <- p
}

pdf("03_Analysis/Results/figures/colonization_success/full_counts_by_arrival_comm_per_strain_beeswarm_pilot.pdf", width = 15, height = 20)
marrangeGrob(p_list, nrow = 5, ncol = 4)
dev.off()

```


```{r}

samples_low_detection %>% 
  unique() %>% length()

df_species_count_all %>% filter(Type == "bee") %>% pull(ID) %>%
  unique() %>% length()

df_species_count_all_combined %>% group_by(ID) %>%
  mutate(is_low = ifelse(ID %in% samples_low_detection, "yes", "")) %>%
  summarise(Type, Experiment, Plate, Treatment, is_low, Ct_Mean_bac, Ct_SD_bac, bac_copies, total_counts, sum_present = sum(geq_passed == "1")) %>%
  arrange(Experiment, Type, Treatment, is_low) %>%
  unique %>% write.csv("03_Analysis/Results/tables/all_sample_metadata_ext.csv", row.names = F, quote = F)
```


### colonization success (total bacterial abundance) across all treatments
```{r}
ggplot() +
  geom_bar(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                        filter(!ID %in% samples_low_detection) %>%
                        filter(Type == "bee") %>%
                        filter(Experiment != "none"),
           aes(x = Treatment, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.01,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                        filter(!ID %in% samples_low_detection) %>%
                        filter(Type == "bee") %>%
                        filter(Experiment != "none") %>%
                        group_by(ID) %>%
                        summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                        unique()
                        ,
                aes(x = Treatment, y = abs_counts_total, fill = Treatment_simp),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.5,
                shape = 21,
                stroke = 1,
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  geom_hline(yintercept = 1e+10, linetype = "dotted") +
  geom_hline(yintercept = 1e+9, linetype = "dotted") +
  geom_hline(yintercept = 1e+8, linetype = "dotted") +
  geom_hline(yintercept = 1e+7, linetype = "dotted") +
  labs(x = "Treatment", y = "Absolute bacterial abundance", fill = "Treatment type") +
  scale_fill_manual(values = c(TreatmentColors, c("MD" = "#ffffff"))) +
  facet_wrap(~Experiment, scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_total.pdf", width = 20, height = 20)
```



### colonization success of each bacterial species (total bacterial abundance) across all treatments
```{r}

ggplot() +
  geom_bar(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Experiment == "1") %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "A1", "B1")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
           aes(x = spec_name, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.1,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Experiment == "1") %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "A1", "B1")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                aes(x = spec_name, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  facet_wrap(~Treatment, scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_species_combo1.pdf", width = 20, height = 20)

ggplot() +
  geom_bar(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Experiment == "3") %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "A1", "B1")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
           aes(x = spec_name, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.1,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Experiment == "3") %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "A1", "B1")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                aes(x = spec_name, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  facet_wrap(~Treatment, scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_species_combo1_rep.pdf", width = 20, height = 20)


ggplot() +
  geom_bar(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A2-0", "B2-0", "A2-B2", "B2-A2", "A2", "B2")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
           aes(x = spec_name, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.1,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A2-0", "B2-0", "A2-B2", "B2-A2", "A2", "B2")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                aes(x = spec_name, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  facet_wrap(~Treatment, scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_species_combo2.pdf", width = 20, height = 20)

ggplot() +
  geom_bar(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A3-0", "B3-0", "A3-B3", "B3-A3", "A3", "B3")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
           aes(x = spec_name, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.1,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A3-0", "B3-0", "A3-B3", "B3-A3", "A3", "B3")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                aes(x = spec_name, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  facet_wrap(~Treatment, scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_species_combo3.pdf", width = 20, height = 20)


ggplot() +
  geom_bar(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A4-0", "B4-0", "A4-B4", "B4-A4", "A4", "B4")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
           aes(x = spec_name, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.1,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A4-0", "B4-0", "A4-B4", "B4-A4", "A4", "B4")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                aes(x = spec_name, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  facet_wrap(~Treatment, scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_species_combo4.pdf", width = 20, height = 20)


ggplot() +
  geom_bar(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A5-0", "B5-0", "A5-B5", "B5-A5", "A5", "B5")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
           aes(x = spec_name, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.1,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A5-0", "B5-0", "A5-B5", "B5-A5", "A5", "B5")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                aes(x = spec_name, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  facet_wrap(~Treatment, scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_species_combo5.pdf", width = 20, height = 20)


ggplot() +
  geom_bar(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A6-0", "B6-0", "A6-B6", "B6-A6", "A6", "B6")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
           aes(x = spec_name, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.1,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A6-0", "B6-0", "A6-B6", "B6-A6", "A6", "B6")) %>%
                        # filter(Experiment == "1") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                aes(x = spec_name, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  facet_wrap(~Treatment, scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_species_combo6.pdf", width = 20, height = 20)



ggplot() +
  geom_bar(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am1-B1", "Bm1-A1")) %>%
                        filter(Experiment == "3") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
           aes(x = spec_name, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.1,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am1-B1", "Bm1-A1")) %>%
                        filter(Experiment == "3") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                aes(x = spec_name, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  facet_wrap(~Treatment, scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_species_combom1.pdf", width = 20, height = 20)



ggplot() +
  geom_bar(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am4-B1", "Bm4-A1")) %>%
                        filter(Experiment == "3") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
           aes(x = spec_name, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.1,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am4-B1", "Bm4-A1")) %>%
                        filter(Experiment == "3") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                aes(x = spec_name, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  facet_wrap(~Treatment, scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_species_combom4.pdf", width = 20, height = 20)



ggplot() +
  geom_bar(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am6-B1", "Bm6-A1")) %>%
                        filter(Experiment == "3") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
           aes(x = spec_name, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.1,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am6-B1", "Bm6-A1")) %>%
                        filter(Experiment == "3") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                aes(x = spec_name, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  facet_wrap(~Treatment, scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_species_combom6.pdf", width = 20, height = 20)



ggplot() +
  geom_bar(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am7-B1", "Bm7-A1")) %>%
                        filter(Experiment == "3") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
           aes(x = spec_name, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.1,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am7-B1", "Bm7-A1")) %>%
                        filter(Experiment == "3") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                aes(x = spec_name, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  facet_wrap(~Treatment, scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_species_combom7.pdf", width = 20, height = 20)



ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                        mutate(spec_name = gsub("Bifidobacterium", "B.", spec_name)) %>%
                        mutate(spec_name = gsub("Lactobacillus", "L.", spec_name)) %>%
                        mutate(spec_name = gsub("Bombilactobacillus", "B.", spec_name)) %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        # filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am7-B1", "Bm7-A1")) %>%
                        filter(Experiment == "pilot") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
           aes(x = spec_name, y = sample_geq),
           fill = "#b9b9b9",
           alpha = 0.1,
           position = "dodge",
           stat = "identity") +
  geom_beeswarm(data = df_species_count_all_combined %>%
                        mutate(spec_name = gsub("Bifidobacterium", "B.", spec_name)) %>%
                        mutate(spec_name = gsub("Lactobacillus", "L.", spec_name)) %>%
                        mutate(spec_name = gsub("Bombilactobacillus", "B.", spec_name)) %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                        # filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am7-B1", "Bm7-A1")) %>%
                        filter(Experiment == "pilot") %>%
                        mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                aes(x = spec_name, y = abs_counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
  labs(y = "Total bacterial abundance", x = "Species", fill = "Strain") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  geom_hline(yintercept = 1e+4, linetype = "dashed") +
  geom_hline(yintercept = 1e+5, linetype = "dashed") +
  geom_hline(yintercept = 1e+6, linetype = "dashed") +
  geom_hline(yintercept = 1e+7, linetype = "dotted") +
  geom_hline(yintercept = 1e+8, linetype = "dotted") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  facet_wrap(~factor(Treatment, treatment_order_pilot), scale = "free_x") +
  make_theme(setFill = F, palettefill = "Set1",
             setCol = F, x_angle = 45, x_size = 10,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/colonization_success_species_combo_pilot.pdf", width = 20, height = 20)

```


```{r}
# for each strain check how well colonization works across experiments in only
# and second...

combos_full <- c("1", "2", "3", "4", "5", "6", "1\'")

combos_drop <- c("1\'", "m1", "m4", "m6", "m7")

# shape for experiment, outline color for limit passed, abs_counts for y, fill for strain
# box plot also for abs count x axis is treatment
p_list <- list()
for (a_strain in strains_uniq) {
  strain_spec_name <- df_species_count_all %>%
                        select(Strain, spec_name_uniq) %>%
                        unique() %>%
                        filter(Strain == a_strain) %>%
                        pull(spec_name_uniq)
  p <- ggplot() +
        geom_bar(data = df_species_count_all %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_full)) %>%
                              # mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = CommunityCombo, y = sample_geq),
                      fill = "#b9b9b9",
                      alpha = 0.1,
                      position = "dodge",
                      stat = "identity"
                      ) +
        geom_boxplot(data = df_species_count_all %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_full)) %>%
                              # mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = CommunityCombo, y = abs_counts, fill = Treatment_simp),
                      color = "#000000",
                      width = 0.5,
                      outlier.shape = NA
                      ) +
        geom_point(data = df_species_count_all %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_full)) %>%
                              # mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                      aes(x = CommunityCombo, y = abs_counts, fill = Treatment_simp, color = geq_passed, alpha = geq_passed),
                      color = "#000000", size = 2
                      ) +
        scale_y_log10(
              breaks = scales::trans_breaks("log10", function(x) 10^x),
              labels = scales::trans_format("log10", scales::math_format(10^.x)),
              # axis limits
              limits = c(1, 1e+10)
          ) +
        labs(y = "Total bacterial abundance (log10)", x = "Treatment", fill = "Treatment") +
        facet_wrap(~Treatment_simp, scale = "free_x", ncol = 4) +
        ggtitle(paste0("Strain: ", strain_spec_name)) +
        scale_fill_manual(values = TreatmentColors) +
        geom_hline(yintercept = 1e+4, linetype = "dashed") +
        geom_hline(yintercept = 1e+5, linetype = "dashed") +
        geom_hline(yintercept = 1e+6, linetype = "dashed") +
        geom_hline(yintercept = 1e+7, linetype = "dotted") +
        geom_hline(yintercept = 1e+8, linetype = "dotted") +
        geom_hline(yintercept = 1e+10, linetype = "dotted") +
        scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
        scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
        make_theme(setFill = F, palettefill = "Set1",
                  setCol = F, x_angle = 45,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1)
  p_list[[a_strain]] <- p
}

marrangeGrob(p_list, ncol = 2, nrow = 3)

pdf("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_full.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 2, nrow = 3)
dev.off()



p_list <- list()
for (a_strain in strains_uniq) {
  strain_spec_name <- df_species_count_all %>%
                        select(Strain, spec_name_uniq) %>%
                        unique() %>%
                        filter(Strain == a_strain) %>%
                        pull(spec_name_uniq)
  p <- ggplot() +
        geom_bar(data = df_species_count_all %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_drop)) %>%
                              # mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = CommunityCombo, y = sample_geq),
                      fill = "#b9b9b9",
                      alpha = 0.1,
                      position = "dodge",
                      stat = "identity"
                      ) +
        geom_boxplot(data = df_species_count_all %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_drop)) %>%
                              # mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = CommunityCombo, y = abs_counts, fill = Treatment_simp),
                      color = "#000000",
                      width = 0.5,
                      outlier.shape = NA
                      ) +
        geom_point(data = df_species_count_all %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_drop)) %>%
                              # mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)),
                      aes(x = CommunityCombo, y = abs_counts, fill = Treatment_simp, color = geq_passed, alpha = geq_passed),
                      color = "#000000", size = 2
                      ) +
        scale_y_log10(
              breaks = scales::trans_breaks("log10", function(x) 10^x),
              labels = scales::trans_format("log10", scales::math_format(10^.x)),
              # axis limits
              limits = c(1, 1e+10)
          ) +
        labs(y = "Total bacterial abundance (log10)", x = "Treatment", fill = "Treatment") +
        facet_wrap(~Treatment_simp, scale = "free_x", ncol = 6) +
        ggtitle(paste0("Strain: ", strain_spec_name)) +
        scale_fill_manual(values = TreatmentColors) +
        geom_hline(yintercept = 1e+4, linetype = "dashed") +
        geom_hline(yintercept = 1e+5, linetype = "dashed") +
        geom_hline(yintercept = 1e+6, linetype = "dashed") +
        geom_hline(yintercept = 1e+7, linetype = "dotted") +
        geom_hline(yintercept = 1e+8, linetype = "dotted") +
        geom_hline(yintercept = 1e+10, linetype = "dotted") +
        scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
        scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
        make_theme(setFill = F, palettefill = "Set1",
                  setCol = F, x_angle = 45,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1)
  p_list[[a_strain]] <- p
}

marrangeGrob(p_list, ncol = 2, nrow = 3)

pdf("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_drop.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 2, nrow = 3)
dev.off()

```


```{r}
# use the same data as above but now for beeswarm instead of box  point

p_list <- list()
for (a_strain in strains_uniq) {
  strain_spec_name <- df_species_count_all %>%
                        select(Strain, spec_name_uniq) %>%
                        unique() %>%
                        filter(Strain == a_strain) %>%
                        pull(spec_name_uniq)
  p <- ggplot() +
        geom_bar(data = df_species_count_all %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_full)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = CommunityCombo, y = sample_geq),
                      fill = "#b9b9b9",
                      alpha = 0.1,
                      position = "dodge",
                      stat = "identity"
                      ) +
        geom_beeswarm(data = df_species_count_all %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_full)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = CommunityCombo, y = abs_counts, fill = Treatment_simp, shape = CommunityCombo, color = geq_passed, alpha = geq_passed),
                      size = 2,
                      cex = 4,
                      corral = "wrap",
                      corral.width = 0.75,
                      alpha = 0.95,
                      # shape = 21,
                      stroke = 1,
                      ) +
        scale_y_log10(
            breaks = scales::trans_breaks("log10", function(x) 10^x),
            labels = scales::trans_format("log10", scales::math_format(10^.x))
          ) +
        labs(y = "Total bacterial abundance (log10)", x = "Treatment", fill = "Treatment") +
        facet_wrap(~Treatment_simp, scale = "free_x", ncol = 4) +
        ggtitle(paste0("Strain: ", strain_spec_name)) +
        scale_shape_manual(values = c("1" = 22,
                                      "2" = 21,
                                      "3" = 20,
                                      "4" = 23,
                                      "5" = 24,
                                      "6" = 25,
                                      "1_rep" = 0,
                                      "1\'" = 0,
                                      "m1" = 21,
                                      "m4" = 22,
                                      "m6" = 23,
                                      "m7" = 24
                                    )) +
        scale_fill_manual(values = TreatmentColors) +
        geom_hline(yintercept = 1e+4, linetype = "dashed") +
        geom_hline(yintercept = 1e+5, linetype = "dashed") +
        geom_hline(yintercept = 1e+6, linetype = "dashed") +
        geom_hline(yintercept = 1e+7, linetype = "dotted") +
        geom_hline(yintercept = 1e+8, linetype = "dotted") +
        geom_hline(yintercept = 1e+10, linetype = "dotted") +
        scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
        scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
        make_theme(setFill = F, palettefill = "Set1",
                  setCol = F, x_angle = 45,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1)
  p_list[[a_strain]] <- p
}

# marrangeGrob(p_list, ncol = 2, nrow = 3)

pdf("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_full_beeswarm.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 2, nrow = 3)
dev.off()


p_list <- list()
for (a_strain in strains_uniq) {
  strain_spec_name <- df_species_count_all %>%
                        select(Strain, spec_name_uniq) %>%
                        unique() %>%
                        filter(Strain == a_strain) %>%
                        pull(spec_name_uniq)
  p <- ggplot() +
        geom_bar(data = df_species_count_all %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_full)) %>%
                              mutate(abs_counts_org = ifelse(abs_counts_org == 0, 0.1, abs_counts_org)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = CommunityCombo, y = sample_geq),
                      fill = "#b9b9b9",
                      alpha = 0.1,
                      position = "dodge",
                      stat = "identity"
                      ) +
        geom_beeswarm(data = df_species_count_all %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_full)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = CommunityCombo, y = abs_counts, fill = Treatment_simp, shape = CommunityCombo, color = geq_passed, alpha = geq_passed),
                      size = 2.5,
                      cex = 4,
                      corral = "wrap",
                      corral.width = 0.75,
                      alpha = 0.95,
                      shape = 21,
                      stroke = 1,
                      ) +
        scale_y_log10(
            breaks = scales::trans_breaks("log10", function(x) 10^x),
            labels = scales::trans_format("log10", scales::math_format(10^.x))
          ) +
        labs(y = "Total bacterial abundance (log10)", x = "Treatment", fill = "Treatment") +
        # facet_wrap(~Treatment_simp, scale = "free_x", ncol = 4) +
        ggtitle(paste0("Strain: ", strain_spec_name)) +
        geom_hline(yintercept = 1e+4, linetype = "dashed") +
        geom_hline(yintercept = 1e+5, linetype = "dashed") +
        geom_hline(yintercept = 1e+6, linetype = "dashed") +
        geom_hline(yintercept = 1e+7, linetype = "dotted") +
        geom_hline(yintercept = 1e+8, linetype = "dotted") +
        geom_hline(yintercept = 1e+10, linetype = "dotted") +
        scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
        scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
        scale_fill_manual(values = TreatmentColors) +
        make_theme(setFill = F, palettefill = "Set1",
                  setCol = F, x_angle = 45,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1)
  p_list[[a_strain]] <- p
}

# marrangeGrob(p_list, ncol = 2, nrow = 3)

pdf("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_full_combined_beeswarm.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 2, nrow = 3)
dev.off()


p_list <- list()
for (a_strain in strains_uniq) {
  strain_spec_name <- df_species_count_all %>%
                        select(Strain, spec_name_uniq) %>%
                        unique() %>%
                        filter(Strain == a_strain) %>%
                        pull(spec_name_uniq)
  p <- ggplot() +
        geom_bar(data = df_species_count_all %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_drop)) %>%
                              mutate(abs_counts_org = ifelse(abs_counts_org == 0, 0.1, abs_counts_org)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = CommunityCombo, y = sample_geq),
                      fill = "#b9b9b9",
                      alpha = 0.1,
                      position = "dodge",
                      stat = "identity"
                      ) +
        geom_beeswarm(data = df_species_count_all %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_drop)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = CommunityCombo, y = abs_counts, fill = Treatment_simp, color = geq_passed, alpha = geq_passed),
                      size = 2.5,
                      cex = 4,
                      corral = "wrap",
                      corral.width = 0.75,
                      alpha = 0.95,
                      shape = 21,
                      stroke = 1,
                      ) +
        scale_y_log10(
            breaks = scales::trans_breaks("log10", function(x) 10^x),
            labels = scales::trans_format("log10", scales::math_format(10^.x))
          ) +
        labs(y = "Total bacterial abundance (log10)", x = "Treatment", fill = "Treatment") +
        facet_wrap(~Treatment_simp, scale = "free_x", ncol = 6) +
        ggtitle(paste0("Strain: ", strain_spec_name)) +
        scale_fill_manual(values = TreatmentColors) +
        geom_hline(yintercept = 1e+4, linetype = "dashed") +
        geom_hline(yintercept = 1e+5, linetype = "dashed") +
        geom_hline(yintercept = 1e+6, linetype = "dashed") +
        geom_hline(yintercept = 1e+7, linetype = "dotted") +
        geom_hline(yintercept = 1e+8, linetype = "dotted") +
        geom_hline(yintercept = 1e+10, linetype = "dotted") +
        scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
        scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
        make_theme(setFill = F, palettefill = "Set1",
                  setCol = F, x_angle = 45,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1)
  p_list[[a_strain]] <- p
}

# marrangeGrob(p_list, ncol = 2, nrow = 3)

pdf("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_drop_beeswarm.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 2, nrow = 3)
dev.off()

```


```{r}
# freq distribution histogram of each strain arriving first and second 

df_species_count_all_arr <- df_species_count_all %>%
                              mutate(arrival = Vectorize(get_first_second_only_arriver)(Treatment, Strain))

p_list <- list()
for (a_strain in strains_uniq) {
  strain_spec_name <- df_species_count_all_arr %>%
                        select(Strain, spec_name_uniq) %>%
                        unique() %>%
                        filter(Strain == a_strain) %>%
                        pull(spec_name_uniq)
  p <- ggplot() +
        geom_density(data = df_species_count_all_arr %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_full)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = abs_counts, fill = arrival),
                      # color = "#000000",
                      alpha = 0.5,
                      # position = "dodge",
                      # size = 2.5,
                      # cex = 4,
                      # corral = "wrap",
                      # corral.width = 0.75,
                      ) +
        # facet_wrap(~CommunityCombo, scale = "free_x") +
        ggtitle(paste0("Strain: ", strain_spec_name)) +
        # scale_fill_manual(values = TreatmentColors) +
        scale_x_log10(
          breaks = scales::trans_breaks("log10", function(x) 10^x),
          labels = scales::trans_format("log10", scales::math_format(10^.x))
        ) +
        make_theme(setFill = T, palettefill = "Spectral",
                  setCol = F, x_angle = 45,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1)

  p_list[[a_strain]] <- p
}

marrangeGrob(p_list, ncol = 2, nrow = 3)

pdf("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_full_arrival_density.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 2, nrow = 3)
dev.off()


p_list <- list()
for (a_strain in strains_uniq) {
  strain_spec_name <- df_species_count_all_arr %>%
                        select(Strain, spec_name_uniq) %>%
                        unique() %>%
                        filter(Strain == a_strain) %>%
                        pull(spec_name_uniq)
  p <- ggplot() +
        geom_density(data = df_species_count_all_arr %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_full)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = abs_counts, fill = arrival),
                      # color = "#000000",
                      alpha = 0.5,
                      position = "stack",
                      # size = 2.5,
                      # cex = 4,
                      # corral = "wrap",
                      # corral.width = 0.75,
                      ) +
        facet_wrap(~CommunityCombo, scale = "free_x") +
        ggtitle(paste0("Strain: ", strain_spec_name)) +
        # scale_fill_manual(values = TreatmentColors) +
        scale_x_log10(
          breaks = scales::trans_breaks("log10", function(x) 10^x),
          labels = scales::trans_format("log10", scales::math_format(10^.x))
        ) +
        make_theme(setFill = T, palettefill = "Spectral",
                  setCol = F, x_angle = 45,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1)

  p_list[[a_strain]] <- p
}

marrangeGrob(p_list, ncol = 2, nrow = 3)

pdf("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_full_arrival_density_by_combo.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 2, nrow = 3)
dev.off()



p_list <- list()
for (a_strain in strains_uniq) {
  strain_spec_name <- df_species_count_all_arr %>%
                        select(Strain, spec_name_uniq) %>%
                        unique() %>%
                        filter(Strain == a_strain) %>%
                        pull(spec_name_uniq)
  p <- ggplot() +
        geom_density(data = df_species_count_all_arr %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_drop)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = abs_counts, fill = arrival),
                      # color = "#000000",
                      alpha = 0.5,
                      # position = "dodge",
                      # size = 2.5,
                      # cex = 4,
                      # corral = "wrap",
                      # corral.width = 0.75,
                      ) +
        # facet_wrap(~CommunityCombo, scale = "free_x") +
        ggtitle(paste0("Strain: ", strain_spec_name)) +
        # scale_fill_manual(values = TreatmentColors) +
        scale_x_log10(
          breaks = scales::trans_breaks("log10", function(x) 10^x),
          labels = scales::trans_format("log10", scales::math_format(10^.x))
        ) +
        make_theme(setFill = T, palettefill = "Spectral",
                  setCol = F, x_angle = 45,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1)

  p_list[[a_strain]] <- p
}

marrangeGrob(p_list, ncol = 2, nrow = 3)

pdf("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_drop_arrival_density.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 2, nrow = 3)
dev.off()


p_list <- list()
for (a_strain in strains_uniq) {
  strain_spec_name <- df_species_count_all_arr %>%
                        select(Strain, spec_name_uniq) %>%
                        unique() %>%
                        filter(Strain == a_strain) %>%
                        pull(spec_name_uniq)
  p <- ggplot() +
        geom_density(data = df_species_count_all_arr %>%
                              filter(Strain == a_strain) %>%
                              filter(Type == "bee") %>%
                              mutate(CommunityCombo = ifelse(Experiment == "3" & CommunityCombo == "1", "1\'", CommunityCombo)) %>%
                              filter(CommunityCombo %in% c("1\'", combos_drop)) %>%
                              mutate(Treatment_simp = ifelse(Type == "bee", Vectorize(simplify_treatment)(Treatment), Treatment)) %>%
                              # group_by(ID, spec_name) %>%
                              # summarise(abs_counts_total = sum(abs_counts), ID, Treatment, Experiment, Treatment_simp) %>%
                              unique(),
                      aes(x = abs_counts, fill = arrival),
                      # color = "#000000",
                      alpha = 0.5,
                      position = "stack",
                      # size = 2.5,
                      # cex = 4,
                      # corral = "wrap",
                      # corral.width = 0.75,
                      ) +
        facet_wrap(~CommunityCombo, scale = "free_x") +
        ggtitle(paste0("Strain: ", strain_spec_name)) +
        # scale_fill_manual(values = TreatmentColors) +
        scale_x_log10(
          breaks = scales::trans_breaks("log10", function(x) 10^x),
          labels = scales::trans_format("log10", scales::math_format(10^.x))
        ) +
        make_theme(setFill = T, palettefill = "Spectral",
                  setCol = F, x_angle = 45,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1)

  p_list[[a_strain]] <- p
}

marrangeGrob(p_list, ncol = 2, nrow = 3)

pdf("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/colonization_success_drop_arrival_density_by_combo.pdf", width = 20, height = 20)
marrangeGrob(p_list, ncol = 2, nrow = 3)
dev.off()


```



```{r}
Count_limit <- 10
Rel_Count_limit <- 0.000001
Total_Count_limit <- 100

# # # # # # # # # # # # # # #
post hoc analyses to decide on whether it was ok with the thresholds or lack thereof
# # # # # # # # # # # # # # #
df_species_count_all # needs to be made to return here
system("mkdir 03_Analysis/Results/figures/priority_effects_experiment/threshold_selection")

ggplot() +
  geom_point(data = df_species_count_all,
                aes(x = Ct_Mean_bac, y = total_counts, color = Type)) +
  geom_vline(xintercept = LOD_ct, linetype = "dashed") +
  geom_hline(yintercept = Total_Count_limit, linetype = "dashed") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(palettecolor = "Spectral")
ggsave("03_Analysis/Results/figures/priority_effects_experiment/threshold_selection/total_counts_vs_ct.pdf")

df_species_count_all %>%
  filter(limit_passed == "No") %>%
  # select(total_counts, Ct_Mean_bac, ID, Type, Treatment) %>%
  unique %>% view

ggplot() +
  geom_point(data = df_species_count_all,
                aes(x = Ct_Mean_bac, y = Counts, color = Strain,
                shape = Type),
                size = 2
                ) +
  geom_vline(xintercept = LOD_ct, linetype = "dashed") +
  geom_hline(yintercept = Count_limit, linetype = "dashed") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_color_manual(values = strainColorsUniq) +
  make_theme(setCol = F, leg_pos = "right",
             guide_nrow = 22
  )
ggsave("03_Analysis/Results/figures/priority_effects_experiment/threshold_selection/counts_vs_ct.pdf")

ggplot() +
  geom_point(data = df_species_count_all,
                aes(x = Ct_Mean_bac, y = rel_counts, color = Strain,
                shape = Type),
                size = 2
                ) +
  geom_vline(xintercept = LOD_ct, linetype = "dashed") +
  geom_hline(yintercept = Rel_Count_limit, linetype = "dashed") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_color_manual(values = strainColorsUniq) +
  make_theme(setCol = F, leg_pos = "right",
             guide_nrow = 22
  )
ggsave("03_Analysis/Results/figures/priority_effects_experiment/threshold_selection/rel_counts_vs_ct.pdf")

# show cases where A1-B1 is missing a strain because A1-0 was also missing it
# within each experiment
df_species_count_all %>%
  mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment)) %>%
  filter(!is.na(Experiment)) %>%
  filter(Experiment != "none") %>%
  select(Strain, spec_name, Experiment, Type, CommunityCombo, Treatment_simp, limit_passed) %>%
  group_by(Strain, spec_name, Experiment, Type, CommunityCombo, Treatment_simp) %>%
  # summarize as number of Yes in limit passed
  reframe(limit_passed = paste0(sum(limit_passed == "Yes"), "/", length(limit_passed))) %>%
  filter(Type == "bee") %>%
  pivot_wider(names_from = Treatment_simp, values_from = limit_passed) %>%
  arrange(spec_name, Strain, CommunityCombo) %>%
  write.csv("03_Analysis/Results/figures/priority_effects_experiment/threshold_selection/bee_counts_across_exps.csv")
  filter(grepl("^0/", A)) %>% view
  # filter(grepl("0/", A) | grepl("0/", AB) | grepl("0/", B) | grepl("0/", BA)) %>% view

df_species_count_all %>%
  mutate(Treatment_simp = Vectorize(simplify_treatment)(Treatment)) %>%
  filter(!is.na(Experiment)) %>%
  filter(Experiment != "none") %>%
  select(Strain, spec_name, Experiment, Type, CommunityCombo, Treatment_simp, limit_passed) %>%
  group_by(Strain, spec_name, Experiment, Type, CommunityCombo, Treatment_simp) %>%
  # summarize as number of Yes in limit passed
  reframe(limit_passed = paste0(sum(limit_passed == "Yes"), "/", length(limit_passed))) %>%
  filter(Type != "bee") %>%
  pivot_wider(names_from = Treatment_simp, values_from = limit_passed) %>%
  filter(A1 == "0/2" & B1 == "0/2") %>%
  arrange(spec_name, Strain, CommunityCombo) %>% view

# # # # # # # # # # # # # # #
```

Total load of bacteria vs number of strains detected

```{r}

df_count_vs_load <- df_species_count_all %>%
                      filter(Type == "bee") %>%
                      filter(geq_passed == "1") %>%
                      filter(!ID %in% samples_low_detection) %>%
                      group_by(ID) %>%
                      summarise(Treatment_simp, CommunityCombo, total_load = bac_copies, n_strains = n_distinct(Strain))

ggplot() +
  geom_point(data = df_count_vs_load,
                # aes(x = n_strains, y = total_load),
                aes(x = n_strains, y = total_load, color = Treatment_simp),
    size = 2
  ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  # facet_wrap(~Treatment_simp) +
  stat_cor(data = df_count_vs_load,
                aes(x = n_strains, y = total_load),
            p.accuracy = 0.001, r.accuracy = 0.01, method = "spearman") +
  geom_smooth(data = df_count_vs_load,
                aes(x = n_strains, y = total_load),
                method = "lm", se = T) +
  stat_cor(data = df_count_vs_load %>% filter(n_strains >= 10),
                aes(x = n_strains, y = total_load),
            label.x.npc = "right", label.y.npc = "top",
            label.x = 15,
            p.accuracy = 0.001, r.accuracy = 0.01, method = "spearman") +
  geom_smooth(data = df_count_vs_load %>% filter(n_strains >= 10),
                aes(x = n_strains, y = total_load),
                method = "lm", se = T) +
  labs(y = "Total bacterial load", x = "Number of strains detected", color = "Treatment") +
  geom_vline(xintercept = 10, linetype = "dashed") +
  geom_vline(xintercept = 22, linetype = "solid") +
  scale_color_manual(values = TreatmentColors) +
  make_theme(setFill = F, setCol = F, x_angle = 45,
             leg_pos = "right", guide_nrow = 6,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/total_load_vs_n_strains.pdf")


df_load_per_species <- df_species_count_all %>%
                      filter(Experiment != "3") %>%
                      mutate(Treatment_simp = ifelse(Treatment_simp == "extraction_blank", "blank", Treatment_simp)) %>%
                      filter(spec_name != "unknown") %>%
                      filter(!is.na(spec_name)) %>%
                      filter(Type == "bee") %>%
                      filter(geq_passed == "1") %>%
                      filter(!ID %in% samples_low_detection) %>%
                      group_by(ID, spec_name) %>%
                      summarise(Treatment_simp, CommunityCombo, species_load = sum(abs_counts))

pvals <- df_load_per_species %>%
            group_by(spec_name) %>%
            wilcox_test(species_load ~ Treatment_simp, p.adjust.method = "BH",
                        comparisons = list(c("A", "AB"),
                                           c("B", "BA")
                                      )
            )
            

ggplot() +
  geom_boxplot(data = df_load_per_species,
                aes(x = Treatment_simp, y = species_load, fill = Treatment_simp),
                # aes(x = spec_name, y = species_load, color = Treatment_simp),
    outlier.shape = NA
  ) +
  geom_point(data = df_load_per_species,
                aes(x = Treatment_simp, y = species_load),
    size = 1,
    color = "#000000"
  ) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  # # add wilcox on top of the boxplots and add BH correction of p values
  # geom_signif(data = df_load_per_species,
  #               aes(x = Treatment_simp, y = species_load),
  #               test = "wilcox.test", test.args = list(exact = F, p.adjust.method = "BH"),
  #               comparisons = list(c("A", "AB"),
  #                                  c("B", "BA")
  #                             ),
  #               map_signif_level = T,
  #               facet_variable = "spec_name") +
  geom_signif(data = pvals, aes(xmin = group1, xmax = group2, annotations = p.adj.signif, y_position = y.position),
              manual = TRUE,
              inherit.aes = FALSE, tip_length = 0) +
  facet_wrap(~spec_name, scale = "free_x", ncol = 3) +
  labs(y = "Total bacterial load (sum of strain of species)",
       x = "Treatment", fill = "Treatment") +
  scale_color_manual(values = TreatmentColors) +
  scale_fill_manual(values = TreatmentColors) +
  make_theme(setFill = F, setCol = F, x_angle = 45,
             leg_pos = "right", guide_nrow = 6,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/species_load_vs_treatment.pdf")

```

```{r}
```


# Explore orthogroups!

```{r}

bifido_gene_counts_mat <- read.csv("03_Analysis/IsolateGenomes/Compare_genomes/OrthogroupsAndFunctions/Bifido_Orthogroups.GeneCount.tsv", sep = "\t") %>%
                          select(any_of(c("Orthogroup", strains_all)))

# rename columns using strain_name_spec_dict_uniq
colnames(bifido_gene_counts_mat) <- unlist(lapply(colnames(bifido_gene_counts_mat), function(x) {
  if (x %in% strains_all) {
    strain_name_spec_dict_uniq[[x]]
  } else {
    x
  }
}))


bifido_gene_counts_mat <- bifido_gene_counts_mat %>%
                          # remove those rows with all columns zero
                          filter(rowSums(bifido_gene_counts_mat[, -1] > 0) > 0)

bifido_gene_counts_mat_shared <- bifido_gene_counts_mat %>%
                                  # get rows present >0 all columns
                                  filter(rowSums(bifido_gene_counts_mat[, -1] > 0) >= 6)

bifido_gene_counts_mat_unique <- bifido_gene_counts_mat %>%
                                  # get rows present >0 all columns
                                  filter(rowSums(bifido_gene_counts_mat[, -1] > 0) == 1)

bifido_gene_counts_mat_shared_by_pairs <- bifido_gene_counts_mat %>%
                                          filter(rowSums(bifido_gene_counts_mat[, -1] > 0) == 2)


lacto_gene_counts_mat <- read.csv("03_Analysis/IsolateGenomes/Compare_genomes/OrthogroupsAndFunctions/Lacto_Orthogroups.GeneCount.tsv", sep = "\t") %>%
                          select(any_of(c("Orthogroup", strains_all))) 

colnames(lacto_gene_counts_mat) <- unlist(lapply(colnames(lacto_gene_counts_mat), function(x) {
  if (x %in% strains_all) {
    strain_name_spec_dict_uniq[[x]]
  } else {
    x
  }
}))

lacto_gene_counts_mat <- lacto_gene_counts_mat %>%
                          # remove those rows with all columns zero
                          filter(rowSums(lacto_gene_counts_mat[, -1] > 0) > 0)

lacto_gene_counts_mat_shared <- lacto_gene_counts_mat %>%
                                  # get rows present >0 all columns
                                  filter(rowSums(lacto_gene_counts_mat[, -1] > 0) >= 6)

lacto_gene_counts_mat_unique <- lacto_gene_counts_mat %>%
                                  # get rows present >0 all columns
                                  filter(rowSums(lacto_gene_counts_mat[, -1] > 0) == 1)

lacto_gene_counts_mat_shared_by_pairs <- lacto_gene_counts_mat %>%
                                          filter(rowSums(lacto_gene_counts_mat[, -1] > 0) == 2)

lacto_gene_counts_mat_shared_by_pairs %>%
  arrange(`Lactobacillus apis-ESL0263`, `Lactobacillus apis-ESL0185`) %>% view

```

Can we detect any of these ASVs in natural samples!?

```{r}

# readRDS("/nas/FAC/FBM/DMF/pengel/spirit/D2c/aprasad/20211018_aprasad_ApisCrossSpeciesAnalysis/Analysis/20240516-AshaVisit/PacBio2023/03_Dada2Results/ASV_counts.csv")

system("mkdir -p 03_Analysis/FindNatural_16S")
dada2_natural_bees <- readRDS("/nas/FAC/FBM/DMF/pengel/spirit/D2c/aprasad/20211018_aprasad_ApisCrossSpeciesAnalysis/Analysis/20240516-AshaVisit/PacBio2023/RDS/dada2.rds")
seqtable_natural_bees <- makeSequenceTable(dada2_natural_bees); dim(seqtable_natural_bees)
seqtable_nochim_natural_bees <- removeBimeraDenovo(seqtable_natural_bees, method = "consensus", multithread = TRUE, verbose = TRUE); dim(seqtable_nochim_natural_bees)
write.csv(seqtable_nochim_natural_bees, "03_Analysis/FindNatural_16S/seqtable_nochim.csv")

# system("source ~/.bashrc && conda activate 20230313_scripts_env && 03_Analysis/infer_species_counts.py -s 03_Analysis/FindNatural_16S/seqtable_nochim.csv -o 03_Analysis/FindNatural_16S/species_table.csv")
system("source ~/.bashrc && conda activate 20230313_scripts_env && 03_Analysis/infer_species_counts.py -m -s 03_Analysis/FindNatural_16S/seqtable_nochim.csv -o 03_Analysis/FindNatural_16S/species_table_updated.csv")

qpcr_df_natural <- read.csv("/users/aprasad/work_aprasad/BACKUP_current/20230313_apis_species_comparison/results/figures/qpcr_plot_df.csv")

df_species_count_natural <- read.csv("03_Analysis/FindNatural_16S/species_table_updated.csv", row.name = 1) %>%
                              rownames_to_column("ID") %>%
                              mutate(ID = Vectorize(remove_extension)(ID, "_filt_reads.fastq.gz")) %>%
                              pivot_longer(!ID, names_to = "Strain", values_to = "Counts") %>%
                              group_by(ID) %>%
                              mutate(total_counts = sum(Counts)) %>%
                              mutate(rel_counts = Counts/total_counts) %>%
                              # select(-total_counts) %>%
                              ungroup() %>%
                              left_join(qpcr_df_natural %>%
                                          filter(Host == "Apis mellifera") %>%
                                          mutate(ID = Sample.Name, bac_copies = copy_num, Ct_Mean_bac = Ct_mean) %>%
                                          select(c(ID, bac_copies, Ct_Mean_bac)) %>%
                                            unique(), by = c("ID")) %>%                              
                              mutate(abs_counts_org = bac_copies * rel_counts) %>%
                              mutate(limit_passed = ifelse(Ct_Mean_bac > LOD_ct | is.na(Ct_Mean_bac), "No", "Yes")) %>%
                              # mutate(limit_passed = ifelse(Ct_Mean_bac > LOD_ct | is.na(Ct_Mean_bac) | total_counts < Total_Count_limit | Counts < Count_limit , "No", "Yes")) %>%
                              mutate(limit_passed = as.factor(limit_passed)) %>%
                              # calculate sample threshold and adjust zeros
                              group_by(ID) %>%
                              left_join(df_info %>% select(strain, spec_name), by = c("Strain" = "strain")) %>%
                              mutate(spec_name_uniq = paste0(spec_name, " (", Strain, ")")) %>%
                              mutate(total_reads = sum(Counts, na.rm = T)) %>%
                              mutate(sample_geq = bac_copies/total_reads) %>%
                              ungroup() %>%
                              mutate(abs_counts = ifelse(is.na(abs_counts_org), 0, abs_counts_org)) %>%
                              mutate(geq_passed = ifelse(abs_counts >= sample_geq, 1, 0)) %>%
                              mutate(geq_passed = as.factor(geq_passed)) %>%
                              mutate(abs_counts = ifelse(abs_counts == 0, 1, abs_counts))

ggplot() +
  geom_beeswarm(data = df_species_count_natural,
                aes(x = Strain, y = abs_counts, fill = Strain, color = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
        scale_y_log10(
              breaks = scales::trans_breaks("log10", function(x) 10^x),
              labels = scales::trans_format("log10", scales::math_format(10^.x)),
              # axis limits
              limits = c(1, 1e+10)
          ) +
        geom_path(data = df_species_count_natural,
                aes(x = Strain, y = abs_counts, group = ID, color = geq_passed),
                color = "#000000",
                alpha = 0.5,
                size = 0.5) +
        facet_wrap(~spec_name, scale = "free_x") +
        labs(y = "Total bacterial abundance (log10)", x = "Treatment", fill = "Treatment") +
        scale_fill_manual(values = strainColorsUniq) +
        scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
        scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
        make_theme(setFill = F, palettefill = "Set1",
                  setCol = F, x_angle = 45,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/FindNatural_16S/species_counts_natural.pdf", width = 20, height = 20)

# system("source ~/.bashrc && conda activate 20230313_scripts_env && 03_Analysis/infer_species_counts.py -s 03_Analysis/IsolateAmplicons/seqtable_nochim.csv -o 03_Analysis/IsolateAmplicons/species_table.csv")

phylo_obj_natural <- readRDS("/nas/FAC/FBM/DMF/pengel/spirit/D2c/aprasad/20211018_aprasad_ApisCrossSpeciesAnalysis/Analysis/20240516-AshaVisit/PacBio2023/04_PhyloseqObject/PhyloSeq_Object.rds")

df_natural_counts_all <- psmelt(phylo_obj_natural) %>%
                      filter(grepl("mellifera", Host)) %>% # master filter
                      filter(!is.na(OTU)) %>%
                      filter(!is.na(Genus)) %>%
                      left_join(qpcr_df_natural %>%
                                # filter(Host == "Apis mellifera") %>%
                                mutate(ID = Sample.Name, bac_copies = copy_num, Ct_Mean_bac = Ct_mean) %>%
                                select(c(ID, bac_copies, Ct_Mean_bac)) %>%
                                  unique, by = c("Sample" = "ID")) %>%
                      mutate(abs_counts_org = bac_copies * Abundance) %>%
                      mutate(limit_passed = ifelse(Ct_Mean_bac > LOD_ct | is.na(Ct_Mean_bac), "No", "Yes")) %>%
                      # mutate(limit_passed = ifelse(Ct_Mean_bac > LOD_ct | is.na(Ct_Mean_bac) | total_counts < Total_Count_limit | Counts < Count_limit , "No", "Yes")) %>%
                      mutate(limit_passed = as.factor(limit_passed)) %>%
                      mutate(abs_counts = ifelse(limit_passed == "Yes", abs_counts_org, NA)) %>%
                      mutate(present_100 = ifelse(Abundance > 100, 1, 0)) %>%
                      mutate(present_100 = as.factor(present_100)) %>%
                      mutate(present = ifelse(Abundance > 0, 1, 0)) %>%
                      mutate(present = as.factor(present)) %>%
                      # calculate sample threshold and adjust zeros
                      group_by(Sample) %>%
                      left_join(df_info %>% select(strain, spec_name), by = c("OTU" = "strain")) %>%
                      mutate(spec_name_uniq = paste0(spec_name, " (", OTU, ")")) %>%
                      mutate(geq_passed = ifelse(abs_counts >= 1e+4, "1", "0")) %>%
                      mutate(geq_passed = as.factor(geq_passed)) %>%
                      mutate(total_reads = sum(Abundance, na.rm = T)) %>%
                      mutate(sample_geq = bac_copies/total_reads) %>%
                      ungroup() %>%
                      # sample geq is the number of copies that corresponds to one read for that sample
                      # any strain having an abundance less than this is considered not detected
                      # so we can set the abundance of these strains to this value
                      # in plots, we can show the strains that are not detected as the threshold value
                      mutate(abs_counts = ifelse(abs_counts_org == 0, 1, abs_counts_org)) %>%
                      mutate(geq_passed = ifelse(abs_counts_org > sample_geq, 1, 0)) %>%
                      mutate(geq_passed = as.factor(geq_passed)) %>%
                      mutate(Species = paste0(Genus, " ", Species)) %>%
                      group_by(Genus) %>%
                          mutate(sum_check = sum(Abundance, na.rm = T)) %>%
                          ungroup() %>%
                          filter(sum_check > 0)


df_natural_counts_filt <- df_natural_counts_all %>%
                          group_by(Genus) %>%
                          mutate(sum_check = sum(Abundance, na.rm = T)) %>%
                          ungroup() %>%
                          filter(sum_check > 0) %>%
                          filter(Genus %in% c("Lactobacillus", "Bifidobacterium", "Bombilactobacillus"))


ggplot(data = df_natural_counts_all %>%
                        filter(grepl("mellifera", Host)) %>%
                        # remove asvs only found in 1 sample
                        group_by(OTU) %>%
                        mutate(present_check = ifelse(Abundance > 0, 1, 0)) %>%
                        mutate(num_check = sum(present_check)) %>%
                        filter(num_check >= 2) %>%
                        ungroup() %>%
                        group_by(Species) %>%
                          mutate(sum_check = sum(Abundance, na.rm = T)) %>%
                          ungroup() %>%
                          filter(sum_check > 0) %>%
                        filter(!is.na(Genus)) %>% filter(!is.na(Species)),
                aes(x = OTU, y = abs_counts, color = geq_passed, group = Sample)) +
  geom_beeswarm(fill = "#df5bff",
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
        scale_y_log10(
              breaks = scales::trans_breaks("log10", function(x) 10^x),
              labels = scales::trans_format("log10", scales::math_format(10^.x))
              # # axis limits
              # limits = c(1, 1e+10)
          ) +
        geom_line() +
        facet_wrap(~Species, scale = "free_x", ncol = 3) +
        labs(y = "Total bacterial abundance (log10)", x = "Treatment", fill = "Treatment") +
        # scale_fill_manual(values = strainColorsUniq) +
        scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
        scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
        geom_hline(yintercept = 1e+4, linetype = "dashed") +
        make_theme(setFill = T, palettefill = "Set1",
                  setCol = F, x_angle = 45,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/FindNatural_16S/species_counts_natural_all.pdf", width = 20, height = 20)
# make a correlation heatmap plot between ASVs based on "Abundance"




# group by otus! - phylotypes.. 97%
dna <- refseq(phylo_obj_natural)
names(dna) <- names(asv_seqs)
# write this out
writeXStringSet(dna, "03_Analysis/FindNatural_16S/asv_sequences.fasta", format = "fasta")

# cd-hit env
# /work/FAC/FBM/DMF/pengel/spirit/aprasad/snakemake-conda-envs/c41bb15eb344238b9c5a180c961ff099_

system("mkdir -p /scratch/aprasad/cdhit_16S")
system("cp 03_Analysis/FindNatural_16S/asv_sequences.fasta /scratch/aprasad/cdhit_16S")
system("cp 03_Analysis/FindNatural_16S/asv_sequences_extended.fasta /scratch/aprasad/cdhit_16S")

# Sinteractive -m150G -t01:00:00
# cd /scratch/aprasad/cdhit_16S
# conda activate /work/FAC/FBM/DMF/pengel/spirit/aprasad/snakemake-conda-envs/c41bb15eb344238b9c5a180c961ff099_

"""
cd-hit -i asv_sequences.fasta -o asv_90_clusters -c 0.90 -n 5
cd-hit -i asv_sequences.fasta -o asv_95_clusters -c 0.95 -n 5
cd-hit -i asv_sequences.fasta -o asv_97_clusters -c 0.97 -n 5
cd-hit -i asv_sequences.fasta -o asv_99_clusters -c 0.99 -n 5
cd-hit -i asv_sequences.fasta -o asv_995_clusters -c 0.995 -n 5
cd-hit -i asv_sequences.fasta -o asv_998_clusters -c 0.998 -n 5
cd-hit -i asv_sequences.fasta -o asv_999_clusters -c 0.999 -n 5

cd-hit -i asv_sequences_extended.fasta -o asv_ext_90_clusters -c 0.90 -n 5
cd-hit -i asv_sequences_extended.fasta -o asv_ext_95_clusters -c 0.95 -n 5
cd-hit -i asv_sequences_extended.fasta -o asv_ext_97_clusters -c 0.97 -n 5
cd-hit -i asv_sequences_extended.fasta -o asv_ext_99_clusters -c 0.99 -n 5
cd-hit -i asv_sequences_extended.fasta -o asv_ext_995_clusters -c 0.995 -n 5
cd-hit -i asv_sequences_extended.fasta -o asv_ext_998_clusters -c 0.998 -n 5
cd-hit -i asv_sequences_extended.fasta -o asv_ext_999_clusters -c 0.999 -n 5
"""

system("mkdir -p 03_Analysis/FindNatural_16S/cd-hit_output")
system("cp /scratch/aprasad/cdhit_16S/* 03_Analysis/FindNatural_16S/cd-hit_output")

clust_threshold <- "995"


cdhit_file <- paste0("03_Analysis/FindNatural_16S/cd-hit_output/asv_ext_", clust_threshold, "_clusters.clstr")  # Replace with your actual file
output_csv <- paste0("03_Analysis/FindNatural_16S/cd-hit_output/asv_ext_", clust_threshold, "_clusters_parsed.csv")
cluster_info_file <- paste0("03_Analysis/FindNatural_16S/cd-hit_output/asv_ext_mod_", clust_threshold, "_summary.csv")

# Read the file
lines <- readLines(cdhit_file)

# Initialize variables
cluster_data <- data.frame(ASV_ID = character(), Cluster = character(), stringsAsFactors = FALSE)


# line = "14	1447aa, >1253... at 1:1447:26:1472/99.79%"
# line = "15	1523aa, >13-ESL0200... *"
# ">\\d+(-\\w+)*"

# line = "0	1488aa, >ASV93... at 99.93%"
# line = "3	1562aa, >22-ESL0394... *"
# ">\\w+(-\\w+)*"


current_cluster <- ""

# Loop through the lines
for (line in lines) {
  line <- trimws(line)  # Remove leading/trailing spaces
  
  if (grepl("^>Cluster", line)) {
    # Extract the cluster number
    current_cluster <- gsub("^>Cluster ", "", line)
  } else {
    # Extract the ASV ID (the number after ">")
    match <- regmatches(line, regexpr(">\\w+(-\\w+)*", line))
    if (length(match) > 0) {
      asv_id <- gsub(">", "", match)
      cluster_data <- rbind(cluster_data, data.frame(ASV_ID = asv_id, Cluster = current_cluster))
    }
  }
}

# Save to CSV
write.csv(cluster_data, output_csv, row.names = FALSE)

split_by_char <- function(x, char, ind) {
  if (grepl(char, x)) {
    strsplit(x, char)[[1]][[ind]]
  } else {
    x
  }
}

cluster_info <- read.csv(cluster_info_file)

cluster_data <- read.csv(output_csv) %>%
                  left_join(cluster_info %>%
                              mutate(Cluster.ID = as.integer(Cluster.ID)) %>%
                              select(Cluster.ID, Species),
                  by = c("Cluster" = "Cluster.ID")) %>%
                  mutate(Species_identified = Species) %>%
                  select(-Species)


df_natural_counts_all_ext <- df_natural_counts_all %>%
                              left_join(cluster_data, by = c("OTU" = "ASV_ID")) %>%
                              # go away empty and singleton ASVs
                              group_by(OTU) %>%
                              mutate(present_check = ifelse(Abundance > 0, 1, 0)) %>%
                              mutate(num_check = sum(present_check)) %>%
                              filter(num_check >= 5) %>%
                              ungroup() %>%
                              mutate(Cluster_simp = Vectorize(split_by_char)(Cluster, "-", 2)) %>%
                              mutate(Cluster_simp = ifelse(Species_identified == "", Cluster_simp, Species_identified)) %>%
                              mutate(Cluster_simp = ifelse(Species_identified == "" & !is.na(Species.1), paste0(Cluster_simp, "-", Species), Cluster_simp)) %>%
                              mutate(Cluster_simp = ifelse(Species_identified == "" & is.na(Species.1) & !is.na(Genus), paste0(Cluster_simp, "-", Genus, " unk"), Cluster_simp))


ggplot() +
  geom_beeswarm(data = df_natural_counts_all_ext %>%
                        # filter(grepl("mellifera", Host)) %>%
                        # remove asvs only found in 1 sample
                        group_by(OTU) %>%
                        mutate(present_check = ifelse(Abundance > 0, 1, 0)) %>%
                        mutate(num_check = sum(present_check)) %>%
                        filter(num_check >= 2) %>%
                        ungroup() %>%
                        group_by(Cluster_simp) %>%
                        # group_by(Cluster) %>%
                          mutate(sum_check = sum(Abundance, na.rm = T)) %>%
                          mutate(clust_size = n_distinct(OTU)) %>%
                          filter(clust_size > 1) %>%
                          ungroup() %>%
                          # filter(sum_check > 0) %>%
                        filter(!is.na(Genus)),
                aes(x = OTU, y = abs_counts, alpha = geq_passed, color = geq_passed),
                fill = "#999999",
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                shape = 21,
                stroke = 1,
                ) +
        scale_y_log10(
              breaks = scales::trans_breaks("log10", function(x) 10^x),
              labels = scales::trans_format("log10", scales::math_format(10^.x))
              # # axis limits
              # limits = c(1, 1e+10)
          ) +
        geom_line(data = df_natural_counts_all_ext %>%
                        # remove asvs only found in 1 sample
                        group_by(OTU) %>%
                        # filter(grepl("mellifera", Host)) %>%
                        mutate(present_check = ifelse(Abundance > 0, 1, 0)) %>%
                        mutate(num_check = sum(present_check)) %>%
                        filter(num_check >= 2) %>%
                        ungroup() %>%
                        group_by(Cluster_simp) %>%
                        # group_by(Cluster) %>%
                          mutate(sum_check = sum(Abundance, na.rm = T)) %>%
                          mutate(clust_size = n_distinct(OTU)) %>%
                          filter(clust_size > 1) %>%
                          ungroup() %>%
                          # filter(sum_check > 0) %>%
                        filter(!is.na(Genus)),
                aes(x = OTU, y = abs_counts, group = Sample)) +
                # aes(x = OTU, y = abs_counts, group = Sample, color = Host)) +
        facet_wrap(~Cluster_simp, scale = "free_x", ncol = 6) +
        # facet_wrap(~Cluster, scale = "free_x", ncol = 6) +
        labs(y = "Total bacterial abundance (log10)", x = "Treatment", fill = "Treatment") +
        # scale_fill_manual(values = strainColorsUniq) +
        scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
        scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
        # scale_color_manual(values = HostColors) +
        geom_hline(yintercept = 1e+4, linetype = "dashed") +
        make_theme(setFill = T, palettecolor = "Spectral", palettefill = "Set1", setCol = F,
                   x_angle = 45,
                  guide_nrow = 23, leg_pos = "right",
                  x_hj = 1, x_vj = 1, y_hj = 1)
ggsave(paste0("03_Analysis/FindNatural_16S/species_counts_natural_all_", clust_threshold, "_clust_simp.pdf"), width = 20, height = 20)
# ggsave(paste0("03_Analysis/FindNatural_16S/species_counts_natural_all_hosts_", clust_threshold, "_clust_simp.pdf"), width = 20, height = 20)
# ggsave("03_Analysis/FindNatural_16S/species_counts_natural_all_995_clust.pdf", width = 20, height = 20)


df_natural_counts_all_ext %>%
  filter(grepl("mellifera", Host)) %>%
  filter(geq_passed == "1") %>%
  group_by(Genus) %>%
  summarise(Cluster_simp = n_distinct(Cluster_simp), Genus) %>% unique() %>% view



# (pacbio_ampli_r_env) (base) aprasad@curnagl /nas/FAC/FBM/DMF/pengel/spirit/D2c/aprasad/20240399_aprasad_PriorityEffects$ 
# mamba install bioconductor-sssssss
# mamba install bioconductor-decipher
# both th eabove failed for decipher!
# 1.26.0 - original dada2

genera_of_int <- c("Lactobacillus", "Bifidobacterium", "Bombilactobacillus",
   "Frischella", "Snodgrassella", "Gilliamella", 
   "Commensalibacter", "Bartonella", "Dysgonomonas"
)

system("mkdir 03_Analysis/FindNatural_16S/heatmaps/")

for (my_genus in genera_of_int) {
  df_wide_gen <- df_natural_counts_all_ext %>%
                  filter(Genus == my_genus) %>%
                  # filter(grepl("mellifera", Host)) %>%
                  filter(!is.na(abs_counts)) %>%
                  select(Sample, OTU, abs_counts) %>%
                  pivot_wider(names_from = OTU, values_from = abs_counts)
  if (dim(df_wide_gen)[2] > 1) {
    print(my_genus)

    df_gen_mat <- df_wide_gen %>%
                  column_to_rownames("Sample") %>%
                  as.matrix() %>% t()

    clusters <- data.frame(OTU = rownames(df_gen_mat)) %>%
                  left_join(df_natural_counts_all_ext %>%
                            select(OTU, Cluster) %>%
                            unique(), by = "OTU") %>%
                            pull(Cluster)
    
    clusters_simp <- data.frame(OTU = rownames(df_gen_mat)) %>%
                  left_join(df_natural_counts_all_ext %>%
                            select(OTU, Cluster_simp) %>%
                            unique(), by = "OTU") %>%
                            pull(Cluster_simp)

    species <- unlist(lapply(clusters_simp, function(x) strsplit(x, "-")[[1]][2]))
    
    top_anno = rowAnnotation(Cluster = as.character(clusters_simp))

    row_sep <- as.factor(clusters_simp)
    
    # col_sep <- as.factor(clusters_simp)

    ht_gen <- Heatmap(df_gen_mat,
               name = "Counts",
               col = colorRamp2(c(0, 1000, 100000, 1000000, 100000000), c(brewer.pal(9, "PuBu")[1], brewer.pal(9, "PuBu")[2], brewer.pal(9, "PuBu")[6], brewer.pal(9, "PuBu")[7], brewer.pal(9, "PuBu")[8])),
              #  col = colorRamp2(c(-1, 0, 1), c("#b2182b", "#ffffff", "#2166ac")),
               cluster_rows = TRUE,
               cluster_columns = TRUE,
               column_title_rot = 0,
               column_title_gp = gpar(fontsize = 12),
               cluster_row_slices = T,
               border = T,
               right_annotation = top_anno,
               row_title_gp = gpar(fontsize = 12),
               row_title_rot = 0,
               show_row_names = T,
               show_heatmap_legend = T,
               row_title = "ASVs",
               column_title = "ASVs",
               row_split = row_sep,
              #  column_split = col_sep,
              #  heatmap_legend_param = list(title = "Spearman Correlation", at = c(-1, 0, 1), labels = c("-1", "0", "1"))
               heatmap_legend_param = list(title = "Counts", at = c(0, 1000, 100000, 1000000, 100000000), labels = c("0", "10", "100", "1000", "10000"))
    )
    # Draw the heatmap
    pdf(paste0("03_Analysis/FindNatural_16S/heatmaps/", my_genus, "_heatmap.pdf"), width = 20, height = 20)
    draw(ht_gen, heatmap_legend_side = "right", annotation_legend_side = "top")
    dev.off()

  }
}

for (my_genus in genera_of_int) {
  df_plot_gen <- df_natural_counts_all_ext %>%
                  filter(Genus == my_genus) %>%
                  group_by(Cluster) %>%
                  mutate(sum_check = sum(Abundance, na.rm = T)) %>%
                  mutate(clust_size = n_distinct(OTU)) %>%
                  filter(clust_size > 1) %>%
                  ungroup()
  if (dim(df_plot_gen)[1] >= 2) {
    ggplot() +
      geom_point(data = df_plot_gen,
                aes(x = OTU, y = abs_counts, fill = geq_passed),
                size = 2.5,
                alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
      # geom_beeswarm(data = df_plot_gen,
      #           aes(x = OTU, y = abs_counts, fill = geq_passed, color = geq_passed),
      #           size = 2.5,
      #           cex = 4,
      #           corral = "wrap",
      #           corral.width = 0.75,
      #           alpha = 0.95,
      #           shape = 21,
      #           stroke = 1,
      #           ) +
      scale_y_log10(
                  breaks = scales::trans_breaks("log10", function(x) 10^x),
                  labels = scales::trans_format("log10", scales::math_format(10^.x))
                  # # axis limits
                  # limits = c(1, 1e+10)
              ) +
            geom_line(data = df_plot_gen,
                aes(x = OTU, y = abs_counts, group = Sample, color = Host)) +
            facet_wrap(~Cluster, scale = "free_x", ncol = 6) +
            # facet_wrap(~Cluster, scale = "free_x", ncol = 6) +
            # scale_fill_manual(values = strainColorsUniq) +
            # scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
            scale_color_manual(values = HostColors) +
            scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
            geom_hline(yintercept = 1e+4, linetype = "dashed") +
            make_theme(setFill = T, palettefill = "Set1",
                      setCol = F, x_angle = 45,
                      guide_nrow = 23, leg_pos = "right",
                      x_hj = 1, x_vj = 1, y_hj = 1)
      ggsave(paste0("03_Analysis/FindNatural_16S/heatmaps/", my_genus, "_point_plot.pdf"))
  }

  
  df_plot_gen <- df_natural_counts_all_ext %>%
                  filter(Genus == my_genus) %>%
                  group_by(Cluster_simp) %>%
                  mutate(sum_check = sum(Abundance, na.rm = T)) %>%
                  mutate(clust_size = n_distinct(OTU)) %>%
                  filter(clust_size > 1) %>%
                  ungroup()

  if (dim(df_plot_gen)[1] >= 2) {
    ggplot() +
      geom_point(data = df_plot_gen,
                aes(x = OTU, y = abs_counts, fill = geq_passed),
                size = 2.5,
                alpha = 0.95,
                shape = 21,
                stroke = 1,
                ) +
      scale_y_log10(
                  breaks = scales::trans_breaks("log10", function(x) 10^x),
                  labels = scales::trans_format("log10", scales::math_format(10^.x))
                  # # axis limits
                  # limits = c(1, 1e+10)
              ) +
            geom_line(data = df_plot_gen,
                aes(x = OTU, y = abs_counts, group = Sample, color = Host)) +
            facet_wrap(~Cluster_simp, scale = "free_x", ncol = 6) +
            # facet_wrap(~Cluster, scale = "free_x", ncol = 6) +
            # scale_fill_manual(values = strainColorsUniq) +
            # scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
            scale_color_manual(values = HostColors) +
            scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
            geom_hline(yintercept = 1e+4, linetype = "dashed") +
            make_theme(setFill = T, palettefill = "Set1",
                      setCol = F, x_angle = 45,
                      guide_nrow = 23, leg_pos = "right",
                      x_hj = 1, x_vj = 1, y_hj = 1)
      ggsave(paste0("03_Analysis/FindNatural_16S/heatmaps/", my_genus, "cluster_simp_point_plot.pdf"))
  }


  for (a_cluster in unique(df_plot_gen$Cluster_simp)) {
    system(paste0("mkdir -p 03_Analysis/FindNatural_16S/heatmaps/", my_genus))
    df_cor_clust <- df_plot_gen %>%
                    filter(Cluster_simp == a_cluster)
    num_otus <- length(unique(df_cor_clust$OTU))
    if (num_otus > 1) {
      my_cor_mat <- cor(df_cor_clust %>% select(Sample, OTU, abs_counts) %>% pivot_wider(names_from = OTU, values_from = abs_counts) %>% column_to_rownames("Sample"))
      pdf(paste0("03_Analysis/FindNatural_16S/heatmaps/", my_genus, "/", a_cluster, "_correlation.pdf"), width = 20, height = 20)
      corrplot(my_cor_mat, method = "circle", tl.col = "black")
      dev.off()
    }

  }
}



```

<!-- Archived code -->

Code used for exploratory analyses - Not Up to date!

```{r}
seqtable_nochim <- read.csv("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/seqtable_nochim.csv")
asv_sequences <- colnames(seqtable_nochim)
sample_names <- unlist(lapply(rownames(seqtable_nochim), function(x) remove_extension(x, "_filt_reads.fastq.gz")))
sample_names <- unlist(lapply(sample_names, function(x) paste0("Exp-1-Bees_", x)))
dna <- Biostrings::DNAStringSet(asv_sequences)
names(dna) <- c(1:length(dna))
# write
Biostrings::writeXStringSet(dna, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/asv_sequences.fasta", format = "fasta", width = 10000)


seqtable_nochim <- read.csv("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/seqtable_nochim.csv")
asv_sequences <- colnames(seqtable_nochim)
sample_names <- unlist(lapply(rownames(seqtable_nochim), function(x) remove_extension(x, "_filt_reads.fastq.gz")))
sample_names <- unlist(lapply(sample_names, function(x) paste0("Exp-2-Bees_", x)))
dna <- Biostrings::DNAStringSet(asv_sequences)
names(dna) <- c(1:length(dna))
# write
Biostrings::writeXStringSet(dna, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/asv_sequences.fasta", format = "fasta", width = 10000)


seqtable_nochim <- read.csv("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/seqtable_nochim.csv")
asv_sequences <- colnames(seqtable_nochim)
sample_names <- unlist(lapply(rownames(seqtable_nochim), function(x) remove_extension(x, "_filt_reads.fastq.gz")))
sample_names <- unlist(lapply(sample_names, function(x) paste0("Exp-3-Bees_", x)))
dna <- Biostrings::DNAStringSet(asv_sequences)
names(dna) <- c(1:length(dna))
# write
Biostrings::writeXStringSet(dna, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/asv_sequences.fasta", format = "fasta", width = 10000)

```


```{r}
<!-- unsed -->
df_pvals_1 <- df_counts_collected_all %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)) %>%
                      # filter(geq_passed == "1") %>%
                      filter(CommunityCombo %in% c("m1", "m4", "m6", "m7", "1_rep")) %>%
                      group_by(Strain) %>%
                      filter(!Strain %in% c("ESL0820", "ESL0825")) %>%
                      group_by(Strain) %>%
                      wilcox_test(counts ~ arrival, data = ., p.adjust.method = "BH", alternative = "less",
                                  comparisons = list(
                                    c("second_1_rep", "second_m1"),
                                    c("second_1_rep", "second_m4"),
                                    c("second_1_rep", "second_m6"),
                                    c("second_1_rep", "second_m7")
                                  )
                      )

df_pvals_4 <- df_counts_collected_all %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)) %>%
                      filter(CommunityCombo %in% c("m1", "m4", "m6", "m7", "1_rep")) %>%
                      group_by(Strain) %>%
                      filter(!Strain %in% c("ESL0827", "ESL0199")) %>%
                      wilcox_test(counts ~ arrival, data = ., p.adjust.method = "BH",
                                  comparisons = list(c("first_1_rep", "first_m4"),
                                                     c("first_1_rep", "only_1_rep")
                                  )
                      )
df_pvals_6 <- df_counts_collected_all %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)) %>%
                      filter(CommunityCombo %in% c("m1", "m4", "m6", "m7", "1_rep")) %>%
                      group_by(Strain) %>%
                      filter(!Strain %in% c("ESL0295", "ESL0394")) %>%
                      wilcox_test(counts ~ arrival, data = ., p.adjust.method = "BH",
                                  comparisons = list(c("first_1_rep", "first_m6"),
                                                     c("first_1_rep", "only_1_rep")
                                  )
                      )
df_pvals_7 <- df_counts_collected_all %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)) %>%
                      filter(CommunityCombo %in% c("m1", "m4", "m6", "m7", "1_rep")) %>%
                      group_by(Strain) %>%
                      filter(!Strain %in% c("ESL0353_ESL0185", "ESL0263")) %>%
                      wilcox_test(counts ~ arrival, data = ., p.adjust.method = "BH",
                                  comparisons = list(c("first_1_rep", "first_m7"),
                                                     c("first_1_rep", "only_1_rep")
                                  )
                      )
df_all_vals <- bind_rows(df_pvals_1, df_pvals_4, df_pvals_6, df_pvals_7) %>% unique()
dim(df_all_vals)
df_all_vals_not_ns <- bind_rows(df_pvals_1, df_pvals_4, df_pvals_6, df_pvals_7) %>%
                filter(p.adj.signif != "ns") %>%
                  mutate(y.position = case_when(
                    group1 == "first_1_rep" & group2 == "first_m1" ~ 10,
                    group1 == "first_1_rep" & group2 == "only_1_rep" ~ 11,
                    group1 == "first_1_rep" & group2 == "first_m4" ~ 8,
                    group1 == "first_1_rep" & group2 == "only_1_rep" ~ 9,
                    group1 == "first_1_rep" & group2 == "first_m6" ~ 10,
                    group1 == "first_1_rep" & group2 == "only_1_rep" ~ 11,
                    group1 == "first_1_rep" & group2 == "first_m7" ~ 8,
                    group1 == "first_1_rep" & group2 == "only_1_rep" ~ 9,
                    TRUE ~ 10
                  )
                  ) %>% unique()
dim(df_all_vals)[1] - dim(df_all_vals_not_ns)[1]
```

### Check community composition of aliquots

```{r}
df_alq_heatmap <- df_species_count_e0 %>%
                    filter(grepl("alq", Treatment))
mat_alq_heatmap <- df_alq_heatmap %>% select(Sample, Strain, abs_counts) %>%
                      pivot_wider(names_from = Strain, values_from = abs_counts) %>%
                      column_to_rownames("Sample") %>%
                      as.matrix()
species_names <- data.frame(strain = colnames(mat_alq_heatmap)) %>% left_join(df_info, by = c("strain" = "strain")) %>% pull(spec_name)
genus_names <- unlist(lapply(species_names, function(x) strsplit(x, " ")[[1]][[1]]))
treatment <- data.frame(sample = rownames(mat_alq_heatmap)) %>% left_join(df_meta_pilot %>% select(Sample, Treatment), by = c("sample" = "Sample")) %>% pull(Treatment)
col_split <- genus_names %>% as.factor() %>% as.character()
row_split <- treatment %>% as.factor() %>% as.character()
top_anno <- HeatmapAnnotation(Species = as.character(species_names),
                                col = list(Species = speciesColors_uniq)
                                )
colnames(mat_alq_heatmap) <- data.frame(strain = colnames(mat_alq_heatmap)) %>%
                                left_join(df_info %>% select(spec_name, strain), by = c("strain")) %>%
                                mutate(spec_name = paste0(spec_name, " (", strain, ")")) %>%
                                pull(spec_name)
heatmap_obj = Heatmap(mat_alq_heatmap,
                      name = "Absolute count",
                      col = colorRamp2(c(0, 100, 1000, 10000, 100000, 1000000, 10000000, 100000000, 1000000000), 
                                       c(brewer.pal(9, "PuBu")[1], brewer.pal(9, "PuBu")[2], brewer.pal(9, "PuBu")[3], brewer.pal(9, "PuBu")[4], brewer.pal(9, "PuBu")[5], brewer.pal(9, "PuBu")[6], brewer.pal(9, "PuBu")[7], brewer.pal(9, "PuBu")[8], brewer.pal(9, "PuBu")[9])),
                      heatmap_legend_param = list(at = c(0, 100, 1000, 10000, 100000, 1000000, 10000000, 100000000, 1000000000),
                                                  legend_height = unit(5, "cm"),
                                                  labels = c(0, 100, 1000, 10000, 100000, 1000000, 10000000, 100000000, 1000000000)),
                      row_split = row_split,
                      column_split = col_split,
                      top_annotation = top_anno,
                      cluster_rows = F,
                      cluster_columns = F,
                      column_title_rot = 40,
                      cluster_row_slices = T,
                      border = T,
                      column_names_side = "top",
                      column_names_rot = 45,
                      column_title_side = "bottom",
                      row_title_rot = 0,
                      show_row_names = T,
                      show_row_dend = F,
                      show_column_dend = F,
                      show_heatmap_legend = T)
draw(heatmap_obj)
pdf("03_Analysis/Results/figures/pilot_experiment/alq_heatmap.pdf", width = 10, height = 15)
draw(heatmap_obj)
dev.off()

ggplot() +
  geom_tile(data = df_alq_heatmap, aes(x = Sample, y = Strain, fill = abs_counts)) +
  facet_grid(spec_name~factor(Treatment, c("alq-A", "alq-B", "alq-AB", "alq-SW")), scale = "free") +
  scale_fill_gradient2(
    low = "#ffffff",
    mid = brewer.pal(9, "PuBu")[4],
    high = brewer.pal(9, "BuGn")[7],
    na.value = "#ffffff",
    midpoint = 3.5,
    trans = "log10"
  ) +
  make_theme(setFill = F, leg_pos = "right", modify_guide = F,
             x_angle = 45, x_hj = 1, x_vj = 1,
             y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/alq_heatmap_ggplot_abs.pdf", width = 10, height = 15)

ggplot() +
  geom_tile(data = df_alq_heatmap, aes(x = Sample, y = Strain, fill = present)) +
  facet_grid(spec_name~factor(Treatment, c("alq-A", "alq-B", "alq-AB", "alq-SW")), scale = "free") +
  scale_fill_manual(values = c("0" = "#ffffff", "1" = brewer.pal(6, "PuBu")[6])) +
  make_theme(setFill = F, leg_pos = "right", modify_guide = F,
             x_angle = 45, x_hj = 1, x_vj = 1,
             y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/alq_heatmap_ggplot_presence.pdf", width = 10, height = 15)
```

### Plot species counts per sample

```{r}
ggplot() +
  geom_tile(data = df_species_count_e0 %>% filter(!grepl("alq", Treatment)), aes(x = Sample, y = Strain, fill = abs_counts)) +
  facet_grid(spec_name~factor(Treatment), scale = "free") +
  scale_fill_gradient2(
    low = "#ffffff",
    mid = brewer.pal(9, "PuBu")[4],
    high = brewer.pal(9, "BuGn")[7],
    na.value = "#ffffff",
    midpoint = 6,
    trans = "log10"
  ) +
  make_theme(setFill = F, leg_pos = "right", modify_guide = F,
             x_angle = 45, x_hj = 1, x_vj = 1,
             y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_heatmap_ggplot_abs.pdf", width = 15, height = 20)

ggplot() +
  geom_tile(data = df_species_count_e0 %>%
              filter(
                Treatment %in% c("0-3", "A-3", "B-3", "AB-1", "BA-1")
              ), aes(x = Sample, y = Strain, fill = abs_counts)) +
  facet_grid(spec_name~factor(Treatment), scale = "free") +
  scale_fill_gradient2(
    low = "#ffffff",
    mid = brewer.pal(9, "PuBu")[4],
    high = brewer.pal(9, "BuGn")[7],
    na.value = "#ffffff",
    midpoint = 6,
    trans = "log10"
  ) +
  make_theme(setFill = F, leg_pos = "right", modify_guide = F,
             x_angle = 45, x_hj = 1, x_vj = 1,
             y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_heatmap_ggplot_day1_abs.pdf", width = 15, height = 20)

ggplot() +
  geom_tile(data = df_species_count_e0 %>%
              filter(
                Treatment %in% c("0-3", "A-3", "B-3", "AB-3", "BA-3")
              ), aes(x = Sample, y = Strain, fill = abs_counts)) +
  facet_grid(spec_name~factor(Treatment), scale = "free") +
  scale_fill_gradient2(
    low = "#ffffff",
    mid = brewer.pal(9, "PuBu")[4],
    high = brewer.pal(9, "BuGn")[7],
    na.value = "#ffffff",
    midpoint = 6,
    trans = "log10"
  ) +
  make_theme(setFill = F, leg_pos = "right", modify_guide = F,
             x_angle = 45, x_hj = 1, x_vj = 1,
             y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_heatmap_ggplot_day3_abs.pdf", width = 15, height = 20)

ggplot() +
  geom_tile(data = df_species_count_e0 %>% filter(!grepl("alq", Treatment)), aes(x = Sample, y = Strain, fill = present)) +
  facet_grid(spec_name~factor(Treatment), scale = "free") +
  scale_fill_manual(values = c("0" = "#ffffff", "1" = brewer.pal(6, "PuBu")[6])) +
  make_theme(setFill = F, leg_pos = "right", modify_guide = F,
             x_angle = 45, x_hj = 1, x_vj = 1,
             y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_heatmap_ggplot_presence.pdf", width = 15, height = 20)
```

### Plot species counts per sample for day 1 and day 3

```{r}
# do the boxplot and beeswarm for the pilot experiment
# df_species_count_e0 %>% pull(Treatment) %>% unique()
df_plot_treatments <- df_species_count_e0 %>%
                        filter(Treatment %in% c("0-3", "A-0", "B-0", "AB-1", "BA-1", "alq-A", "alq-B", "alq-SW")) %>%
                        mutate(Type = ifelse(Treatment == "alq-SW", "controls", "bee")) %>%
                        mutate(Type = ifelse(Treatment == "alq-A", "controls", Type)) %>%
                        mutate(Type = ifelse(Treatment == "alq-B", "controls", Type)) %>%
                        mutate(Type = ifelse(grepl("0-3", Treatment), "controls", Type))

ggplot() +
  geom_boxplot(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = Strain),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts_org, group = Strain, fill = Strain, color = limit_passed),
                shape = 21, stroke = 1,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("No" = "#b9b9b9", "Yes" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_boxplot_ggplot_abs_day1.pdf", width = 15, height = 20)

ggplot() +
  geom_beeswarm(data = df_plot_treatments %>%
                        mutate(abs_counts_org = ifelse(abs_counts_org == 0, 0.1, abs_counts_org)),
                aes(x = Treatment, y = abs_counts_org, fill = Strain, group = Strain, color = limit_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.9,
                shape = 21,
                stroke = 1,
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("No" = "#b9b9b9", "Yes" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_beeswarm_ggplot_abs_day1.pdf", width = 15, height = 20)

df_plot_treatments <- df_species_count_e0 %>%
                        filter(Treatment %in% c("0-3", "A-3", "B-3", "AB-1", "BA-1", "alq-A", "alq-B", "alq-SW")) %>%
                        mutate(Type = ifelse(Treatment == "alq-SW", "controls", "bee")) %>%
                        mutate(Type = ifelse(Treatment == "alq-A", "controls", Type)) %>%
                        mutate(Type = ifelse(Treatment == "alq-B", "controls", Type)) %>%
                        mutate(Type = ifelse(grepl("0-3", Treatment), "controls", Type))

ggplot() +
  geom_boxplot(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = Strain),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts_org, group = Strain, fill = Strain, color = limit_passed),
                # color = "black",
                shape = 21,
                stroke = 1,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("No" = "#b9b9b9", "Yes" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_boxplot_ggplot_abs_day1_day3control.pdf", width = 15, height = 20)

ggplot() +
  geom_beeswarm(data = df_plot_treatments %>%
                        mutate(abs_counts_org = ifelse(abs_counts_org == 0, 0.1, abs_counts_org)),
                aes(x = Treatment, y = abs_counts_org, fill = Strain, group = Strain, color = limit_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.9,
                shape = 21,
                stroke = 1,
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("No" = "#b9b9b9", "Yes" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_beeswarm_ggplot_abs_day1_day3control.pdf", width = 15, height = 20)


df_plot_treatments <- df_species_count_e0 %>%
                        filter(Treatment %in% c("0-3", "A-3", "B-3", "AB-3", "BA-3", "alq-A", "alq-B", "alq-SW")) %>%
                        mutate(Type = ifelse(Treatment == "alq-SW", "controls", "bee")) %>%
                        mutate(Type = ifelse(Treatment == "alq-A", "controls", Type)) %>%
                        mutate(Type = ifelse(Treatment == "alq-B", "controls", Type)) %>%
                        mutate(Type = ifelse(grepl("0-3", Treatment), "controls", Type))

ggplot() +
  geom_boxplot(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = Strain),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts_org, group = Strain, fill = Strain, color = limit_passed),
                shape = 21,
                stroke = 1,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("No" = "#b9b9b9", "Yes" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_boxplot_ggplot_abs_day3.pdf", width = 15, height = 20)

ggplot() +
  geom_beeswarm(data = df_plot_treatments %>%
                        mutate(abs_counts_org = ifelse(abs_counts_org == 0, 0.1, abs_counts_org)),
                aes(x = Treatment, y = abs_counts_org, fill = Strain, group = Strain, color = limit_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.9,
                shape = 21,
                stroke = 1,
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("No" = "#b9b9b9", "Yes" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_beeswarm_ggplot_abs_day3.pdf", width = 15, height = 20)

df_plot_treatments <- df_species_count_e0 %>%
                        filter(Treatment %in% c("A-0", "A-3", "B-3", "A-0", "B-0", "0-3", "alq-A", "alq-B", "alq-SW")) %>%
                        mutate(Type = ifelse(Treatment == "alq-SW", "controls", "bee")) %>%
                        mutate(Type = ifelse(Treatment == "alq-A", "controls", Type)) %>%
                        mutate(Type = ifelse(Treatment == "alq-B", "controls", Type)) %>%
                        mutate(Type = ifelse(grepl("0-3", Treatment), "controls", Type))

ggplot() +
  geom_boxplot(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = Strain),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts_org, group = Strain, fill = Strain, color = limit_passed),
                color = "black",
                shape = 21,
                stroke = 1,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("No" = "#b9b9b9", "Yes" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_boxplot_ggplot_abs_d0_v_d3.pdf", width = 15, height = 20)

ggplot() +
  geom_beeswarm(data = df_plot_treatments %>%
                        mutate(abs_counts_org = ifelse(abs_counts_org == 0, 0.1, abs_counts_org)),
                aes(x = Treatment, y = abs_counts_org, fill = Strain, group = Strain, color = limit_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.9,
                shape = 21,
                stroke = 1,
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("No" = "#b9b9b9", "Yes" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_beeswarm_ggplot_abs_d0_v_d3.pdf", width = 15, height = 20)

df_plot_treatments <- df_species_count_e0 %>%
                        filter(Treatment %in% c("A-3", "B-3", "AB-0", "0-3", "alq-A", "alq-B", "alq-SW")) %>%
                        mutate(Type = ifelse(Treatment == "alq-SW", "controls", "bee")) %>%
                        mutate(Type = ifelse(Treatment == "alq-A", "controls", Type)) %>%
                        mutate(Type = ifelse(Treatment == "alq-B", "controls", Type)) %>%
                        mutate(Type = ifelse(grepl("0-3", Treatment), "controls", Type))

ggplot() +
  geom_boxplot(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = Strain),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts_org, group = Strain, fill = Strain, color = limit_passed),
                # color = "black",
                shape = 21,
                stroke = 1,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("No" = "#b9b9b9", "Yes" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_boxplot_ggplot_abs_only_v_together.pdf", width = 15, height = 20)

ggplot() +
  geom_beeswarm(data = df_plot_treatments %>%
                        mutate(abs_counts = ifelse(abs_counts == 0, 0.1, abs_counts)),
                aes(x = Treatment, y = abs_counts, fill = Strain, group = Strain, color = limit_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.9,
                shape = 21,
                stroke = 1,
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("No" = "#b9b9b9", "Yes" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_beeswarm_ggplot_abs_only_v_together.pdf", width = 15, height = 20)


df_plot_treatments <- df_species_count_e0 %>%
                        filter(Treatment %in% c("A-3", "B-3", "AB-3", "BA-3", "AB-0", "0-3", "alq-A", "alq-B", "alq-SW")) %>%
                        mutate(Type = ifelse(Treatment == "alq-SW", "controls", "bee")) %>%
                        mutate(Type = ifelse(Treatment == "alq-A", "controls", Type)) %>%
                        mutate(Type = ifelse(Treatment == "alq-B", "controls", Type)) %>%
                        mutate(Type = ifelse(grepl("0-3", Treatment), "controls", Type))

ggplot() +
  geom_boxplot(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = Strain),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts_org, group = Strain, fill = Strain, color = limit_passed),
                # color = "black",
                shape = 21,
                stroke = 1,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("No" = "#b9b9b9", "Yes" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_boxplot_ggplot_abs_d3_w_together.pdf", width = 15, height = 20)

ggplot() +
  geom_beeswarm(data = df_plot_treatments %>%
                        mutate(abs_counts_org = ifelse(abs_counts_org == 0, 0.1, abs_counts_org)),
                aes(x = Treatment, y = abs_counts_org, fill = Strain, group = Strain, color = limit_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.9,
                shape = 21,
                stroke = 1,
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("No" = "#b9b9b9", "Yes" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/pilot_experiment/species_beeswarm_ggplot_abs_d3_w_together.pdf", width = 15, height = 20)


# df_plot_treatments <- df_species_count_e0 %>%
#                         filter(Treatment %in% c("0-3", "A-3", "B-3", "AB-1", "BA-1", "alq-A", "alq-B")) %>%

  
```

### Inspect taxonomy

Redo this section later if asked for by someone. Did this before and now not really using in further analysis.

```{r}
system("wget https://zenodo.org/record/3986799/files/silva_nr99_v138_wSpecies_train_set.fa.gz  -O 03_Analysis/Results/tables/silva_nr99_v138_wSpecies_train_set.fa.gz")
system("wget https://zenodo.org/record/3986799/files/silva_nr99_v138_wSpecies_train_set.fa.gz  -O 03_Analysis/Results/tables/silva_nr99_v138_train_set.fa.gz")
system("wget https://zenodo.org/record/3986799/files/silva_species_assignment_v138.fa.gz  -O 03_Analysis/Results/tables/silva_species_assignment_v138.fa.gz")

silva_DB_assignTax <- "03_Analysis/Results/tables/silva_nr99_v138_wSpecies_train_set.fa.gz"
silva_DB_assignSpec <- "03_Analysis/Results/tables/silva_species_assignment_v138.fa.gz"
custom_taxonomy <- "00_StrainSelection/custom_db_from_isolates.fa"

dada2_1 <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/RDS/dada2.rds")
seqtable_1 <- makeSequenceTable(dada2_1); dim(seqtable_1)
seqtable_nochim_1 <- removeBimeraDenovo(seqtable_1, method = "consensus", multithread = TRUE, verbose = TRUE); dim(seqtable_1)

dada2_2 <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/RDS/dada2.rds")
seqtable_2 <- makeSequenceTable(dada2_2); dim(seqtable_2)
seqtable_nochim_2 <- removeBimeraDenovo(seqtable_2, method = "consensus", multithread = TRUE, verbose = TRUE); dim(seqtable_2)

dada2_3 <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/RDS/dada2.rds")
seqtable_3 <- makeSequenceTable(dada2_3); dim(seqtable_3)
seqtable_nochim_3 <- removeBimeraDenovo(seqtable_3, method = "consensus", multithread = TRUE, verbose = TRUE); dim(seqtable_3)

taxonomy_1 <- assignTaxonomy(seqtable_1, silva_DB_assignTax, minBoot = 80, multithread=TRUE, verbose = TRUE)
# saveRDS(taxonomy_1, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/taxonomy_before_species_assigned.RDS")
taxonomy_1 <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/taxonomy.RDS")
taxonomy_1_spec <- addSpecies(taxonomy_1, silva_DB_assignSpec, allowMultiple = F, verbose = T)
# saveRDS(taxonomy_1_spec, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/taxonomy_spec.RDS")
taxonomy_1_spec <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/taxonomy_spec.RDS")
taxonomy_1_custom <- addSpecies(taxonomy_1_spec, custom_taxonomy, allowMultiple = F, verbose = T)
saveRDS(taxonomy_1_custom, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/taxonomy_custom.RDS")
write.csv(taxonomy_1_custom, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/taxonomy_custom.csv")
# add counts to taxonomy table
taxonomy_1_custom <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/taxonomy_custom.RDS") %>%
                      as.data.frame(.name_repair = "minimal")
colnames(taxonomy_1_custom) <- c("Kingdom","Phylum","Class","Order","Family","Genus","Species","Species_exact","Strain")
taxonomy_1_custom_w_counts <- t(seqtable_nochim_1) %>% as.data.frame %>% rownames_to_column("ASV") %>%
                                left_join(taxonomy_1_custom %>% rownames_to_column("ASV"), by = c("ASV"))
write.csv(taxonomy_1_custom_w_counts, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment1/taxonomy_custom_w_counts.csv")

taxonomy_2 <- assignTaxonomy(seqtable_2, silva_DB_assignTax, minBoot = 80, multithread=TRUE, verbose = TRUE)
saveRDS(taxonomy_2, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/taxonomy_before_species_assigned.RDS")
taxonomy_2 <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/taxonomy.RDS")
taxonomy_2_spec <- addSpecies(taxonomy_2, silva_DB_assignSpec, allowMultiple = F, verbose = T)
saveRDS(taxonomy_2_spec, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/taxonomy_spec.RDS")
taxonomy_2_spec <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/taxonomy_spec.RDS")
taxonomy_2_custom <- addSpecies(taxonomy_2_spec, custom_taxonomy, allowMultiple = F, verbose = T)
saveRDS(taxonomy_2_custom, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/taxonomy_custom.RDS")
write.csv(taxonomy_2_custom, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/taxonomy_custom.csv")
# add counts to taxonomy table
taxonomy_2_custom <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/taxonomy_custom.RDS") %>%
                      as.data.frame(.name_repair = "minimal")
colnames(taxonomy_2_custom) <- c("Kingdom","Phylum","Class","Order","Family","Genus","Species","Species_exact","Strain")
taxonomy_2_custom_w_counts <- t(seqtable_nochim_2) %>% as.data.frame %>% rownames_to_column("ASV") %>%
                                left_join(taxonomy_2_custom %>% rownames_to_column("ASV"), by = c("ASV"))
write.csv(taxonomy_2_custom_w_counts, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment2/taxonomy_custom_w_counts.csv")

taxonomy_3 <- assignTaxonomy(seqtable_3, silva_DB_assignTax, minBoot = 80, multithread=TRUE, verbose = TRUE)
saveRDS(taxonomy_3, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/taxonomy_before_species_assigned.RDS")
taxonomy_3 <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/taxonomy.RDS")
taxonomy_3_spec <- addSpecies(taxonomy_3, silva_DB_assignSpec, allowMultiple = F, verbose = T)
saveRDS(taxonomy_3_spec, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/taxonomy_spec.RDS")
taxonomy_3_spec <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/taxonomy_spec.RDS")
taxonomy_3_custom <- addSpecies(taxonomy_3_spec, custom_taxonomy, allowMultiple = F, verbose = T)
saveRDS(taxonomy_3_custom, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/taxonomy_custom.RDS")
write.csv(taxonomy_3_custom, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/taxonomy_custom.csv")
# add counts to taxonomy table
taxonomy_3_custom <- readRDS("03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/taxonomy_custom.RDS") %>%
                      as.data.frame(.name_repair = "minimal")
colnames(taxonomy_3_custom) <- c("Kingdom","Phylum","Class","Order","Family","Genus","Species","Species_exact","Strain")
taxonomy_3_custom_w_counts <- t(seqtable_nochim_3) %>% as.data.frame %>% rownames_to_column("ASV") %>%
                                left_join(taxonomy_3_custom %>% rownames_to_column("ASV"), by = c("ASV"))
write.csv(taxonomy_3_custom_w_counts, "03_Analysis/PriorityEffectsExperiment/00_preprocessing/Experiment3/taxonomy_custom_w_counts.csv")
```

```{r}
ggplot() +
  geom_bar(data = df_total_reads_compare,
        aes(x = ID, y = total_reads, fill = Treatment_simp), stat = "identity") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = TreatmentColors) +
  facet_wrap(~ CommunityCombo, scale = "free") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Total reads", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 7,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/total_reads_per_sample.pdf")

ggplot() +
  geom_bar(data = df_total_reads_compare %>% filter(Type == "bee"),
        aes(x = ID, y = total_reads, fill = Treatment_simp), stat = "identity") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = TreatmentColors) +
  facet_wrap(~ CommunityCombo, scale = "free") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Total reads", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 7,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/total_reads_per_sample_bees.pdf")


ggplot() +
  geom_bar(data = df_total_reads_compare,
        aes(x = ID, y = bac_copies, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_total_reads_compare,
        aes(x = ID, y = total_reads, fill = Treatment_simp), color = "#000000", alpha = 0.5, stat = "identity") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  scale_fill_manual(values = TreatmentColors) +
  facet_wrap(~ CommunityCombo, scale = "free") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Bacteria copies", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 7,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/bac_copies_per_sample.pdf")


ggplot() +
  geom_bar(data = df_total_reads_compare %>% filter(Type == "bee"),
        aes(x = ID, y = bac_copies, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_total_reads_compare,
        aes(x = ID, y = total_reads, fill = Treatment_simp), color = "#000000", alpha = 0.5, stat = "identity") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Bacterial copies for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  facet_wrap(~ CommunityCombo, scale = "free") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Bacteria copies", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 7,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/bac_copies_per_sample_bees.pdf")



ggplot() +
  geom_bar(data = df_total_reads_compare,
        aes(x = ID, y = sample_geq, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_total_reads_compare,
        aes(x = ID, y = total_reads, fill = Treatment_simp), color = "#000000", alpha = 0.5, stat = "identity") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  facet_wrap(~ CommunityCombo, scale = "free") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 7,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/GEQ_per_sample_bees.pdf")


ggplot() +
  geom_bar(data = df_total_reads_compare,
        aes(x = ID, y = total_abs, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_total_reads_compare,
        aes(x = ID, y = sample_geq, fill = Treatment_simp), color = "#000000", alpha = 0.5, stat = "identity") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  facet_wrap(~ CommunityCombo, scale = "free") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 7,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/GEQ_sum_abs_per_sample_bees.pdf")

ggplot() +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "1" & Experiment == "1") %>% filter(Type == "bee"),
        aes(x = Strain, y = abs_counts_org, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "1" & Experiment == "1") %>% filter(Type == "bee"),
        aes(x = Strain, y = sample_geq), color = "#000000", alpha = 0, stat = "identity", linewidth = 0.2) +
  facet_wrap(~ ID, scale = "free_x") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 4,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/Strain_abundace_w_threshold_1.pdf")

ggplot() +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "1" & Experiment == "3") %>% filter(Type == "bee"),
        aes(x = Strain, y = abs_counts_org, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "1" & Experiment == "3") %>% filter(Type == "bee"),
        aes(x = Strain, y = sample_geq), color = "#000000", alpha = 0, stat = "identity", linewidth = 0.2) +
  facet_wrap(~ ID, scale = "free_x") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 4,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/Strain_abundace_w_threshold_1_rep.pdf")


ggplot() +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "2") %>% filter(Type == "bee"),
        aes(x = Strain, y = abs_counts_org, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "2") %>% filter(Type == "bee"),
        aes(x = Strain, y = sample_geq), color = "#000000", alpha = 0, stat = "identity", linewidth = 0.2) +
  facet_wrap(~ ID, scale = "free_x") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 4,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/Strain_abundace_w_threshold_2.pdf")

ggplot() +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "3") %>% filter(Type == "bee"),
        aes(x = Strain, y = abs_counts_org, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "3") %>% filter(Type == "bee"),
        aes(x = Strain, y = sample_geq), color = "#000000", alpha = 0, stat = "identity", linewidth = 0.2) +
  facet_wrap(~ ID, scale = "free_x") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 4,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/Strain_abundace_w_threshold_3.pdf")

ggplot() +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "4") %>% filter(Type == "bee"),
        aes(x = Strain, y = abs_counts_org, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "4") %>% filter(Type == "bee"),
        aes(x = Strain, y = sample_geq), color = "#000000", alpha = 0, stat = "identity", linewidth = 0.2) +
  facet_wrap(~ ID, scale = "free_x") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 4,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/Strain_abundace_w_threshold_4.pdf")

ggplot() +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "5") %>% filter(Type == "bee"),
        aes(x = Strain, y = abs_counts_org, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "5") %>% filter(Type == "bee"),
        aes(x = Strain, y = sample_geq), color = "#000000", alpha = 0, stat = "identity", linewidth = 0.2) +
  facet_wrap(~ ID, scale = "free_x") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 4,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/Strain_abundace_w_threshold_5.pdf")

ggplot() +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "6") %>% filter(Type == "bee"),
        aes(x = Strain, y = abs_counts_org, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "6") %>% filter(Type == "bee"),
        aes(x = Strain, y = sample_geq), color = "#000000", alpha = 0, stat = "identity", linewidth = 0.2) +
  facet_wrap(~ ID, scale = "free_x") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 4,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/Strain_abundace_w_threshold_6.pdf")

ggplot() +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "m1") %>% filter(Type == "bee"),
        aes(x = Strain, y = abs_counts_org, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "m1") %>% filter(Type == "bee"),
        aes(x = Strain, y = sample_geq), color = "#000000", alpha = 0, stat = "identity", linewidth = 0.2) +
  facet_wrap(~ ID, scale = "free_x") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 4,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/Strain_abundace_w_threshold_m1.pdf")

ggplot() +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "m4") %>% filter(Type == "bee"),
        aes(x = Strain, y = abs_counts_org, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "m4") %>% filter(Type == "bee"),
        aes(x = Strain, y = sample_geq), color = "#000000", alpha = 0, stat = "identity", linewidth = 0.2) +
  facet_wrap(~ ID, scale = "free_x") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 4,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/Strain_abundace_w_threshold_m4.pdf")

ggplot() +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "m6") %>% filter(Type == "bee"),
        aes(x = Strain, y = abs_counts_org, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "m6") %>% filter(Type == "bee"),
        aes(x = Strain, y = sample_geq), color = "#000000", alpha = 0, stat = "identity", linewidth = 0.2) +
  facet_wrap(~ ID, scale = "free_x") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 4,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/Strain_abundace_w_threshold_m6.pdf")

ggplot() +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "m7") %>% filter(Type == "bee"),
        aes(x = Strain, y = abs_counts_org, fill = Treatment_simp), stat = "identity") +
  geom_bar(data = df_species_count_all %>% filter(CommunityCombo == "m7") %>% filter(Type == "bee"),
        aes(x = Strain, y = sample_geq), color = "#000000", alpha = 0, stat = "identity", linewidth = 0.2) +
  facet_wrap(~ ID, scale = "free_x") +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  ggtitle("Threshold for each sample and total number reads shown by black border") +
  scale_fill_manual(values = TreatmentColors) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  labs(x = "Sample", y = "Sample GEQ", fill = "Treatment") +
  make_theme(setFill = F, setCol = F, x_size = 4,
             x_angle = 45, x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/colonization_success/Strain_abundace_w_threshold_m7.pdf")

```

## Check community composition bees across experiments

```{r}
df_species_count_all_exp <- df_species_count_all %>%
                            filter(Experiment != "none")
for (combo in unique(df_species_count_all_exp$CommunityCombo)) {
  if (combo == "SW") {
    ggplot() +
        geom_tile(data = df_species_count_all %>%
                          filter(CommunityCombo == combo),
        aes(x = ID, y = Strain, fill = abs_counts_org)) +
        facet_grid(spec_name~Type+Treatment, scale = "free", space = "free") +
        scale_fill_gradient2(
          low = "#ffffff",
          mid = brewer.pal(9, "PuBu")[4],
          high = brewer.pal(9, "BuGn")[7],
          na.value = "#ffffff",
          midpoint = 3.5,
          trans = "log10"
        ) +
        make_theme(theme = theme_bw(), setFill = F, leg_pos = "right", modify_guide = F,
                  x_angle = 45, x_hj = 1, x_vj = 1,
                  y_hj = 1)
        ggsave(paste0("03_Analysis/Results/figures/priority_effects_experiment/bee_gut_abs_counts/bee_gut_comp_heatmap_comm_", combo, ".pdf"), width = 10, height = 15)
  } else {
    if (grepl("m", combo)) {
      ggplot() +
        geom_tile(data = df_species_count_all %>%
                          filter(Experiment == "3") %>%
                          filter(CommunityCombo %in% c("1", combo)),
        aes(x = ID, y = Strain, fill = abs_counts_org)) +
        facet_grid(spec_name~Type+Treatment, scale = "free", space = "free") +
        scale_fill_gradient2(
          low = "#ffffff",
          mid = brewer.pal(9, "PuBu")[4],
          high = brewer.pal(9, "BuGn")[7],
          na.value = "#ffffff",
          midpoint = 3.5,
          trans = "log10"
        ) +
        make_theme(theme = theme_bw(), setFill = F, leg_pos = "right", modify_guide = F,
                  x_angle = 45, x_hj = 1, x_vj = 1,
                  y_hj = 1)
        ggsave(paste0("03_Analysis/Results/figures/priority_effects_experiment/bee_gut_abs_counts/bee_gut_comp_heatmap_comm_", combo, ".pdf"), width = 10, height = 15)
    } else {
      ggplot() +
        geom_tile(data = df_species_count_all %>%
                          filter(Experiment != "3") %>%
                          filter(CommunityCombo == combo),
        aes(x = ID, y = Strain, fill = abs_counts_org)) +
        facet_grid(spec_name~Type+Treatment, scale = "free", space = "free") +
        scale_fill_gradient2(
          low = "#ffffff",
          mid = brewer.pal(9, "PuBu")[4],
          high = brewer.pal(9, "BuGn")[7],
          na.value = "#ffffff",
          midpoint = 3.5,
          trans = "log10"
        ) +
        make_theme(theme = theme_bw(), setFill = F, leg_pos = "right", modify_guide = F,
                  x_angle = 45, x_hj = 1, x_vj = 1,
                  y_hj = 1)
        ggsave(paste0("03_Analysis/Results/figures/priority_effects_experiment/bee_gut_abs_counts/bee_gut_comp_heatmap_comm_", combo, ".pdf"), width = 10, height = 15)
    }
  }
}
combo = "1"
ggplot() +
        geom_tile(data = df_species_count_all %>%
                          filter(CommunityCombo == combo),
        aes(x = ID, y = Strain, fill = abs_counts_org)) +
        facet_grid(spec_name~Type+Treatment, scale = "free", space = "free") +
        scale_fill_gradient2(
          low = "#ffffff",
          mid = brewer.pal(9, "PuBu")[4],
          high = brewer.pal(9, "BuGn")[7],
          na.value = "#ffffff",
          midpoint = 3.5,
          trans = "log10"
        ) +
        make_theme(theme = theme_bw(), setFill = F, leg_pos = "right", modify_guide = F,
                  x_angle = 45, x_hj = 1, x_vj = 1,
                  y_hj = 1)
ggsave(paste0("03_Analysis/Results/figures/priority_effects_experiment/bee_gut_abs_counts/bee_gut_comp_heatmap_comm_", combo, "_both.pdf"), width = 10, height = 15)
```


### Plot strain counts across selected treatments

#### Community combination 1 in Experiment 1

```{r}
df_plot_treatments <- df_species_count_all %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "A1", "B1", "MD", "extraction_blank")) %>%
                        filter(Experiment == "1") %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments %>% filter(geq_passed == 1),
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), color = geq_passed, alpha = geq_passed),
                # color = "black",
                stroke = 1,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_combo_1.pdf", width = 20, height = 20)

# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm_combo_1.pdf", width = 20, height = 20)
```

#### Community combination 1 in Experiment 1 and 3

```{r}
df_plot_treatments <- df_species_count_all %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "A1", "B1", "MD", "extraction_blank")) %>%
                        filter(Experiment %in% c("1", "3")) %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), shape = Experiment, color = geq_passed, alpha = geq_passed),
                # color = "black",
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_shape_manual(values = c("1" = 24, "3" = 21)) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_combo_1_both.pdf", width = 20, height = 20)


# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity", color = "#000000", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, shape = Experiment),
                size = 2,
                cex = 4,
                corral = "gutter",
                corral.width = 0.75,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_shape_manual(values = c("1" = 24, "3" = 21)) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm_combo_1_both.pdf", width = 20, height = 20)

```

#### Community combination 2 in Experiment 1

```{r}
df_plot_treatments <- df_species_count_all %>%
                        filter(Treatment %in% c("A2-0", "B2-0", "A2-B2", "B2-A2", "A2", "B2", "MD", "extraction_blank")) %>%
                        filter(Experiment == "1") %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments %>% filter(geq_passed == 1),
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), color = geq_passed, alpha = geq_passed),
                # color = "black",
                stroke = 1,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_combo_2.pdf", width = 20, height = 20)

# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm_combo_2.pdf", width = 20, height = 20)
```

#### Community combination 3 in Experiment 1

```{r}
df_plot_treatments <- df_species_count_all %>%
                        filter(Treatment %in% c("A3-0", "B3-0", "A3-B3", "B3-A3", "A3", "B3", "MD", "extraction_blank")) %>%
                        filter(Experiment == "1") %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments %>% filter(geq_passed == 1),
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), color = geq_passed, alpha = geq_passed),
                # color = "black",
                stroke = 1,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_combo_3.pdf", width = 20, height = 20)

# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm_combo_3.pdf", width = 20, height = 20)
```

#### Community combination 4 in Experiment 2

```{r}
df_plot_treatments <- df_species_count_all %>%
                        filter(Treatment %in% c("A4-0", "B4-0", "A4-B4", "B4-A4", "A4", "B4", "MD", "extraction_blank")) %>%
                        filter(Experiment == "2") %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments %>% filter(geq_passed == 1),
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), color = geq_passed, alpha = geq_passed),
                # color = "black",
                stroke = 1,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_combo_4.pdf", width = 20, height = 20)

# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm_combo_4.pdf", width = 20, height = 20)
```

#### Community combination 5 in Experiment 2

```{r}
df_plot_treatments <- df_species_count_all %>%
                        filter(Treatment %in% c("A5-0", "B5-0", "A5-B5", "B5-A5", "A5", "B5", "MD", "extraction_blank")) %>%
                        filter(Experiment == "2") %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments %>% filter(geq_passed == 1),
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), color = geq_passed, alpha = geq_passed),
                # color = "black",
                stroke = 1,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_combo_5.pdf", width = 20, height = 20)

# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm_combo_5.pdf", width = 20, height = 20)
```

#### Community combination 6 in Experiment 2

```{r}
df_plot_treatments <- df_species_count_all %>%
                        filter(Treatment %in% c("A6-0", "B6-0", "A6-B6", "B6-A6", "A6", "B6", "MD", "extraction_blank")) %>%
                        filter(Experiment == "2") %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments %>% filter(geq_passed == 1),
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), color = geq_passed, alpha = geq_passed),
                # color = "black",
                stroke = 1,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_combo_6.pdf", width = 20, height = 20)

# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm_combo_6.pdf", width = 20, height = 20)
```

#### Community dropout condition m1 in Experiment 3

This drops out the species Bifidobacterium apousia.

```{r}
df_plot_treatments <- df_species_count_all %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am1-B1", "Bm1-A1", "A1", "B1", "MD", "dropout_community", "extraction_blank")) %>%
                        filter(CommunityCombo %in% c("SW", "1", "m1")) %>%
                        filter(Experiment == "3") %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "aliquot", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type)) %>%
                        mutate(Treatment = ifelse(ID == "am1", "Am1", Treatment)) %>%
                        mutate(Treatment = ifelse(ID == "bm1", "Bm1", Treatment))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments %>% filter(geq_passed == 1),
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), color = geq_passed, alpha = geq_passed),
                # color = "black",
                stroke = 1,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_combo_m1.pdf", width = 20, height = 20)

# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm_combo_m1.pdf", width = 20, height = 20)
```

#### Community dropout condition m4 in Experiment 3

This drops out Bifidobacterium sp1.

```{r}
df_plot_treatments <- df_species_count_all %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am4-B1", "Bm4-A1", "A1", "B1", "MD", "dropout_community", "extraction_blank")) %>%
                        filter(CommunityCombo %in% c("SW", "1", "m4")) %>%
                        filter(Experiment == "3") %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "aliquot", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type)) %>%
                        mutate(Treatment = ifelse(ID == "am4", "Am4", Treatment)) %>%
                        mutate(Treatment = ifelse(ID == "bm4", "Bm4", Treatment))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments %>% filter(geq_passed == 1),
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), color = geq_passed, alpha = geq_passed),
                # color = "black",
                stroke = 1,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_combo_m4.pdf", width = 20, height = 20)

# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm_combo_m4.pdf", width = 20, height = 20)
```

#### Community dropout condition m6 in Experiment 3

This drops out Bombilactobacillus mellis. 

```{r}
df_plot_treatments <- df_species_count_all %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am6-B1", "Bm6-A1", "A1", "B1", "MD", "dropout_community", "extraction_blank")) %>%
                        filter(CommunityCombo %in% c("SW", "1", "m6")) %>%
                        filter(Experiment == "3") %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "aliquot", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type)) %>%
                        mutate(Treatment = ifelse(ID == "am6", "Am6", Treatment)) %>%
                        mutate(Treatment = ifelse(ID == "bm6", "Bm6", Treatment))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments %>% filter(geq_passed == 1),
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), color = geq_passed, alpha = geq_passed),
                # color = "black",
                stroke = 1,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_combo_m6.pdf", width = 20, height = 20)

# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm_combo_m6.pdf", width = 20, height = 20)
```


#### Community dropout condition m7 in Experiment 3

This drops out Lactobacillus apis.

```{r}
df_plot_treatments <- df_species_count_all %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1", "Am7-B1", "Bm7-A1", "A1", "B1", "MD", "dropout_community", "extraction_blank")) %>%
                        filter(CommunityCombo %in% c("SW", "1", "m7")) %>%
                        filter(Experiment == "3") %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "aliquot", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type)) %>%
                        mutate(Treatment = ifelse(ID == "am7", "Am7", Treatment)) %>%
                        mutate(Treatment = ifelse(ID == "bm7", "Bm7", Treatment))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments %>% filter(geq_passed == 1),
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), color = geq_passed, alpha = geq_passed),
                # color = "black",
                stroke = 1,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_combo_m7.pdf", width = 20, height = 20)

# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment, group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = Treatment, y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm_combo_m7.pdf", width = 20, height = 20)
```


#### Community all across all treatments


```{r}
df_plot_treatments <- df_species_count_all %>%
                        mutate(CommunityCombo = ifelse(CommunityCombo == "1" & Experiment == "3", "1\'", CommunityCombo)) %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "aliquot", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type)) %>%
                        mutate(Type = ifelse(Treatment == "extraction_blank", "controls", Type)) %>%
                        mutate(Treatment = ifelse(ID == "am7", "Am7", Treatment)) %>%
                        mutate(Treatment = ifelse(ID == "bm7", "Bm7", Treatment))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment_simp, group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments %>% filter(geq_passed == "1"),
                aes(x = Treatment_simp, y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = Treatment_simp, y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), color = geq_passed, alpha = geq_passed),
                # color = "black",
                stroke = 1,
                shape = 21,
                size = 1,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_all.pdf", width = 20, height = 20)

# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = Treatment_simp, group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = Treatment_simp, y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 1,
                cex = 3,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 0.5
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Type, scale = "free_x") +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm_all.pdf", width = 20, height = 20)


```


```{r}
a
```


```{r}
ggplot()  +
  geom_bar(data = df_species_count_all_combined %>% filter(Experiment != "pilot"),
                aes(x = arrival, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_species_count_all_combined %>% filter(Experiment != "pilot"),
                aes(x = arrival, y = abs_counts),
                outlier.shape = NA
  ) +
  geom_point(data = df_species_count_all_combined %>% filter(Experiment != "pilot"),
                aes(x = arrival, y = abs_counts, group = Strain, fill = CommunityCombo, color = geq_passed, alpha = geq_passed),
                # color = "black",
                # stroke = 1,
                size = 2,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~spec_name_uniq, scale = "free_x") +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = T, palettefill = "Spectral", setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_arrival/all_boxplot.pdf", width = 20, height = 20)

ggplot()  +
  geom_bar(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(!grepl("m", CommunityCombo)),
                aes(x = arrival, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(!grepl("m", CommunityCombo)),
                aes(x = arrival, y = abs_counts),
                outlier.shape = NA
  ) +
  geom_point(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(!grepl("m", CommunityCombo)),
                aes(x = arrival, y = abs_counts, group = Strain, fill = CommunityCombo, color = geq_passed, alpha = geq_passed),
                # color = "black",
                # stroke = 1,
                size = 2,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~spec_name_uniq, scale = "free_x") +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = T, palettefill = "Spectral", setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_arrival/all_full_boxplot.pdf", width = 20, height = 20)

ggplot()  +
  geom_bar(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(grepl("m", CommunityCombo)),
                aes(x = arrival, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(grepl("m", CommunityCombo)),
                aes(x = arrival, y = abs_counts),
                outlier.shape = NA
  ) +
  geom_point(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(grepl("m", CommunityCombo)),
                aes(x = arrival, y = abs_counts, group = Strain, fill = CommunityCombo, color = geq_passed, alpha = geq_passed),
                # color = "black",
                # stroke = 1,
                size = 2,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~spec_name_uniq, scale = "free_x") +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = T, palettefill = "Spectral", setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_arrival/all_drop_boxplot.pdf", width = 20, height = 20)


for (a_strain in c(strains, "ESL0183_ESL0262", "ESL0353_ESL0185")) {
  if (a_strain == "ESL0185") {
    next
  }
  if (a_strain == "ESL0183") {
    next
  }
  if (a_strain == "ESL0197") {
    next
  }
  if (a_strain == "ESL1028") {
    next
  }
  print(a_strain)
  ggplot()  +
    geom_bar(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(Strain == a_strain) %>% filter(arrival %in% c("first", "second", "only")),
                  aes(x = arrival, y = sample_geq),
              position = "identity",
              fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
    geom_beeswarm(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(Strain == a_strain) %>% filter(arrival %in% c("first", "second", "only")),
                  aes(x = arrival, y = abs_counts, fill = Treatment_simp, group = CommunityCombo, color = geq_passed, alpha = geq_passed),
                  size = 2.5,
                  cex = 4,
                  corral = "wrap",
                  corral.width = 0.75,
                  alpha = 0.75,
                  shape = 21,
                  stroke = 1,
                  # color = "black"
    ) +
    # geom_boxplot(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(Strain == a_strain),
    #               aes(x = arrival, y = abs_counts),
    #               outlier.shape = NA
    # ) +
    # geom_point(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(Strain == a_strain),
    #               aes(x = arrival, y = abs_counts, group = Strain, fill = CommunityCombo, color = limit_passed),
    #               # color = "black",
    #               # stroke = 1,
    #               size = 1,
    #               shape = 21,
    #               position = position_dodge(width = .5)
    # ) +
    facet_grid(~CommunityCombo) +
    scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
    scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
    scale_fill_manual(values = TreatmentColors) +
    scale_y_log10(
      breaks = scales::trans_breaks("log10", function(x) 10^x),
      labels = scales::trans_format("log10", scales::math_format(10^.x))
    ) +
    make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
              guide_nrow = 23, leg_pos = "right",
              x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave(paste0("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_arrival/bee_gut_beeswarm_", a_strain, ".pdf"), width = 20, height = 20)
}


for (a_strain in c(strains, "ESL0183_ESL0262", "ESL0353_ESL0185")) {
  if (a_strain == "ESL0185") {
    next
  }
  if (a_strain == "ESL0183") {
    next
  }
  print(a_strain)
  ggplot()  +
    geom_bar(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(Strain == a_strain),
                  aes(x = arrival, y = sample_geq),
              position = "identity",
              fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
    geom_beeswarm(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(Strain == a_strain),
                  aes(x = arrival, y = abs_counts, fill = Treatment_simp, group = CommunityCombo, color = geq_passed, alpha = geq_passed),
                  size = 2.5,
                  cex = 4,
                  corral = "wrap",
                  corral.width = 0.75,
                  alpha = 0.75,
                  shape = 21,
                  stroke = 1,
                  # color = "black"
    ) +
    # geom_boxplot(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(Strain == a_strain),
    #               aes(x = arrival, y = abs_counts),
    #               outlier.shape = NA
    # ) +
    # geom_point(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>% filter(Strain == a_strain),
    #               aes(x = arrival, y = abs_counts, group = Strain, fill = CommunityCombo, color = limit_passed),
    #               # color = "black",
    #               # stroke = 1,
    #               size = 1,
    #               shape = 21,
    #               position = position_dodge(width = .5)
    # ) +
    facet_grid(~CommunityCombo) +
    scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
    scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
    scale_fill_manual(values = TreatmentColors) +
    scale_y_log10(
      breaks = scales::trans_breaks("log10", function(x) 10^x),
      labels = scales::trans_format("log10", scales::math_format(10^.x))
    ) +
    make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
              guide_nrow = 23, leg_pos = "right",
              x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave(paste0("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_arrival/bee_gut_beeswarm_ext_", a_strain, ".pdf"), width = 20, height = 20)
}
```


Plot community profiles. To show colonization success etc.

```{r}
ggplot() +
  geom_bar(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>%
                    filter(Experiment %in% c("1", "2")) %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = rel_counts, fill = factor(Strain, strains_uniq)),
            stat = "identity",
            color = "#000000",
            linewidth = 0.25
            ) +
  facet_wrap(~CommunityCombo + Treatment, scale = "free", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/rel_counts_by_treatment/bee_gut_barplot_rel_counts.pdf", width = 20, height = 20)

ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    filter(Experiment %in% c("1", "2")) %>%
                    filter(Type != "bee") %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = rel_counts, fill = factor(Strain, strains_uniq)),
            stat = "identity",
            color = "#000000",
            linewidth = 0.25
            ) +
  facet_wrap(~CommunityCombo + Treatment, scale = "free", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/rel_counts_by_treatment/bee_gut_barplot_rel_counts_alq.pdf", width = 20, height = 20)

ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    filter(Experiment %in% c("1", "2")) %>%
                    filter(Type != "bee") %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = abs_counts_org, fill = factor(Strain, strains_uniq)),
            stat = "identity",
            color = "#000000",
            linewidth = 0.25
            ) +
  facet_wrap(~CommunityCombo + Treatment, scale = "free", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/rel_counts_by_treatment/bee_gut_barplot_rel_counts_alq_abs_eq.pdf", width = 20, height = 20)

ggplot() +
  geom_bar(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>%
                    filter(Experiment %in% c("1", "2")) %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = abs_counts_org, fill = factor(Strain, strains_uniq)),
            stat = "identity",
            color = "#000000",
            linewidth = 0.25
            ) +
  facet_wrap(~CommunityCombo + Treatment, scale = "free", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/rel_counts_by_treatment/bee_gut_barplot_rel_counts_abs_eq.pdf", width = 20, height = 20)

ggplot() +
  geom_bar(data = df_species_count_all_combined %>% filter(Experiment != "pilot") %>%
                    filter(Experiment %in% c("1", "2")) %>%
                    filter(Type == "bee") %>%
                    filter(Strain != "unknown") %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = rel_counts, fill = factor(Strain, strains_uniq)),
            stat = "identity",
            color = "#000000",
            linewidth = 0.25
            ) +
  facet_wrap(~CommunityCombo + Treatment, scale = "free", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/rel_counts_by_treatment/bee_gut_barplot_rel_counts_wo_unknown.pdf", width = 20, height = 20)


treatment_order_pilot <- c(
  "A-3_14", "A-3", "B-3_14", "B-3",
  "AB-1", "AB-3", "BA-1", "BA-3",
  "alq-A", "alq-AB", "alq-B", "alq-SW",
  "AB-0", "0-3", "A-0", "B-0"
)

ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    filter(!Strain %in% c(strain_without_species_pair, "ESL0183_ESL0262")) %>%
                    # left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("pilot")) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(Type == "bee") %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_pilot), scale = "free_x", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_pilot.pdf", width = 20, height = 20)

ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    filter(!Strain %in% c(strain_without_species_pair, "ESL0183_ESL0262")) %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("pilot")) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(Type == "bee") %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_pilot), scale = "free_x", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_pilot_filt.pdf", width = 20, height = 20)

treatment_order_e1 <- c(
  "A1-0", "A1-B1", "B1-A1", "B1-0",
  "A2-0", "A2-B2", "B2-A2", "B2-0",
  "A3-0", "A3-B3", "B3-A3", "B3-0"
)

ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("1")) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(Type == "bee") %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_e1), scale = "free_x", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_e1.pdf", width = 20, height = 20)


ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("1")) %>%
                    filter(!Strain %in% c(strain_without_species_pair)) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(Type == "bee") %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_e1), scale = "free_x", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_e1_filt.pdf", width = 20, height = 20)


treatment_order_e1_ctrl <- c(
  "A1-0", "A1-B1", "B1-A1", "B1-0",
  "A1", "B1", "A2", "B2",
  "A2-0", "A2-B2", "B2-A2", "B2-0",
  "A3-0", "A3-B3", "B3-A3", "B3-0",
  "A3", "B3", "MD", "extraction_blank"
)

ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("1")) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    # filter(Type == "bee") %>%
                    filter(!Treatment %in% c("SW")),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_e1_ctrl), scale = "free_x", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_e1_ctrl.pdf", width = 20, height = 20)


ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("1")) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(!Strain %in% c(strain_without_species_pair)) %>%
                    # filter(Type == "bee") %>%
                    filter(!Treatment %in% c("SW")),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_e1_ctrl), scale = "free_x", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_e1_filt_ctrl.pdf", width = 20, height = 20)

treatment_order_e2 <- c(
  "A4-0", "A4-B4", "B4-A4", "B4-0",
  "A5-0", "A5-B5", "B5-A5", "B5-0",
  "A6-0", "A6-B6", "B6-A6", "B6-0"
)

ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("2")) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(Type == "bee") %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_e2), scale = "free_x", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_e2.pdf", width = 20, height = 20)


ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("2")) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(Type == "bee") %>%
                    filter(!Strain %in% c(strain_without_species_pair)) %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_e2), scale = "free_x", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_e2_filt.pdf", width = 20, height = 20)

treatment_order_e2_ctrl <- c(
  "A4-0", "A4-B4", "B4-A4", "B4-0",
  "A4", "B4", "A5", "B5",
  "A5-0", "A5-B5", "B5-A5", "B5-0",
  "A6-0", "A6-B6", "B6-A6", "B6-0",
  "A6", "B6", "MD", "extraction_blank"
)

ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("2")) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(!Treatment %in% c(NA)),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_e2_ctrl), scale = "free_x", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_e2_ctrl.pdf", width = 20, height = 20)


ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("2")) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(!Strain %in% c(strain_without_species_pair)) %>%
                    filter(!Treatment %in% c(NA)),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_e2_ctrl), scale = "free_x", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_e2_filt_ctrl.pdf", width = 20, height = 20)


treatment_order_e3 <- c(
  "A1-0", "B1-0",
  "A1-B1", "B1-A1",
  "Am1-B1", "Bm1-A1",
  "Am4-B1", "Bm4-A1",
  "Am6-B1", "Bm6-A1",
  "Am7-B1", "Bm7-A1"
)

ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("3")) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(Type == "bee") %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_e3), scale = "free_x", ncol = 2) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_e3.pdf", width = 20, height = 20)


ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("3")) %>%
                    filter(!Strain %in% c(strain_without_species_pair)) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(Type == "bee") %>%
                    filter(!Treatment %in% c("MD", "extraction_blank")),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_e3), scale = "free_x", ncol = 2) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_e3_filt.pdf", width = 20, height = 20)


treatment_order_e3_ctrl <- c(
  "A1-0", "B1-0",
  "A1-B1", "B1-A1",
  "Am1-B1", "Bm1-A1",
  "Am4-B1", "Bm4-A1",
  "Am6-B1", "Bm6-A1",
  "Am7-B1", "Bm7-A1",
  "A1", "B1",
  "dropout_community",
  "MD", "extraction_blank"
)

ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("3")) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(!Treatment %in% c(NA)),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~factor(Treatment, treatment_order_e3_ctrl)+CommunityCombo, scale = "free_x", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_e3_ctrl.pdf", width = 20, height = 20)


ggplot() +
  geom_bar(data = df_species_count_all_combined %>%
                    left_join(df_species_count_all_combined %>% filter(Experiment != "pilot") %>% select(Strain, CommunityCombo, ID, arrival) %>% distinct(), by = c("Strain", "CommunityCombo", "ID")) %>%
                    filter(Experiment %in% c("3")) %>%
                    filter(!Strain %in% c(strain_without_species_pair)) %>%
                    mutate(abs_counts_org_log = ifelse(abs_counts_org > 0, log10(abs_counts_org), 0)) %>%
                    filter(!Treatment %in% c(NA)),
           aes(x = ID, y = abs_counts_org_log,
               fill = factor(Strain, strains_uniq),
               color = arrival,
               group = factor(Strain, strains_uniq)),
            stat = "identity",
            position = "stack",
            linewidth = 0.5
            ) +
  facet_wrap(~CommunityCombo+factor(Treatment, treatment_order_e3_ctrl), scale = "free_x", ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("first" = "black",
                                "second" = "grey",
                                "both" = "grey",
                                "unadded" = "grey",
                                "only" = "grey")) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", max_colors = 22,
             x_hj = 1, x_vj = 1, y_hj = 1) +
  theme(legend.box.margin = margin())
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_bar_plots/bee_gut_barplot_abs_counts_e3_filt_ctrl.pdf", width = 20, height = 20)

# check 185, 263 arrival order not consistent in e3_filt_ctrl plot

```


#### Community dropout conditions in Experiment 3

```{r}
df_plot_treatments <- df_species_count_all %>%
                        filter(Treatment %in% c("A1-0", "B1-0", "A1-B1", "B1-A1",
                                                "Am1-B1", "Bm1-A1",
                                                "Am4-B1", "Bm4-A1",
                                                "Am6-B1", "Bm6-A1",
                                                "Am7-B1", "Bm7-A1",
                                                "A1", "B1", "MD", "dropout_community", "extraction_blank")) %>%
                        # filter(CommunityCombo %in% c("SW", "1", "m7")) %>%
                        filter(Experiment == "3") %>%
                        filter(spec_name %in% c("Bifidobacterium apousia", "Bifidobacterium sp1.", "Bombilactobacillus mellis", "Lactobacillus apis")) %>%
                        mutate(Type = ifelse(Treatment == "MD", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "blank", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "aliquot", "controls", Type)) %>%
                        mutate(Type = ifelse(Type == "bacteria", "controls", Type)) %>%
                        filter(Type == "bee") %>%
                        mutate(keep = ifelse((spec_name == "Bifidobacterium apousia" & CommunityCombo %in% c("1", "m1")), 1, 0)) %>%
                        mutate(keep = ifelse((spec_name == "Bifidobacterium sp1." & CommunityCombo %in% c("1", "m4")), 1, keep)) %>%
                        mutate(keep = ifelse((spec_name == "Bombilactobacillus mellis" & CommunityCombo %in% c("1", "m6")), 1, keep)) %>%
                        mutate(keep = ifelse((spec_name == "Lactobacillus apis" & CommunityCombo %in% c("1", "m7")), 1, keep)) %>%
                        filter(keep == 1)

pvals_1_les <- df_plot_treatments %>%
              group_by(spec_name, Strain) %>%  # Facet-specific grouping
              filter(Strain %in% comm_B1) %>%
              filter(geq_passed == "1") %>%
              wilcox_test(abs_counts ~ Treatment_simp, alternative = "less",
                          comparisons = list(c("B", "AmB"), c("AmB", "AB"), c("B", "AB"))) %>%
              adjust_pvalue(method = "BH") %>%  # Correct multiple tes
              mutate(p.adj.signif = ifelse(p.adj < 0.05, "*", "ns")) %>%  # Add significance annotation
              mutate(p.signif = ifelse(p < 0.05, "*", "ns")) %>% # Add significance annotation
              mutate(p.signif = ifelse(p < 0.01, "**", p.signif)) %>% # Add significance annotation
              mutate(y.position = case_when(
                group1 == "AmB" & group2 == "B" ~ 8,
                group1 == "AB" & group2 == "AmB" ~ 8.5,
                group1 == "AB" & group2 == "B" ~ 9
              ))

pvals_1_gre <- df_plot_treatments %>%
              group_by(spec_name, Strain) %>%  # Facet-specific grouping
              filter(Strain %in% comm_B1) %>%
              filter(geq_passed == "1") %>%
              wilcox_test(abs_counts ~ Treatment_simp, alternative = "greater",
                          comparisons = list(c("B", "AmB"), c("AmB", "AB"), c("B", "AB"))) %>%
              adjust_pvalue(method = "BH") %>%  # Correct multiple tes
              mutate(p.adj.signif = ifelse(p.adj < 0.05, "*", "ns")) %>%  # Add significance annotation
              mutate(p.signif = ifelse(p < 0.05, "*", "ns")) %>% # Add significance annotation
              mutate(p.signif = ifelse(p < 0.01, "**", p.signif)) %>% # Add significance annotation
              mutate(y.position = case_when(
                group1 == "AmB" & group2 == "B" ~ 8,
                group1 == "AB" & group2 == "AmB" ~ 8.5,
                group1 == "AB" & group2 == "B" ~ 9
              ))

pvals_2_les <- df_plot_treatments %>%
              group_by(spec_name, Strain) %>%  # Facet-specific grouping
              filter(geq_passed == "1") %>%
              filter(Strain != "ESL0827") %>%
              filter(Strain %in% c(comm_A1, "ESL0353_ESL0185")) %>%
              wilcox_test(abs_counts ~ Treatment_simp, alternative = "less",
                          comparisons = list(c("A", "BmA"), c("BmA", "BA"), c("A", "BA"))) %>%
              adjust_pvalue(method = "BH") %>%  # Correct multiple tes
              mutate(p.adj.signif = ifelse(p.adj < 0.05, "*", "ns")) %>%  # Add significance annotation
              mutate(p.signif = ifelse(p < 0.05, "*", "ns")) %>% # Add significance annotation
              mutate(p.signif = ifelse(p < 0.01, "**", p.signif)) %>% # Add significance annotation
              mutate(y.position = case_when(
                group1 == "A" & group2 == "BmA" ~ 8,
                group1 == "BA" & group2 == "BmA" ~ 8.5,
                group1 == "A" & group2 == "BA" ~ 9
              ))

pvals_2_gre <- df_plot_treatments %>%
              group_by(spec_name, Strain) %>%  # Facet-specific grouping
              filter(geq_passed == "1") %>%
              filter(Strain != "ESL0827") %>%
              filter(Strain %in% c(comm_A1, "ESL0353_ESL0185")) %>%
              wilcox_test(abs_counts ~ Treatment_simp, alternative = "greater",
                          comparisons = list(c("A", "BmA"), c("BmA", "BA"), c("A", "BA"))) %>%
              adjust_pvalue(method = "BH") %>%  # Correct multiple tes
              mutate(p.adj.signif = ifelse(p.adj < 0.05, "*", "ns")) %>%  # Add significance annotation
              mutate(p.signif = ifelse(p < 0.05, "*", "ns")) %>% # Add significance annotation
              mutate(p.signif = ifelse(p < 0.01, "**", p.signif)) %>% # Add significance annotation
              mutate(y.position = case_when(
                group1 == "A" & group2 == "BmA" ~ 8,
                group1 == "BA" & group2 == "BmA" ~ 8.5,
                group1 == "A" & group2 == "BA" ~ 9
              ))
       

pvals_1 <- pvals_1_les %>% mutate(comp_type = "less") %>%
          bind_rows(pvals_1_gre %>% mutate(comp_type = "greater"))

pvals_2 <- pvals_2_les %>% mutate(comp_type = "less") %>%
          bind_rows(pvals_2_gre %>% mutate(comp_type = "greater"))

ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = factor(Treatment_simp, c("A", "B","AmB", "BmA", "AB", "BA")), group = ID, y = sample_geq),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_plot_treatments %>% filter(geq_passed == 1),
                aes(x = factor(Treatment_simp, c("A", "B","AmB", "BmA", "AB", "BA")), y = abs_counts, fill = factor(Strain, strains_uniq)),
                outlier.shape = NA
  ) +
  geom_point(data = df_plot_treatments,
                aes(x = factor(Treatment_simp, c("A", "B","AmB", "BmA", "AB", "BA")), y = abs_counts, group = Strain, fill = factor(Strain, strains_uniq), color = geq_passed, alpha = geq_passed),
                # color = "black",
                stroke = 1,
                shape = 21,
                position = position_dodge(width = .5)
  ) +
  facet_wrap(~ spec_name+Strain, scale = "free", nrow = 4) +
  geom_signif(data = pvals_1 %>% filter(p.signif != "ns"), aes(xmin = group1, xmax = group2, annotations = p.signif, y_position = y.position, color = comp_type),
              manual = TRUE,
              inherit.aes = FALSE, tip_length = 0) +
  geom_signif(data = pvals_2 %>% filter(p.signif != "ns"), aes(xmin = group1, xmax = group2, annotations = p.signif, y_position = y.position, color = comp_type),
              manual = TRUE,
              inherit.aes = FALSE, tip_length = 0) +
  # choose strains in df_plot_treatments$Strain for color scale from strainColors
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000", "less" = "orange", "greater" = "darkgreen")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_boxplot_comm_only_drop.pdf", width = 10, height = 15)

# do the same as above but as a beeswarm plot
ggplot() +
  geom_bar(data = df_plot_treatments,
            aes(x = factor(Treatment_simp, c("A", "B","AmB", "BmA", "AB", "BA")), group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments,
                aes(x = factor(Treatment_simp, c("A", "B","AmB", "BmA", "AB", "BA")), y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Strain, scale = "free", nrow = 4) +
  geom_signif(data = pvals_1 %>% filter(p.signif != "ns"), aes(xmin = group1, xmax = group2, annotations = p.signif, y_position = y.position, color = comp_type),
              manual = TRUE,
              inherit.aes = FALSE, tip_length = 0) +
  geom_signif(data = pvals_2 %>% filter(p.signif != "ns"), aes(xmin = group1, xmax = group2, annotations = p.signif, y_position = y.position, color = comp_type),
              manual = TRUE,
              inherit.aes = FALSE, tip_length = 0) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
  ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm__only_drop_all.pdf", width = 10, height = 15)


ggplot() +
  geom_bar(data = df_plot_treatments %>%
                    # filter(Treatment_simp %in% c("AB", "BA", "AmB", "BmA")) %>%
                    group_by(Strain, Treatment_simp) %>%
                    filter(geq_passed == "1") %>%
                    mutate(n = n()) %>%
                    filter(n > 0) %>%
                    ungroup() %>%
                    select(-n),
            aes(x = factor(Treatment_simp, c("AB", "BA", "AmB", "BmA")), group = ID, y = sample_geq),
            position = "identity", linewidth = 0,
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_beeswarm(data = df_plot_treatments %>%
                    # filter(Treatment_simp %in% c("AB", "BA", "AmB", "BmA")) %>%
                    group_by(Strain, Treatment_simp) %>%
                    filter(geq_passed == "1") %>%
                    mutate(n = n()) %>%
                    filter(n > 0) %>%
                    ungroup() %>%
                    select(-n),
                aes(x = factor(Treatment_simp, c("B" ,"A", "AB", "BA", "AmB", "BmA")), y = abs_counts, fill = factor(Strain, strains_uniq), group = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.8,
                shape = 21,
                stroke = 1
                # color = "black"
  ) +
  facet_wrap(~ spec_name+Strain, scale = "free", nrow = 4) +
  geom_signif(data = pvals_1 %>%
                      # filter(group1 %in% c("AB", "BA", "AmB", "BmA")) %>%
                      # filter(group2 %in% c("AB", "BA", "AmB", "BmA")) %>%
                        filter(p.signif != "ns"), aes(xmin = group1, xmax = group2, annotations = p.signif, y_position = y.position, color = comp_type),
              manual = TRUE,
              inherit.aes = FALSE, tip_length = 0) +
  geom_signif(data = pvals_2 %>%
                      # filter(group1 %in% c("AB", "BA", "AmB", "BmA")) %>%
                      # filter(group2 %in% c("AB", "BA", "AmB", "BmA")) %>%
                        filter(p.signif != "ns"), aes(xmin = group1, xmax = group2, annotations = p.signif, y_position = y.position, color = comp_type),
              manual = TRUE,
              inherit.aes = FALSE, tip_length = 0) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000", "less" = "orange", "greater" = "darkgreen")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(theme = theme_bw(), setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/abs_counts_by_treatment/bee_gut_beeswarm_comm__only_drop.pdf", width = 10, height = 15)

p_vals <- pvals_1 %>%
            bind_rows(pvals_2) %>%
            mutate(Comparison = paste(group1, group2, sep = "_"))
write_csv(p_vals, "03_Analysis/Results/tables/p_vals_dropouts.csv")

```

```{r}
# not in use
# df_species_count_medians_arrival <- df_species_count_all_combined %>% filter(Experiment != "pilot") %>%
#                                       group_by(Strain, CommunityCombo, Treatment) %>%
#                                       mutate(num_samples = n_distinct(ID)) %>%
#                                       filter(!is.na(abs_counts)) %>%
#                                       mutate(num_samples_detected = n_distinct(ID)) %>%
#                                       summarize(arrival, mean = mean(abs_counts), sd = sd(abs_counts), median = median(abs_counts), num_samples, num_samples_detected) %>% 
#                                       unique()

# df_species_count_medians_arrival %>% write.csv("03_Analysis/Results/tables/df_species_count_medians_arrival.csv", row.names = F)

# df_species_count_all_combined %>% filter(Experiment != "pilot")_list <- df_species_count_all_combined %>% filter(Experiment != "pilot") %>%
#   group_by(Strain, CommunityCombo, Treatment) %>%
#   mutate(num_samples = n_distinct(ID)) %>%
#   filter(!is.na(abs_counts)) %>%
#   mutate(num_samples_detected = n_distinct(ID)) %>%
#   filter(!is.na(arrival)) %>%
#   filter(!Treatment %in% c("A-1", "B-1", "MD", "AB-0")) %>%
#   filter(!arrival %in% c("both")) %>%
#   ungroup() %>%
#   select(Strain, CommunityCombo, abs_counts_org, arrival) %>%
#   group_by(Strain, CommunityCombo) %>%
#   pivot_wider(id_cols = c(Strain, CommunityCombo), names_from = arrival, values_from = abs_counts_org, values_fn = ~ paste0(.x, collapse = ","))

# write.csv(lapply(df_species_count_all_combined %>% filter(Experiment != "pilot")_list, function(x) if (is.list(x)) unlist(x) else x), "03_Analysis/Results/tables/df_species_count_all_combined %>% filter(Experiment != "pilot")_list.csv", row.names = F)

# view(df_species_count_all_combined %>% filter(Experiment != "pilot"))
```


```{r}
pvals <- df_counts_collected %>%
  filter(!arrival %in% c("dropout")) %>%
  group_by(Strain) %>%
  wilcox_test(counts ~ arrival, data = .) %>%
  # correct p-values for multiple tes
            group1 == "first" & group2 == "second" ~ 11,
            group1 == "first" & group2 == "only" ~ 10.5,
            group1 == "only" & group2 == "second" ~ 10,
            TRUE ~ 10.6
  )) %>% # Prevent overlap
  mutate(p = p.adjust(p, method = "BH")) %>%
  mutate(pval = ifelse(p < 0.05, "*", "ns.")) %>%
  mutate(pval = ifelse(p < 0.01, "**", pval)) %>%
  mutate(pval = ifelse(p < 0.001, "***", pval))
  

pvals_all <- df_counts_collected_all %>%
  filter(!arrival %in% c("only", "dropout")) %>%
  group_by(Strain) %>%
  summarize(pval = wilcox.test(counts ~ arrival, data = ., p.adjust.method = "BH")$p.value) %>%
  mutate(pval = ifelse(pval < 0.05, "*", "")) %>%
  mutate(pval = ifelse(pval < 0.01, "**", pval)) %>%
  mutate(pval = ifelse(pval < 0.001, "***", pval))

ggplot() +
  geom_bar(data = df_counts_collected %>% filter(!arrival %in% c("dropout")),
            aes(x = arrival, y = sample_geq, group = Strain),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_counts_collected %>% filter(geq_passed == "1") %>% filter(!arrival %in% c("dropout")),
                aes(x = arrival, y = counts, fill = Strain),
                linewidth = 0.5,
                outlier.shape = NA
                ) +
  geom_point(data = df_counts_collected %>% filter(!arrival %in% c("dropout")),
                aes(x = arrival, y = counts, fill = Strain, shape = CommunityCombo, color = geq_passed, alpha = geq_passed),
                size = 3, position = position_jitterdodge(jitter.width = 0.25, dodge.width = 0.25)
                ) +
  facet_wrap(~factor(Strain, strains_uniq), ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_shape_manual(values = c(
    "1" = 22,
    "2" = 21,
    "3" = 20,
    "4" = 23,
    "5" = 24,
    "6" = 25,
    "1_rep" = 0
  )) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  geom_signif(data = pvals, manual = TRUE,
              aes(xmin = group1, xmax = group2, annotations = pval, y_position = y.position),
              tip_length = 0.01, step_increase = 0.1
  ) +
  make_theme(setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/priority_effect_strength_compared/counts_by_arrival_per_strain_with_shapes.pdf", width = 20, height = 20)


ggplot() +
  geom_bar(data = df_counts_collected %>% filter(!arrival %in% c("dropout")),
            aes(x = arrival, y = sample_geq, group = Strain),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.05) +
  geom_boxplot(data = df_counts_collected %>% filter(geq_passed == "1") %>% filter(!arrival %in% c("dropout")),
                aes(x = arrival, y = counts, fill = Strain),
                linewidth = 0.5,
                outlier.shape = NA
                ) +
  geom_point(data = df_counts_collected %>% filter(geq_passed != "1") %>% filter(!arrival %in% c("dropout")),
                aes(x = arrival, y = counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5, shape = 21, position = position_jitterdodge(jitter.width = 0.25, dodge.width = 0.25)
                ) +
  geom_point(data = df_counts_collected %>% filter(geq_passed == "1") %>% filter(!arrival %in% c("dropout")),
                aes(x = arrival, y = counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 2.5, shape = 21, position = position_jitterdodge(jitter.width = 0.25, dodge.width = 0.25)
                ) +
  facet_wrap(~strain_name_spec_dict_uniq, ncol = 6) +
  scale_fill_manual(values = strainColorsUniq) +
  scale_color_manual(values = c("0" = "#b9b9b9", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  geom_signif(data = pvals, manual = TRUE,
              aes(xmin = group1, xmax = group2, annotations = pval, y_position = y.position),
              y_position = 1e+06, tip_length = 0.01
  ) +
  make_theme(setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "none",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/priority_effect_strength_compared/counts_by_arrival_per_strain.pdf", width = 10, height = 15)


ggplot() +
  geom_bar(data = df_counts_collected %>% filter(!arrival %in% c("dropout")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
            aes(x = arrival, y = sample_geq, group = Strain),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.1) +
  geom_boxplot(data = df_counts_collected %>% filter(geq_passed == "1") %>% filter(!arrival %in% c("dropout")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
                aes(x = arrival, y = counts, fill = Strain),
                color = "black",
                linewidth = 0.5,
                outlier.shape = NA
                ) +
  geom_point(data = df_counts_collected %>% filter(!arrival %in% c("dropout")) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
                aes(x = arrival, y = counts, fill = Strain, color = geq_passed, alpha = geq_passed),
                size = 0.5
                ) +
  facet_wrap(~factor(Strain, strains_uniq), ncol = 4) +
  scale_fill_manual(values = strainColorsUniq, labels = strain_name_spec_dict_uniq) +
  scale_color_manual(values = c("0" = "#777777", "1" = "#000000")) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(setFill = F, setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right", x_size = 5,
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/priority_effect_strength_compared/counts_by_arrival_per_strain_across_combos.pdf", width = 20, height = 20)

ggplot() +
  geom_bar(data = df_counts_collected %>% filter(!arrival %in% c("dropout")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
            aes(x = CommunityCombo, y = sample_geq, group = arrival_org),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.1) +
  geom_beeswarm(data = df_counts_collected %>% filter(!arrival %in% c("dropout")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
                aes(x = CommunityCombo, y = counts, fill = arrival_org, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.75,
                shape = 21,
                stroke = 1,
                ) +
  labs(y = "log10(absolute abundance)") +
  facet_wrap(~strain_name_spec_dict_uniq, ncol = 4) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(setFill = T, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/priority_effect_strength_compared/counts_by_arrival_per_strain_across_combos_beeswarm.pdf", width = 20, height = 20)


df_pvals_sec <- df_counts_collected_all %>%
                      filter(arrival %in% c("second")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)) %>%
                      # filter(CommunityCombo %in% c("m1", "m4", "m6", "m7", "1_rep")) %>%
                      filter(geq_passed == "1") %>%
                      filter(!Strain %in% c("ESL0183_ESL0262", "ESL0184",
                                            "ESL0200")) %>%
                      mutate(compare_drop = ifelse(grepl("m", CommunityCombo), "dropout", "full")) %>%
                      group_by(Strain) %>%
                      wilcox_test(counts ~ compare_drop, data = ., p.adjust.method = "BH", alternative = "greater"
                      ) %>%
                      mutate(y.position = 9.5) %>%
                      mutate(p.adj = p.adjust(p, method = "BH")) %>%
                      mutate(p.adj.signif = ifelse(p.adj < 0.05, "*", "ns")) %>%
                      mutate(p.adj.signif = ifelse(p.adj < 0.01, "**", p.adj.signif)) %>%
                      mutate(p.adj.signif = ifelse(p.adj < 0.001, "***", p.adj.signif)) %>%
                      filter(p.adj.signif != "ns") %>%
                      mutate(`strain_name_spec_dict_uniq` = strain_name_spec_dict_uniq[Strain])

ggplot() +
  geom_bar(data = df_counts_collected_all %>%
                      mutate(compare_drop = ifelse(grepl("m", CommunityCombo), "dropout", "full")) %>%
                      filter(arrival %in% c("second")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
            aes(x = compare_drop, y = sample_geq, group = arrival),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.1) +
  geom_boxplot(data = df_counts_collected_all %>%
                      mutate(compare_drop = ifelse(grepl("m", CommunityCombo), "dropout", "full")) %>%
                      filter(arrival %in% c("second")) %>%
                      mutate(arrival_org = arrival) %>%
                      filter(geq_passed == "1") %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
                aes(x = compare_drop, y = counts, fill = compare_drop, color = geq_passed, alpha = geq_passed),
                outlier.shape = NA
                ) +
  geom_point(data = df_counts_collected_all %>%
                      mutate(compare_drop = ifelse(grepl("m", CommunityCombo), "dropout", "full")) %>%
                      filter(arrival %in% c("second")) %>%
                      mutate(arrival_org = arrival) %>%
                      filter(geq_passed != "1") %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
                aes(x = compare_drop, y = counts, fill = compare_drop, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                alpha = 0.75,
                shape = 21,
                stroke = 1,
                ) +
  geom_point(data = df_counts_collected_all %>%
                      mutate(compare_drop = ifelse(grepl("m", CommunityCombo), "dropout", "full")) %>%
                      filter(arrival %in% c("second")) %>%
                      mutate(arrival_org = arrival) %>%
                      filter(geq_passed == "1") %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
                aes(x = compare_drop, y = counts, fill = compare_drop, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                alpha = 0.75,
                shape = 21,
                stroke = 1,
                ) +
  labs(y = "log10(absolute abundance)") +
  geom_signif(data = df_pvals_sec, manual = TRUE,
              aes(xmin = group1, xmax = group2, annotations = p.adj.signif, y_position = y.position),
              tip_length = 0.01, inherit.aes = FALSE
  ) +
  facet_wrap(~strain_name_spec_dict_uniq, ncol = 4) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(setFill = T, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/priority_effect_strength_compared/counts_by_arrival_per_strain_dropout_compare.pdf", width = 20, height = 20)

df_counts_collected_all %>%
  filter(arrival %in% c("second")) %>%
  group_by(Strain, arrival) %>%
  filter(!Strain %in% c("ESL0183_ESL0262", "ESL0184", "ESL0819", "ESL0835", "ESL0261",
                                            "ESL0200")) %>%
  filter(geq_passed == "1") %>%
  summarise(n = n(), .groups = "drop") %>%
  pivot_wider(names_from = arrival, values_from = n)

df_pvals_sec_sep_gre <- df_counts_collected_all %>%
                      filter(arrival %in% c("second")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)) %>%
                      filter(CommunityCombo %in% c("1_rep", "m1", "m4", "m6", "m7")) %>%
                      filter(geq_passed == "1") %>%
                      # filter(!Strain %in% c("ESL0183_ESL0262", "ESL0184", "ESL0819", "ESL0835", "ESL0261",
                      #                       "ESL0200")) %>%

                      mutate(compare_drop = ifelse(grepl("m", CommunityCombo), "dropout", "full")) %>%
                      group_by(Strain) %>%
                      mutate(num = n_distinct(arrival)) %>%
                      filter(num > 1) %>%
                      wilcox_test(counts ~ arrival, data = ., p.adjust.method = "BH", alternative = "greater"
                      ) %>%
                      ungroup() %>%
                      mutate(
                        # standardize pair order to avoid duplicates like A-B and B-A
                        pair_id = paste(pmin(group1, group2), pmax(group1, group2), sep = "_")
                      ) %>%
                      arrange(pair_id) %>%
                      mutate(
                        pair_rank = dense_rank(pair_id),
                        y.position = 9.5 + (pair_rank %% 5) * 0.2  # cycle compactly through 5 tiers
                      ) %>%
                      # mutate(y.position = 9.5 + (row_number() %% 5) * 0.2) %>%
                      mutate(comparison_type = case_when(
                        grepl("first", group1) & grepl("first", group2) ~ "first",
                        grepl("first", group1) & grepl("second", group2) ~ "first-second",
                        grepl("second", group1) & grepl("first", group2) ~ "first-second",
                        grepl("second", group1) & grepl("second", group2) ~ "second",
                        TRUE ~ "other"
                      )) %>%
                      # filter(p.adj.signif != "ns") %>%
                      mutate(`strain_name_spec_dict_uniq` = strain_name_spec_dict_uniq[Strain])

df_pvals_sec_sep_les <- df_counts_collected_all %>%
                      filter(arrival %in% c("second")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)) %>%
                      filter(CommunityCombo %in% c("1_rep", "m1", "m4", "m6", "m7")) %>%
                      filter(geq_passed == "1") %>%
                      filter(!Strain %in% c("ESL0183_ESL0262", "ESL0184",
                                            "ESL0200")) %>%
                      mutate(compare_drop = ifelse(grepl("m", CommunityCombo), "dropout", "full")) %>%
                      group_by(Strain) %>%
                      mutate(num = n_distinct(arrival)) %>%
                      filter(num > 1) %>%
                      wilcox_test(counts ~ arrival, data = ., p.adjust.method = "BH", alternative = "less"
                      ) %>%
                      ungroup() %>%
                      mutate(
                        # standardize pair order to avoid duplicates like A-B and B-A
                        pair_id = paste(pmin(group1, group2), pmax(group1, group2), sep = "_")
                      ) %>%
                      arrange(pair_id) %>%
                      mutate(
                        pair_rank = dense_rank(pair_id),
                        y.position = 9.5 + (pair_rank %% 5) * 0.2  # cycle compactly through 5 tiers
                      ) %>%
                      # mutate(y.position = 9.5 + (row_number() %% 5) * 0.2) %>%
                      mutate(comparison_type = case_when(
                        grepl("first", group1) & grepl("first", group2) ~ "first",
                        grepl("first", group1) & grepl("second", group2) ~ "first-second",
                        grepl("second", group1) & grepl("first", group2) ~ "first-second",
                        grepl("second", group1) & grepl("second", group2) ~ "second",
                        TRUE ~ "other"
                      )) %>%
                      # filter(p.adj.signif != "ns") %>%
                      mutate(`strain_name_spec_dict_uniq` = strain_name_spec_dict_uniq[Strain])

ggplot() +
  geom_bar(data = df_counts_collected_all %>%
                      filter(CommunityCombo %in% c("1_rep", "m1", "m4", "m6", "m7")) %>%
                      mutate(compare_drop = ifelse(grepl("m", CommunityCombo), "dropout", "full")) %>%
                      filter(arrival %in% c("second", "first")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
            aes(x = arrival, y = sample_geq, group = arrival),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.1) +
  geom_beeswarm(data = df_counts_collected_all %>%
                      filter(CommunityCombo %in% c("1_rep", "m1", "m4", "m6", "m7")) %>%
                      mutate(compare_drop = ifelse(grepl("m", CommunityCombo), "dropout", "full")) %>%
                      filter(arrival %in% c("second", "first")) %>%
                      mutate(arrival_org = arrival) %>%
                      filter(geq_passed != "1") %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
                aes(x = arrival, y = counts, fill = compare_drop, shape = arrival_org, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.75,
                # shape = 21,
                stroke = 1,
                ) +
  geom_beeswarm(data = df_counts_collected_all %>%
                      filter(CommunityCombo %in% c("1_rep", "m1", "m4", "m6", "m7")) %>%
                      mutate(compare_drop = ifelse(grepl("m", CommunityCombo), "dropout", "full")) %>%
                      filter(arrival %in% c("second", "first")) %>%
                      mutate(arrival_org = arrival) %>%
                      filter(geq_passed == "1") %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
                aes(x = arrival, y = counts, shape = arrival_org, fill = compare_drop, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                # alpha = 0.75,
                # shape = 21,
                stroke = 1,
                ) +
  labs(y = "log10(absolute abundance)") +
  # geom_signif(data = df_pvals_sec_sep %>% filter(p.adj.signif != "ns"),
  #             manual = TRUE,
  #             aes(xmin = group1, xmax = group2, annotations = p.adj.signif, y_position = y.position),
  #             tip_length = 0.01, inherit.aes = FALSE
  # ) +
  facet_wrap(~strain_name_spec_dict_uniq, ncol = 4) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_shape_manual(values = c("first" = 21, "second" = 24)) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(setFill = T, palettefill = "Set1",
             setCol = F, x_angle = 45,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/priority_effect_strength_compared/counts_by_arrival_per_strain_beeswarm_exp3_and_comb_1.pdf", width = 20, height = 20)

df_pvals_sec_sep <- df_pvals_sec_sep_gre %>%
                      mutate(test = "greater") %>%
                      bind_rows(df_pvals_sec_sep_les %>%
                                mutate(test = "less"))

write.csv(df_pvals_sec_sep, "03_Analysis/Results/tables/df_pvals_all_vals_dropouts.csv", row.names = F)

ggplot() +
  geom_bar(data = df_counts_collected %>% filter(!arrival %in% c("dropout")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
            aes(x = arrival, y = sample_geq, group = arrival_org),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.1) +
  geom_beeswarm(data = df_counts_collected %>% filter(!arrival %in% c("dropout")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
                aes(x = arrival, y = counts, fill = arrival_org, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.75,
                shape = 21,
                stroke = 1,
                ) +
  labs(y = "log10(absolute abundance)") +
  facet_wrap(~strain_name_spec_dict_uniq, ncol = 4) +
  scale_alpha_manual(values = c("0" = 0.5, "1" = 0.9)) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(setFill = T, palettefill = "Set1",
             setCol = F, x_angle = 45, x_size = 5,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/priority_effect_strength_compared/counts_by_arrival_per_strain_across_combos_beeswarm_sep.pdf", width = 20, height = 20)


ggplot() +
  geom_bar(data = df_counts_collected_all %>% filter(!arrival %in% c("dropout")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
            aes(x = arrival, y = sample_geq, group = arrival_org),
            position = "identity",
            fill = "#b9b9b9", stat = "identity", alpha = 0.1) +
  geom_beeswarm(data = df_counts_collected_all %>% filter(geq_passed != "1") %>% filter(!arrival %in% c("dropout")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
                aes(x = arrival, y = counts, fill = arrival_org, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.75,
                shape = 21,
                stroke = 1,
                ) +
  geom_beeswarm(data = df_counts_collected_all %>% filter(geq_passed == "1")  %>% filter(!arrival %in% c("dropout")) %>%
                      mutate(arrival_org = arrival) %>%
                      mutate(arrival = paste0(arrival, "_", CommunityCombo)),
                aes(x = arrival, y = counts, fill = arrival_org, color = geq_passed, alpha = geq_passed),
                size = 2.5,
                cex = 4,
                corral = "wrap",
                corral.width = 0.75,
                alpha = 0.75,
                shape = 21,
                stroke = 1,
                ) +
  labs(y = "log10(absolute abundance)") +  
  facet_wrap(~strain_name_spec_dict_uniq, ncol = 4) +
  scale_alpha_manual(values = c("0" = 0.1, "1" = 0.9)) +
  scale_color_manual(values = c("0" = "#999999", "1" = "#000000")) +
  scale_y_log10(
    breaks = scales::trans_breaks("log10", function(x) 10^x),
    labels = scales::trans_format("log10", scales::math_format(10^.x))
  ) +
  make_theme(setFill = T, palettefill = "Set1",
             setCol = F, x_angle = 45, x_size = 5,
             guide_nrow = 23, leg_pos = "right",
             x_hj = 1, x_vj = 1, y_hj = 1)
ggsave("03_Analysis/Results/figures/priority_effects_experiment/priority_effect_strength_compared/all_counts_by_arrival_per_strain_across_combos_beeswarm_sep.pdf", width = 20, height = 20)


```